<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />


<title>Cell Cycle Analysis</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol Paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Cell Cycle Analysis</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">11/2/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>02_Protocol/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211228code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211228)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211228code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211228)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtree1923ae36314462857191f55817eed51512a6c060targetblank1923ae3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/1923ae36314462857191f55817eed51512a6c060" target="_blank">1923ae3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtree1923ae36314462857191f55817eed51512a6c060targetblank1923ae3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/1923ae36314462857191f55817eed51512a6c060" target="_blank">1923ae3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Metadata/.DS_Store
    Ignored:    Metadata/cell_annotation_markers/.DS_Store
    Ignored:    Paper/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/Unused/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/seeds/.DS_Store
    Ignored:    data/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/EDI_02_Normalisation_HVG_DimRed.Rmd
    Untracked:  analysis/EDI_03_Batch_correction.Rmd
    Untracked:  analysis/EDI_04_Cell_Cycle.Rmd
    Untracked:  analysis/EDI_05_Annotation.Rmd
    Untracked:  analysis/EDI_06_marker_genes.Rmd
    Untracked:  analysis/EDI_07_DA.Rmd
    Untracked:  analysis/EDI_08_Subclustering.Rmd
    Untracked:  analysis/P2_03_Fresh_only_subclustering.Rmd
    Untracked:  analysis/P2_03_FvC_Annotations.Rmd
    Untracked:  analysis/P2_04_FvC_gene_expression_plot.Rmd
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-28-17h-59m-25s-err.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-28-17h-59m-25s-out.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-28-20h-11m-16s-err.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-28-20h-11m-16s-out.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-28-22h-42m-55s-err.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-28-22h-42m-55s-out.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-29-08h-29m-36s-err.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-29-08h-29m-36s-out.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-29-08h-43m-28s-err.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-29-08h-43m-28s-out.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-29-23h-45m-47s-err.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-29-23h-45m-47s-out.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-30-09h-31m-28s-err.txt
    Untracked:  analysis/logs/EDI_01_Loading_and_Preprocessing.Rmd-2022-01-30-09h-31m-28s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-29-11h-16m-12s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-29-11h-16m-12s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-29-11h-36m-32s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-29-11h-36m-32s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-29-11h-42m-03s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-29-11h-42m-03s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-01h-26m-29s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-01h-26m-29s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-11h-38m-30s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-11h-38m-30s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-11h-58m-52s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-11h-58m-52s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-12h-06m-05s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-12h-06m-05s-out.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-12h-30m-16s-err.txt
    Untracked:  analysis/logs/EDI_02_Normalisation_HVG_DimRed.Rmd-2022-01-30-12h-30m-16s-out.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-29-12h-45m-25s-err.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-29-12h-45m-25s-out.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-29-12h-56m-47s-err.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-29-12h-56m-47s-out.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-29-15h-44m-12s-err.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-29-15h-44m-12s-out.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-30-13h-02m-39s-err.txt
    Untracked:  analysis/logs/EDI_03_Batch_correction.Rmd-2022-01-30-13h-02m-39s-out.txt
    Untracked:  analysis/logs/EDI_03_Cell_Cycle.Rmd-2022-01-29-12h-09m-03s-err.txt
    Untracked:  analysis/logs/EDI_03_Cell_Cycle.Rmd-2022-01-29-12h-09m-03s-out.txt
    Untracked:  analysis/logs/EDI_03_Cell_Cycle.Rmd-2022-01-29-12h-39m-55s-err.txt
    Untracked:  analysis/logs/EDI_03_Cell_Cycle.Rmd-2022-01-29-12h-39m-55s-out.txt
    Untracked:  analysis/logs/EDI_04_Cell_Cycle.Rmd-2022-01-29-15h-17m-36s-err.txt
    Untracked:  analysis/logs/EDI_04_Cell_Cycle.Rmd-2022-01-29-15h-17m-36s-out.txt
    Untracked:  analysis/logs/EDI_04_Cell_Cycle.Rmd-2022-01-30-15h-58m-04s-err.txt
    Untracked:  analysis/logs/EDI_04_Cell_Cycle.Rmd-2022-01-30-15h-58m-04s-out.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-29-15h-21m-59s-err.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-29-15h-21m-59s-out.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-29-15h-44m-29s-err.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-29-15h-44m-29s-out.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-29-15h-45m-42s-err.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-29-15h-45m-42s-out.txt
    Untracked:  analysis/logs/EDI_06_marker_genes.Rmd-2022-01-29-15h-50m-44s-err.txt
    Untracked:  analysis/logs/EDI_06_marker_genes.Rmd-2022-01-29-15h-50m-44s-out.txt
    Untracked:  analysis/logs/EDI_07_DA.Rmd-2022-01-29-15h-53m-12s-err.txt
    Untracked:  analysis/logs/EDI_07_DA.Rmd-2022-01-29-15h-53m-12s-out.txt
    Untracked:  analysis/logs/EDI_07_DA.Rmd-2022-01-29-16h-48m-47s-err.txt
    Untracked:  analysis/logs/EDI_07_DA.Rmd-2022-01-29-16h-48m-47s-out.txt
    Untracked:  analysis/logs/EDI_07_DA.Rmd-2022-01-29-18h-15m-08s-err.txt
    Untracked:  analysis/logs/EDI_07_DA.Rmd-2022-01-29-18h-15m-08s-out.txt
    Untracked:  analysis/logs/EDI_08_Subclustering.Rmd-2022-01-29-18h-16m-26s-err.txt
    Untracked:  analysis/logs/EDI_08_Subclustering.Rmd-2022-01-29-18h-16m-26s-out.txt
    Untracked:  analysis/logs/EDI_08_Subclustering.Rmd-2022-01-29-19h-46m-01s-err.txt
    Untracked:  analysis/logs/EDI_08_Subclustering.Rmd-2022-01-29-19h-46m-01s-out.txt
    Untracked:  analysis/logs/EDI_08_Subclustering.Rmd-2022-01-29-19h-56m-34s-err.txt
    Untracked:  analysis/logs/EDI_08_Subclustering.Rmd-2022-01-29-19h-56m-34s-out.txt

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/EDI_01_Loading_and_Preprocessing.Rmd
    Modified:   analysis/P2_01_data_comparison.Rmd
    Deleted:    analysis/Protocol paper/.Rprofile
    Deleted:    analysis/Protocol paper/.gitattributes
    Deleted:    analysis/Protocol paper/.gitignore
    Deleted:    analysis/Protocol paper/Protocol paper.Rproj
    Deleted:    analysis/Protocol paper/README.md
    Deleted:    analysis/Protocol paper/_workflowr.yml
    Deleted:    analysis/Protocol paper/analysis/_site.yml
    Deleted:    analysis/Protocol paper/analysis/about.Rmd
    Deleted:    analysis/Protocol paper/analysis/index.Rmd
    Deleted:    analysis/Protocol paper/analysis/license.Rmd
    Deleted:    analysis/Protocol paper/code/README.md
    Deleted:    analysis/Protocol paper/docs/.nojekyll
    Deleted:    analysis/Protocol paper/output/README.md
    Deleted:    analysis/Unused/EDI_02_Normalisation_HVG_DimRed.Rmd
    Deleted:    analysis/Unused/EDI_03_Batch_correction.Rmd
    Deleted:    analysis/Unused/EDI_04_Cell_Cycle.Rmd
    Deleted:    analysis/Unused/EDI_05_Annotation.Rmd
    Deleted:    analysis/Unused/EDI_06_marker_genes.Rmd
    Deleted:    analysis/Unused/EDI_07_DA.Rmd
    Deleted:    analysis/Unused/EDI_08_Subclustering.Rmd
    Deleted:    analysis/Unused/P2_03_Subclustering.Rmd
    Deleted:    analysis/Unused/P2_04_Annotated_Fresh_v_Culture.Rmd
    Deleted:    analysis/Unused/P2_05_gene_expression_plot.Rmd
    Modified:   analysis/admin.rmd
    Modified:   analysis/index.Rmd
    Deleted:    code/CF_readin_with_UMIs.Rmd
    Deleted:    code/code_snippets.Rmd
    Modified:   code/general_purpose_code.R
    Deleted:    code/rename_protocols.r
    Deleted:    code/test_muscat.Rmd
    Deleted:    code/utils_muscat.r

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>sce &lt;- readRDS(file=&quot;../data/EOS_Files/EDI_EOS3_sce.rds&quot;)
rownames(sce) &lt;- rowData(sce)$ID # might need this for the analysis</code></pre>
<p>We download the marker genes for the cell cycle stages from a github repository. The data comes in a data frame. We split the genes into lists according to the cell cycle phase.</p>
<pre class="r"><code>cc_file &lt;- getURL(&quot;https://raw.githubusercontent.com/hbc/tinyatlas/master/cell_cycle/Homo_sapiens.csv&quot;) 
cell_cycle_genes &lt;- read.csv(text = cc_file)

head(cell_cycle_genes)</code></pre>
<pre><code>  phase          geneID modified
1  G2/M ENSG00000010292  9/13/17
2  G2/M ENSG00000011426  9/13/17
3  G2/M ENSG00000013810  9/13/17
4  G2/M ENSG00000072571  9/13/17
5  G2/M ENSG00000075218  9/13/17
6  G2/M ENSG00000080986  9/13/17</code></pre>
<pre class="r"><code>s_genes &lt;- cell_cycle_genes %&gt;%
  dplyr::filter(phase == &quot;S&quot;) %&gt;%
  pull(&quot;geneID&quot;)

g2m_genes &lt;- cell_cycle_genes %&gt;%
  dplyr::filter(phase == &quot;G2/M&quot;) %&gt;%
  pull(&quot;geneID&quot;)</code></pre>
</div>
<div id="identify-cell-cycle-phases" class="section level1">
<h1>Identify cell cycle phases</h1>
<p>The function we are using for the identification of the cell cycle is a Seurat function. Seurat is a data format similar to the Single Cell Experiment object we generally use in the other scrips of this analysis. So we convert our Single Cell Experiment object into a Seurat object and run the identification of the cell cycle phase. We save the results to our Single Cell Experiment object as we will be using this object class it in our remaining analysis.</p>
<pre class="r"><code>sce_seurat &lt;- as.Seurat(sce, counts=&quot;counts&quot;, data=&quot;logcounts&quot;)

sce_seurat &lt;- CellCycleScoring(sce_seurat,
                               g2m.features = g2m_genes,
                               s.features = s_genes)

sce$phase &lt;- sce_seurat$Phase</code></pre>
</div>
<div id="visualisation" class="section level1">
<h1>Visualisation</h1>
<div id="coloured-by-clusters-label" class="section level2 tabset">
<h2 class="tabset">Coloured by clusters label</h2>
<p>We show the plots once to inspect the clusters. It could be interesting to see that certain clusters might have been formed based on the similarity of the cell cycle phase.</p>
<pre class="r"><code>cat(&quot;\n\n### UMAP \n\n&quot;)</code></pre>
<div id="umap" class="section level3">
<h3>UMAP</h3>
<pre class="r"><code>plotReducedDim(sce, &quot;UMAP&quot;, colour_by=&quot;kgraph_clusters_harmony&quot;, text_by=&quot;kgraph_clusters_harmony&quot;, point_size=0.5) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/plot%20UMAP-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="coloured-by-cell-cycle-phase" class="section level2 tabset">
<h2 class="tabset">Coloured by cell cycle phase</h2>
<pre class="r"><code>plot_stratified_views(sce, &quot;UMAP&quot;, stratify_by=&quot;Protocol&quot;, colour_by=&quot;phase&quot;, text_by=&quot;kgraph_clusters_harmony&quot;, tabset=&quot;###&quot;, point_size=0.5)</code></pre>
<div id="all" class="section level3">
<h3>All</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20protocol-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="tabib_et_al" class="section level3">
<h3>Tabib_et_al</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20protocol-2.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="sole_boldo_et_al" class="section level3">
<h3>Sole_Boldo_et_al</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20protocol-3.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="he_et_al" class="section level3">
<h3>He_et_al</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20protocol-4.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="fresh" class="section level3">
<h3>Fresh</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20protocol-5.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="coloured-by-protocol-grouped-by-cell-cycle-phase" class="section level2 tabset">
<h2 class="tabset">Coloured by protocol &amp; grouped by cell cycle phase</h2>
<pre class="r"><code>plot_stratified_views(sce, &quot;UMAP&quot;, stratify_by=&quot;phase&quot;, colour_by=&quot;Protocol&quot;, text_by=&quot;kgraph_clusters_harmony&quot;, tabset=&quot;###&quot;, point_size=0.5)</code></pre>
<div id="all-1" class="section level3">
<h3>All</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20phase-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="g1" class="section level3">
<h3>G1</h3>
<pre><code>Warning: Removed 1 rows containing missing values (geom_text_repel).</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20phase-2.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="s" class="section level3">
<h3>S</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20phase-3.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="g2m" class="section level3">
<h3>G2M</h3>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/visualise-phases-umap%20by%20phase-4.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="cluster-phase-composition-stacked-barplot" class="section level1 tabset">
<h1 class="tabset">Cluster-phase composition (stacked barplot)</h1>
<pre class="r"><code># Cluster_50
cat(&quot;\n\n## By Cluster \n&quot;)</code></pre>
<div id="by-cluster" class="section level2">
<h2>By Cluster</h2>
<pre class="r"><code>as.data.frame(colData(sce)) %&gt;%
  group_by(kgraph_clusters_harmony, phase) %&gt;%
  summarise(Freq=n()) %&gt;% 
  ggplot(aes(x=kgraph_clusters_harmony, y=Freq, fill=phase, label=Freq)) +
         geom_bar(stat=&quot;identity&quot;, position=&quot;fill&quot;) +
  labs(title=&quot;Frequency of cell phase by cluster&quot;,
       subtitle=&quot;Labels represent absolute number of cells per group&quot;,
       x=&quot;Cluster&quot;,
       y=&quot;Frequency&quot;) + 
  geom_text(size=3, position = position_fill(vjust=0.5))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;kgraph_clusters_harmony&#39;. You can override using the `.groups` argument.</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/average%20doublet%20score%201-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Protocol
cat(&quot;\n\n## By Protocol \n&quot;)</code></pre>
</div>
<div id="by-protocol" class="section level2">
<h2>By Protocol</h2>
<pre class="r"><code>as.data.frame(colData(sce)) %&gt;%
  group_by(Protocol, phase) %&gt;%
  summarise(Freq=n()) %&gt;%
  ggplot(aes(x=Protocol, y=Freq, fill=phase, label=Freq)) +
         geom_bar(stat=&quot;identity&quot;, position=&quot;fill&quot;) +
  labs(title=&quot;Frequency of cell phase by Protocol&quot;,
       subtitle=&quot;Labels represent absolute number of cells per group&quot;,
       x=&quot;Cluster&quot;,
       y=&quot;Frequency&quot;) + 
  geom_text(size=3, position = position_fill(vjust=0.5))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Protocol&#39;. You can override using the `.groups` argument.</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/average%20doublet%20score%201-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Sample
cat(&quot;\n\n## By Sample \n&quot;)</code></pre>
</div>
<div id="by-sample" class="section level2">
<h2>By Sample</h2>
<pre class="r"><code>as.data.frame(colData(sce)) %&gt;%
  group_by(Sample, phase) %&gt;%
  summarise(Freq=n()) %&gt;%
  ggplot(aes(x=Sample, y=Freq, fill=phase, label=Freq)) +
         geom_bar(stat=&quot;identity&quot;, position=&quot;fill&quot;) +
  labs(title=&quot;Frequency of cell phase by Sample&quot;,
       subtitle=&quot;Labels represent absolute number of cells per group&quot;,
       x=&quot;Cluster&quot;,
       y=&quot;Frequency&quot;) + 
  geom_text(size=3, position = position_fill(vjust=0.5)) +
  theme(axis.text.x = element_text(angle = 45,hjust=1), axis.ticks.x=element_blank())</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Sample&#39;. You can override using the `.groups` argument.</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/average%20doublet%20score%201-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cluster-phase-composition-counts" class="section level1 tabset">
<h1 class="tabset">Cluster-phase composition (counts)</h1>
<pre class="r"><code># Cluster_50
cat(&quot;\n\n## By Cluster \n&quot;)</code></pre>
<div id="by-cluster-1" class="section level2">
<h2>By Cluster</h2>
<pre class="r"><code>as.data.frame(colData(sce)) %&gt;%
  group_by(kgraph_clusters_harmony, phase) %&gt;%
  summarise(Freq=n()) %&gt;% 
  ggplot(aes(x=kgraph_clusters_harmony, y=Freq, fill=phase, label=Freq)) +
         geom_bar(stat=&quot;identity&quot;) +
  labs(title=&quot;Frequency of cell phase by cluster&quot;,
       subtitle=&quot;Labels represent absolute number of cells per group&quot;,
       x=&quot;Cluster&quot;,
       y=&quot;Frequency&quot;) + 
  geom_text(size=3)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;kgraph_clusters_harmony&#39;. You can override using the `.groups` argument.</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/average%20doublet%20score%202-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Protocol
cat(&quot;\n\n## By Protocol \n&quot;)</code></pre>
</div>
<div id="by-protocol-1" class="section level2">
<h2>By Protocol</h2>
<pre class="r"><code>as.data.frame(colData(sce)) %&gt;%
  group_by(Protocol, phase) %&gt;%
  summarise(Freq=n()) %&gt;%
  ggplot(aes(x=Protocol, y=Freq, fill=phase, label=Freq)) +
         geom_bar(stat=&quot;identity&quot;) +
  labs(title=&quot;Frequency of cell phase by Protocol&quot;,
       subtitle=&quot;Labels represent absolute number of cells per group&quot;,
       x=&quot;Cluster&quot;,
       y=&quot;Frequency&quot;) + 
  geom_text(size=3)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Protocol&#39;. You can override using the `.groups` argument.</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/average%20doublet%20score%202-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Sample
cat(&quot;\n\n## By Sample \n&quot;)</code></pre>
</div>
<div id="by-sample-1" class="section level2">
<h2>By Sample</h2>
<pre class="r"><code>as.data.frame(colData(sce)) %&gt;%
  group_by(Sample, phase) %&gt;%
  summarise(Freq=n()) %&gt;%
  ggplot(aes(x=Sample, y=Freq, fill=phase, label=Freq)) +
         geom_bar(stat=&quot;identity&quot;) +
  labs(title=&quot;Frequency of cell phase by Sample&quot;,
       subtitle=&quot;Labels represent absolute number of cells per group&quot;,
       x=&quot;Cluster&quot;,
       y=&quot;Frequency&quot;) + 
  geom_text(size=3) +
  theme(axis.text.x = element_text(angle = 45,hjust=1), axis.ticks.x=element_blank())</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Sample&#39;. You can override using the `.groups` argument.</code></pre>
<p><img src="figure/EDI_04_Cell_Cycle.Rmd/average%20doublet%20score%202-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="save-data" class="section level1">
<h1>Save data</h1>
<pre class="r"><code>rownames(sce) &lt;- rowData(sce)$Symbol

file_path &lt;- &quot;../data/EOS_Files/EDI_EOS4_sce.rds&quot;
saveRDS(sce, file = file_path)
upload_file_to_iSEE(file_path)</code></pre>
<pre><code>Command executed: 

scp -i ~/.ssh/id_imls_servers /Users/dominiquepaul/xRobinson\ Group/02_Protocol/data/EOS_Files/EDI_EOS4_sce.rds dominique@imlspenticton.uzh.ch:/home/Shared/retger/synovial/data/protocol_paper_BBDP/sce/EDI_EOS4_sce___2022-01-30.rds</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] stringr_1.4.0               openxlsx_4.2.5             
 [3] RColorBrewer_1.1-2          scater_1.22.0              
 [5] scuttle_1.4.0               dplyr_1.0.7                
 [7] SingleCellExperiment_1.16.0 SummarizedExperiment_1.24.0
 [9] Biobase_2.54.0              GenomicRanges_1.46.1       
[11] GenomeInfoDb_1.30.0         IRanges_2.28.0             
[13] S4Vectors_0.32.3            BiocGenerics_0.40.0        
[15] MatrixGenerics_1.6.0        matrixStats_0.61.0         
[17] ggplot2_3.3.5               hrbrthemes_0.8.6           
[19] SeuratObject_4.0.4          Seurat_4.0.6               
[21] RCurl_1.98-1.5             

loaded via a namespace (and not attached):
  [1] workflowr_1.7.0           systemfonts_1.0.3        
  [3] plyr_1.8.6                igraph_1.2.10            
  [5] lazyeval_0.2.2            splines_4.1.0            
  [7] BiocParallel_1.28.3       listenv_0.8.0            
  [9] scattermore_0.7           digest_0.6.29            
 [11] htmltools_0.5.2           viridis_0.6.2            
 [13] fansi_1.0.2               magrittr_2.0.1           
 [15] ScaledMatrix_1.2.0        tensor_1.5               
 [17] cluster_2.1.2             ROCR_1.0-11              
 [19] globals_0.14.0            extrafont_0.17           
 [21] extrafontdb_1.0           spatstat.sparse_2.1-0    
 [23] colorspace_2.0-2          ggrepel_0.9.1            
 [25] xfun_0.29                 crayon_1.4.2             
 [27] jsonlite_1.7.3            spatstat.data_2.1-2      
 [29] survival_3.2-13           zoo_1.8-9                
 [31] glue_1.6.0                polyclip_1.10-0          
 [33] gtable_0.3.0              zlibbioc_1.40.0          
 [35] XVector_0.34.0            leiden_0.3.9             
 [37] DelayedArray_0.20.0       BiocSingular_1.10.0      
 [39] Rttf2pt1_1.3.8            future.apply_1.8.1       
 [41] abind_1.4-5               scales_1.1.1             
 [43] DBI_1.1.2                 miniUI_0.1.1.1           
 [45] Rcpp_1.0.8                viridisLite_0.4.0        
 [47] xtable_1.8-4              reticulate_1.22          
 [49] spatstat.core_2.3-2       rsvd_1.0.5               
 [51] htmlwidgets_1.5.4         httr_1.4.2               
 [53] ellipsis_0.3.2            ica_1.0-2                
 [55] farver_2.1.0              pkgconfig_2.0.3          
 [57] sass_0.4.0                uwot_0.1.11              
 [59] deldir_1.0-6              utf8_1.2.2               
 [61] labeling_0.4.2            tidyselect_1.1.1         
 [63] rlang_0.4.12              reshape2_1.4.4           
 [65] later_1.3.0               munsell_0.5.0            
 [67] tools_4.1.0               generics_0.1.1           
 [69] ggridges_0.5.3            evaluate_0.14            
 [71] fastmap_1.1.0             yaml_2.2.1               
 [73] goftest_1.2-3             knitr_1.37               
 [75] fs_1.5.2                  fitdistrplus_1.1-6       
 [77] zip_2.2.0                 purrr_0.3.4              
 [79] RANN_2.6.1                sparseMatrixStats_1.6.0  
 [81] pbapply_1.5-0             future_1.23.0            
 [83] nlme_3.1-153              mime_0.12                
 [85] compiler_4.1.0            beeswarm_0.4.0           
 [87] plotly_4.10.0             png_0.1-7                
 [89] spatstat.utils_2.3-0      tibble_3.1.6             
 [91] bslib_0.3.1               stringi_1.7.6            
 [93] highr_0.9                 gdtools_0.2.3            
 [95] lattice_0.20-45           Matrix_1.4-0             
 [97] vctrs_0.3.8               pillar_1.6.4             
 [99] lifecycle_1.0.1           spatstat.geom_2.3-1      
[101] lmtest_0.9-39             jquerylib_0.1.4          
[103] BiocNeighbors_1.12.0      RcppAnnoy_0.0.19         
[105] data.table_1.14.2         cowplot_1.1.1            
[107] bitops_1.0-7              irlba_2.3.5              
[109] httpuv_1.6.4              patchwork_1.1.1          
[111] R6_2.5.1                  promises_1.2.0.1         
[113] KernSmooth_2.23-20        gridExtra_2.3            
[115] vipor_0.4.5               parallelly_1.30.0        
[117] codetools_0.2-18          MASS_7.3-54              
[119] assertthat_0.2.1          rprojroot_2.0.2          
[121] withr_2.4.3               sctransform_0.3.2        
[123] GenomeInfoDbData_1.2.7    mgcv_1.8-38              
[125] parallel_4.1.0            beachmat_2.10.0          
[127] grid_4.1.0                rpart_4.1-15             
[129] tidyr_1.1.4               DelayedMatrixStats_1.16.0
[131] rmarkdown_2.11            Rtsne_0.15               
[133] git2r_0.29.0              shiny_1.7.1              
[135] ggbeeswarm_0.6.0         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
