<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Marker Genes</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol Paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Marker Genes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>02_Protocol/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211228code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211228)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211228code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211228)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtree5f4cea35c55dabe976c91d654f1dd6c72b777689targetblank5f4cea3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/5f4cea35c55dabe976c91d654f1dd6c72b777689" target="_blank">5f4cea3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtree5f4cea35c55dabe976c91d654f1dd6c72b777689targetblank5f4cea3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/5f4cea35c55dabe976c91d654f1dd6c72b777689" target="_blank">5f4cea3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Metadata/.DS_Store
    Ignored:    Metadata/cell_annotation_markers/.DS_Store
    Ignored:    Paper/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/Supple_table_1.png
    Ignored:    analysis/Unused/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/seeds/.DS_Store
    Ignored:    data/
    Ignored:    output/.DS_Store
    Ignored:    output/plots/.DS_Store

Untracked files:
    Untracked:  Metadata/~$cluster_annotation_map.xlsx
    Untracked:  analysis/FVC_quick_marker.Rmd
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-19h-33m-54s-err.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-19h-33m-54s-out.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-19h-54m-46s-err.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-19h-54m-46s-out.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-19h-55m-02s-err.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-19h-55m-02s-out.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-20h-33m-08s-err.txt
    Untracked:  analysis/logs/EDI_05_Annotation.Rmd-2022-01-31-20h-33m-08s-out.txt
    Untracked:  analysis/logs/EDI_06_marker_genes.Rmd-2022-01-31-20h-40m-14s-err.txt
    Untracked:  analysis/logs/EDI_06_marker_genes.Rmd-2022-01-31-20h-40m-14s-out.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-01-31-15h-32m-54s-err.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-01-31-15h-32m-54s-out.txt
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-01-31-16h-25m-56s-err.txt
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-01-31-16h-25m-56s-out.txt
    Untracked:  analysis/logs/P2_03_Fresh_only_subclustering.Rmd-2022-01-31-16h-32m-06s-err.txt
    Untracked:  analysis/logs/P2_03_Fresh_only_subclustering.Rmd-2022-01-31-16h-32m-06s-out.txt
    Untracked:  analysis/logs/P2_03_FvC_Annotations.Rmd-2022-01-31-16h-43m-33s-err.txt
    Untracked:  analysis/logs/P2_03_FvC_Annotations.Rmd-2022-01-31-16h-43m-33s-out.txt
    Untracked:  analysis/logs/P2_04_FvC_gene_expression_plot.Rmd-2022-01-31-16h-48m-06s-err.txt
    Untracked:  analysis/logs/P2_04_FvC_gene_expression_plot.Rmd-2022-01-31-16h-48m-06s-out.txt
    Untracked:  analysis/logs/index.Rmd-2022-01-31-16h-53m-18s-err.txt
    Untracked:  analysis/logs/index.Rmd-2022-01-31-16h-53m-18s-out.txt
    Untracked:  output/plots/Suppl_Figure5.png
    Untracked:  output/plots/Suppl_table_1.csv
    Untracked:  output/plots/Suppl_table_1.png
    Untracked:  output/plots/Suppl_table_2.csv
    Untracked:  output/plots/Suppl_table_2.png
    Untracked:  output/plots/old/
    Untracked:  output/plots/overview_figure_fvc.png
    Untracked:  output/plots_31Jan/

Unstaged changes:
    Modified:   Metadata/cluster_annotation_map.xlsx
    Modified:   analysis/EDI_01_Loading_and_Preprocessing.Rmd
    Modified:   analysis/EDI_03_Batch_correction.Rmd
    Modified:   analysis/EDI_05_Annotation.Rmd
    Modified:   analysis/P2_01_data_comparison.Rmd
    Modified:   analysis/P2_03_Fresh_only_subclustering.Rmd
    Modified:   analysis/P2_03_FvC_Annotations.Rmd
    Modified:   analysis/admin.rmd
    Modified:   analysis/plots.Rmd
    Modified:   code/general_purpose_code.R
    Deleted:    output/plots/figure3.png
    Deleted:    output/plots/figure3_supplemental.png
    Deleted:    output/plots/figure4.png
    Deleted:    output/plots/figure5.png
    Deleted:    output/plots/figure6.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/EDI_06_marker_genes.Rmd</code>) and HTML (<code>docs/EDI_06_marker_genes.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/analysis/EDI_06_marker_genes.Rmd" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
<td>
Added new clustering after meeting 29-Jan-2022
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Skin-cell-protocol-scRNA/239a294b77f8417881912b9058abc84dad2da0ac/docs/EDI_06_marker_genes.html" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
<td>
Added new clustering after meeting 29-Jan-2022
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="reload-the-data" class="section level1">
<h1>Reload the data</h1>
<pre class="r"><code>sce &lt;- readRDS(file=&quot;../data/EOS_Files/EDI_EOS5_sce.rds&quot;)

colLabels(sce) &lt;- sce$kgraph_clusters_walktrap_harmony</code></pre>
</div>
<div id="umap-coloured-by-clusters-and-protocol" class="section level1 tabset">
<h1 class="tabset">UMAP coloured by clusters and protocol</h1>
<pre class="r"><code>cat(&quot;\n\n## UMAP coloured by cluster \n\n&quot;)</code></pre>
<div id="umap-coloured-by-cluster" class="section level2">
<h2>UMAP coloured by cluster</h2>
<pre class="r"><code>plotReducedDim(sce, &quot;UMAP&quot;, colour_by=&quot;label&quot;, text_by=&quot;label&quot;, point_size=0.5) +
  labs(title=&quot;UMAP plot coloured by clusters&quot;) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/EDI_06_marker_genes.Rmd/Visualisation%20of%20clusters-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Visualisation-of-clusters-1">
Past versions of “Visualisation of clusters-1.png”
</button>
</p>
<div id="fig-no-spaces-Visualisation-of-clusters-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Visualisation of clusters-1.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n## UMAP coloured by Protocol \n\n&quot;)</code></pre>
</div>
<div id="umap-coloured-by-protocol" class="section level2">
<h2>UMAP coloured by Protocol</h2>
<pre class="r"><code>plotReducedDim(sce, &quot;UMAP&quot;, colour_by=&quot;Protocol&quot;, point_size=0.5) +
  labs(title=&quot;UMAP plot coloured by protocol&quot;) + 
  theme_ipsum_rc() +
  get_protocol_colours(levels(sce$Protocol))</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/EDI_06_marker_genes.Rmd/Visualisation%20of%20clusters-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Visualisation-of-clusters-2">
Past versions of “Visualisation of clusters-2.png”
</button>
</p>
<div id="fig-no-spaces-Visualisation-of-clusters-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Visualisation of clusters-2.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>wb &lt;- createWorkbook()
addWorksheet(wb, &quot;Intro&quot;)
writeData(wb, &quot;Intro&quot;, data.frame(&quot;Marker genes obtained using only the highly-variable genes&quot;), colNames=FALSE)</code></pre>
</div>
</div>
<div id="marker-genes-for-unsupervised-clusters" class="section level1">
<h1>Marker Genes for unsupervised clusters</h1>
<p>Marker genes can help us identify clusters as specific cell types and also discover smaller changes between subclusters (e.g. changes in activation or differentiation state)</p>
<div id="marker-genes-for-each-cluster-using-the-t-test" class="section level2 tabset">
<h2 class="tabset">Marker genes for each cluster using the t-test</h2>
<pre class="r"><code>sce_marker_genes &lt;- sce[rowData(sce)$is_hvg,]
rownames(sce_marker_genes) &lt;- rowData(sce_marker_genes)$Symbol

markers.all.up &lt;- findMarkers(sce_marker_genes, 
                              test.type=&quot;t&quot;,
                              direction=&quot;up&quot;, 
                              block=sce_marker_genes$Sample)

for (i in 1:length(unique(colLabels(sce_marker_genes)))){
  
  cat(paste0(&quot;### Cluster &quot;, i, &quot;\n&quot;))
  
  interesting.all.up &lt;- markers.all.up[[i]]
  interesting.all.up &lt;- interesting.all.up[interesting.all.up$Top &lt;=10,]
  logFCs &lt;- getMarkerEffects(interesting.all.up, prefix=&quot;logFC&quot;)
  logFCs[is.na(logFCs)] &lt;- 0 # set all NAs to 0
  
  # write to excel
  sheet_name &lt;- paste0(&quot;ttest_pairs_cluster_&quot;, i)
  addWorksheet(wb, sheet_name)
  writeData(wb, sheet_name, logFCs, rowNames=TRUE)
  
  pheatmap(logFCs, breaks=seq(-3, 3, length.out=101),
           main=paste0(&quot;Cluster &quot;,i, &quot;: Marker genes, corrected for batch effects \n (upregulated only)&quot;)) + 
    theme_ipsum_rc()
  
  cat(&quot;\n\n&quot;)
}</code></pre>
<div id="cluster-1" class="section level3">
<h3>Cluster 1</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-1">
Past versions of “Marker Genes upregulated t-test-1.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-1.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-2" class="section level3">
<h3>Cluster 2</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-2">
Past versions of “Marker Genes upregulated t-test-2.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-2.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-3" class="section level3">
<h3>Cluster 3</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-3">
Past versions of “Marker Genes upregulated t-test-3.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-3.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-4" class="section level3">
<h3>Cluster 4</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-4">
Past versions of “Marker Genes upregulated t-test-4.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-4.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-5" class="section level3">
<h3>Cluster 5</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-5">
Past versions of “Marker Genes upregulated t-test-5.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-5.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-6" class="section level3">
<h3>Cluster 6</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-6">
Past versions of “Marker Genes upregulated t-test-6.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-6.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-7" class="section level3">
<h3>Cluster 7</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-7">
Past versions of “Marker Genes upregulated t-test-7.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-7.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-8" class="section level3">
<h3>Cluster 8</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-8">
Past versions of “Marker Genes upregulated t-test-8.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-8.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-9" class="section level3">
<h3>Cluster 9</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-9">
Past versions of “Marker Genes upregulated t-test-9.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-9.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-10" class="section level3">
<h3>Cluster 10</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-10">
Past versions of “Marker Genes upregulated t-test-10.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-10.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-11" class="section level3">
<h3>Cluster 11</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-11">
Past versions of “Marker Genes upregulated t-test-11.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-11.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-12" class="section level3">
<h3>Cluster 12</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-12">
Past versions of “Marker Genes upregulated t-test-12.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-12.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-13" class="section level3">
<h3>Cluster 13</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-13">
Past versions of “Marker Genes upregulated t-test-13.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-13.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-14" class="section level3">
<h3>Cluster 14</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-14">
Past versions of “Marker Genes upregulated t-test-14.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-14.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-15" class="section level3">
<h3>Cluster 15</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/Marker%20Genes%20upregulated%20t-test-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-Genes-upregulated-t-test-15">
Past versions of “Marker Genes upregulated t-test-15.png”
</button>
</p>
<div id="fig-no-spaces-Marker-Genes-upregulated-t-test-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/Marker Genes upregulated t-test-15.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="heatmap-for-t-test-results" class="section level2">
<h2>Heatmap for t-test results</h2>
<pre class="r"><code>top_genes_ttest &lt;- lapply(markers.all.up, function(marker_set){
  marker_set %&gt;%
    as.data.frame() %&gt;%
    filter(Top &lt;= 5) %&gt;%
    rownames()
  }) %&gt;% 
  unlist(use.names=F) %&gt;% 
  unique()

# Matrix to plot the heatmap of expression values. 
# To avoid a too big plot we only keep every 10th cell
sce_plot &lt;- sce_marker_genes[top_genes_ttest, seq(1, dim(sce_marker_genes)[2], 20)]
# We sort by cluster
sce_plot &lt;- sce_plot[,order(sce_plot$label)]
# Anotations
column_annotations &lt;- data.frame(&quot;Cluster&quot;=factor(sce_plot$label), row.names=colnames(sce_plot))
# clip heatmap values to ensure that colour contrast is high
sce_plot &lt;- assay(sce_plot, &quot;logcounts&quot;)
sce_plot[sce_plot &gt;=3] = 3

as.ggplot(pheatmap(sce_plot, annotation_col=column_annotations,
         show_rownames=TRUE, show_colnames=FALSE, cluster_cols=FALSE, silent=TRUE,
         color=inferno(50))) +
  labs(title=&quot;Gene signatures per cluster identified via t-test&quot;,
       subtitle=&quot;Top 5 differentially expressed genes per cluster; values are log-counts clipped at 3&quot;) +
  theme_ipsum_rc()+ 
  theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),legend.position=&quot;none&quot;,
          panel.background=element_blank(),panel.border=element_blank(),panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),plot.background=element_blank())</code></pre>
<p><img src="figure/EDI_06_marker_genes.Rmd/pheatmap%20for%20t-test%20results-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pheatmap-for-t-test-results-1">
Past versions of “pheatmap for t-test results-1.png”
</button>
</p>
<div id="fig-no-spaces-pheatmap-for-t-test-results-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/pheatmap for t-test results-1.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="marker-genes-for-each-cluster-using-the-wilcox" class="section level2 tabset">
<h2 class="tabset">Marker genes for each cluster using the Wilcox</h2>
<p>The Wilcoxon test makes pairwise comparisons between to groups (clusters) which is proportional to the area und the curve (AUC). The scores represent the likelihood of a random cell of another cluster having a higher expression than the reference cluster. Scores of 0 and 1 indicate that the distributions are completely separated.</p>
<pre class="r"><code>markers.all.up &lt;- findMarkers(sce_marker_genes, 
                              test.type=&quot;wilcox&quot;,
                              pval.type=&quot;any&quot;,
                              direction=&quot;up&quot;, 
                              block=sce_marker_genes$Sample)

for (i in 1:length(unique(colLabels(sce_marker_genes)))){
  
  cat(paste0(&quot;### Cluster &quot;, i, &quot;\n&quot;))
  
  interesting.all.up &lt;- markers.all.up[[i]]
  interesting.all.up &lt;- interesting.all.up[interesting.all.up$Top &lt;=10,]
  logFCs &lt;- getMarkerEffects(interesting.all.up, prefix=&quot;AUC&quot;)
  logFCs[is.na(logFCs)] &lt;- 0 # set all NAs to 0
  
   # write to excel
  sheet_name &lt;- paste0(&quot;wilcox_pairs_cluster_&quot;, i)
  addWorksheet(wb, sheet_name)
  writeData(wb, sheet_name, logFCs, rowNames=TRUE)
  
  pheatmap(logFCs, breaks=seq(0, 1, length.out=101),
           main=paste0(&quot;Cluster &quot;,i, &quot;: Marker genes, corrected for batch effects \n (upregulated only)&quot;)) + 
    theme_ipsum_rc()
  
  cat(&quot;\n\n&quot;)
}</code></pre>
<div id="cluster-1-1" class="section level3">
<h3>Cluster 1</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-1">
Past versions of “marker genes wilcox upregulated-1.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-1.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-2-1" class="section level3">
<h3>Cluster 2</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-2">
Past versions of “marker genes wilcox upregulated-2.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-2.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-3-1" class="section level3">
<h3>Cluster 3</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-3">
Past versions of “marker genes wilcox upregulated-3.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-3.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-4-1" class="section level3">
<h3>Cluster 4</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-4">
Past versions of “marker genes wilcox upregulated-4.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-4.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-5-1" class="section level3">
<h3>Cluster 5</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-5">
Past versions of “marker genes wilcox upregulated-5.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-5.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-6-1" class="section level3">
<h3>Cluster 6</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-6">
Past versions of “marker genes wilcox upregulated-6.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-6.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-7-1" class="section level3">
<h3>Cluster 7</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-7">
Past versions of “marker genes wilcox upregulated-7.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-7.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-8-1" class="section level3">
<h3>Cluster 8</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-8">
Past versions of “marker genes wilcox upregulated-8.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-8.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-9-1" class="section level3">
<h3>Cluster 9</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-9">
Past versions of “marker genes wilcox upregulated-9.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-9.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-10-1" class="section level3">
<h3>Cluster 10</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-10">
Past versions of “marker genes wilcox upregulated-10.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-10.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-11-1" class="section level3">
<h3>Cluster 11</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-11">
Past versions of “marker genes wilcox upregulated-11.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-11.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-12-1" class="section level3">
<h3>Cluster 12</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-12">
Past versions of “marker genes wilcox upregulated-12.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-12.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-13-1" class="section level3">
<h3>Cluster 13</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-13">
Past versions of “marker genes wilcox upregulated-13.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-13.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-14-1" class="section level3">
<h3>Cluster 14</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-14">
Past versions of “marker genes wilcox upregulated-14.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-14.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-15-1" class="section level3">
<h3>Cluster 15</h3>
<p><img src="figure/EDI_06_marker_genes.Rmd/marker%20genes%20wilcox%20upregulated-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-marker-genes-wilcox-upregulated-15">
Past versions of “marker genes wilcox upregulated-15.png”
</button>
</p>
<div id="fig-no-spaces-marker-genes-wilcox-upregulated-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/marker genes wilcox upregulated-15.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="heatmap-for-wilcoxon-test-results" class="section level2">
<h2>Heatmap for Wilcoxon test results</h2>
<pre class="r"><code>top_genes_wilcox &lt;- lapply(markers.all.up, function(marker_set){
  marker_set %&gt;%
    as.data.frame() %&gt;%
    filter(Top &lt;= 5) %&gt;%
    rownames()
}) %&gt;% 
  unlist(use.names=F) %&gt;% 
  unique()

# Matrix to plot the heatmap of expression values. 
# To avoid a too big plot we only keep every 10th cell
sce_plot &lt;- sce_marker_genes[top_genes_wilcox, seq(1, dim(sce_marker_genes)[2], 20)]
# We sort by cluster
sce_plot &lt;- sce_plot[,order(sce_plot$label)]
# Anotations
column_annotations &lt;- data.frame(&quot;Cluster&quot;=factor(sce_plot$label), row.names=colnames(sce_plot))
# clip heatmap values to ensure that colour contrast is high
sce_plot &lt;- assay(sce_plot, &quot;logcounts&quot;)
sce_plot[sce_plot &gt;=3] = 3

as.ggplot(pheatmap(sce_plot, annotation_col=column_annotations,
         show_rownames=TRUE, show_colnames=FALSE, cluster_cols=FALSE, silent=TRUE,
         color=inferno(50))) +
  labs(title=&quot;Gene signatures per cluster identified via wilcox test&quot;,
       subtitle=&quot;Top 5 differentially expressed genes per cluster; values are log-counts clipped at 3&quot;) +
  theme_ipsum_rc()+ 
  theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(),legend.position=&quot;none&quot;,
          panel.background=element_blank(),panel.border=element_blank(),panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),plot.background=element_blank())</code></pre>
<p><img src="figure/EDI_06_marker_genes.Rmd/pheatmap%20for%20wilcox%20results-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pheatmap-for-wilcox-results-1">
Past versions of “pheatmap for wilcox results-1.png”
</button>
</p>
<div id="fig-no-spaces-pheatmap-for-wilcox-results-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/docs/figure/EDI_06_marker_genes.Rmd/pheatmap for wilcox results-1.png" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>filename_marker_results &lt;- &quot;../docs/output/EDI_marker_results_hvg_genes.xlsx&quot;
saveWorkbook(wb, file = filename_marker_results, overwrite = TRUE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] stringr_1.4.0                  dplyr_1.0.7                   
 [3] RColorBrewer_1.1-2             viridis_0.6.2                 
 [5] viridisLite_0.4.0              ggplotify_0.1.0               
 [7] pheatmap_1.0.12                scran_1.22.1                  
 [9] scater_1.22.0                  scuttle_1.4.0                 
[11] openxlsx_4.2.5                 hrbrthemes_0.8.6              
[13] ggplot2_3.3.5                  tidySingleCellExperiment_1.4.0
[15] SingleCellExperiment_1.16.0    SummarizedExperiment_1.24.0   
[17] Biobase_2.54.0                 GenomicRanges_1.46.1          
[19] GenomeInfoDb_1.30.0            IRanges_2.28.0                
[21] S4Vectors_0.32.3               BiocGenerics_0.40.0           
[23] MatrixGenerics_1.6.0           matrixStats_0.61.0            

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0          colorspace_2.0-2         
  [3] ellipsis_0.3.2            rprojroot_2.0.2          
  [5] bluster_1.4.0             XVector_0.34.0           
  [7] BiocNeighbors_1.12.0      fs_1.5.2                 
  [9] rstudioapi_0.13           farver_2.1.0             
 [11] ggrepel_0.9.1             fansi_1.0.2              
 [13] sparseMatrixStats_1.6.0   extrafont_0.17           
 [15] knitr_1.37                jsonlite_1.7.3           
 [17] workflowr_1.7.0           Rttf2pt1_1.3.8           
 [19] cluster_2.1.2             compiler_4.1.0           
 [21] httr_1.4.2                dqrng_0.3.0              
 [23] assertthat_0.2.1          Matrix_1.4-0             
 [25] fastmap_1.1.0             lazyeval_0.2.2           
 [27] limma_3.50.0              cli_3.1.0                
 [29] later_1.3.0               BiocSingular_1.10.0      
 [31] htmltools_0.5.2           tools_4.1.0              
 [33] rsvd_1.0.5                igraph_1.2.10            
 [35] gtable_0.3.0              glue_1.6.0               
 [37] GenomeInfoDbData_1.2.7    Rcpp_1.0.8               
 [39] jquerylib_0.1.4           vctrs_0.3.8              
 [41] extrafontdb_1.0           DelayedMatrixStats_1.16.0
 [43] xfun_0.29                 beachmat_2.10.0          
 [45] lifecycle_1.0.1           irlba_2.3.5              
 [47] statmod_1.4.36            edgeR_3.36.0             
 [49] zlibbioc_1.40.0           scales_1.1.1             
 [51] promises_1.2.0.1          parallel_4.1.0           
 [53] yaml_2.2.1                gridExtra_2.3            
 [55] yulab.utils_0.0.4         gdtools_0.2.3            
 [57] sass_0.4.0                stringi_1.7.6            
 [59] highr_0.9                 ScaledMatrix_1.2.0       
 [61] zip_2.2.0                 BiocParallel_1.28.3      
 [63] rlang_0.4.12              pkgconfig_2.0.3          
 [65] systemfonts_1.0.3         bitops_1.0-7             
 [67] evaluate_0.14             lattice_0.20-45          
 [69] purrr_0.3.4               labeling_0.4.2           
 [71] htmlwidgets_1.5.4         cowplot_1.1.1            
 [73] tidyselect_1.1.1          magrittr_2.0.1           
 [75] R6_2.5.1                  generics_0.1.1           
 [77] metapod_1.2.0             DelayedArray_0.20.0      
 [79] DBI_1.1.2                 pillar_1.6.4             
 [81] whisker_0.4               withr_2.4.3              
 [83] RCurl_1.98-1.5            tibble_3.1.6             
 [85] crayon_1.4.2              utf8_1.2.2               
 [87] plotly_4.10.0             rmarkdown_2.11           
 [89] locfit_1.5-9.4            grid_4.1.0               
 [91] data.table_1.14.2         git2r_0.29.0             
 [93] digest_0.6.29             tidyr_1.1.4              
 [95] httpuv_1.6.4              gridGraphics_0.5-1       
 [97] munsell_0.5.0             beeswarm_0.4.0           
 [99] vipor_0.4.5               bslib_0.3.1              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
