<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-03-14" />

<title>Analysis of heat shock proteins</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol Paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of heat shock proteins</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-03-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>02_Protocol/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211228code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211228)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211228code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211228)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulOptimisedskincellprotocolscRNAtreea5a0eb8156ede71c7fa41b073d4ac641a51992b3targetblanka5a0eb8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/tree/a5a0eb8156ede71c7fa41b073d4ac641a51992b3" target="_blank">a5a0eb8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulOptimisedskincellprotocolscRNAtreea5a0eb8156ede71c7fa41b073d4ac641a51992b3targetblanka5a0eb8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/tree/a5a0eb8156ede71c7fa41b073d4ac641a51992b3" target="_blank">a5a0eb8</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rprofile
    Ignored:    .Rproj.user/
    Ignored:    .gitattributes
    Ignored:    Metadata/.DS_Store
    Ignored:    Metadata/cell_annotation_markers/.DS_Store
    Ignored:    Paper/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/Supple_table_1.png
    Ignored:    analysis/Unused/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/
    Ignored:    output/.DS_Store
    Ignored:    output/plots_paper_png/.DS_Store

Untracked files:
    Untracked:  analysis/logs/HSP_analysis.Rmd-2022-03-14-15h-11m-40s-err.txt
    Untracked:  analysis/logs/HSP_analysis.Rmd-2022-03-14-15h-11m-40s-out.txt
    Untracked:  analysis/logs/HSP_analysis.Rmd-2022-03-14-15h-12m-08s-err.txt
    Untracked:  analysis/logs/HSP_analysis.Rmd-2022-03-14-15h-12m-08s-out.txt

Unstaged changes:
    Modified:   analysis/HSP_analysis.Rmd
    Modified:   analysis/admin.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/HSP_analysis.Rmd</code>) and HTML (<code>docs/HSP_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/analysis/HSP_analysis.Rmd" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
<td>
added heatshock analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/21876325449c12f0117d805b23a7e3e1fb99117f/docs/HSP_analysis.html" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
<td>
added heatshock analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="heat-shock-proteins" class="section level1">
<h1>Heat shock proteins</h1>
<p>Heat shock proteins (HSPs) are specific proteins that are made when cells are briefly exposed to temperatures above their normal growth temperature. The synthesis of HSPs is a universal phenomenon, occurring in all plant and animal species studied, including humans.</p>
<p>Some people have demonstrated that during tissue dissociation, digestion results in a stress response. Some proteins are produced by cells in response to exposure to stressful conditions (Heat shock proteins).</p>
<p>In this analysis we examine whether the genes involved in the heat shock are differentially expressed depending on the protocol used.</p>
<p>The genes for HSPs are commonly also termed HSP[…]. We searched the <a href="https://genomics.senescence.info/genes/query.php?search=HSP">GenAge database</a> for the query term “HSP”.</p>
<pre class="r"><code>hsps &lt;- c(&quot;HSP90AA1&quot;, &quot;HSPA9&quot;, &quot;HSPD1&quot;, &quot;HSPA1A&quot;, &quot;HSPA1B&quot;, &quot;HSPA8&quot;)
cat(hsps)</code></pre>
<pre><code>HSP90AA1 HSPA9 HSPD1 HSPA1A HSPA1B HSPA8</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>sce_integrated &lt;- readRDS(file=&quot;../data/EOS_Files/EDI_EOS5_sce.rds&quot;)
# colnames(colData(sce))
# table(sce_integrated$manual_labels_coarse)

# We remove some cell types due to low expression levels
# t(table(sce_integrated$manual_labels_coarse, sce_integrated$Sample))
removed_cell_types &lt;- c(&quot;Mast cells&quot;, &quot;Sweat gland cells&quot;, &quot;Melanocytes/Schwann cells/Neuronal cells&quot;)
sce &lt;- sce_integrated[,!sce_integrated$manual_labels_coarse  %in% removed_cell_types]</code></pre>
<p>Are all of the genes in our data?</p>
<pre class="r"><code># all hsp genes appear to be present in the data
hsps %in% rowData(sce)$Symbol</code></pre>
<pre><code>[1] TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
</div>
<div id="single-run-of-a-pseudobulk-de" class="section level1">
<h1>Single run of a Pseudobulk DE</h1>
<pre class="r"><code># Fresh Sole_Boldo_et_al He_et_al Tabib_et_al
control_protocol &lt;- &quot;Fresh&quot;
test_protocol &lt;- &quot;Sole_Boldo_et_al&quot;

refactorise &lt;- function(sce_object, column_vector){
  if (length(column_vector) == 1){
    colData(sce_object)[,column_vector] &lt;- factor(colData(sce_object)[, column_vector])
  } else {
    for (col in column_vector){
      colData(sce_object)[,col] &lt;- factor(colData(sce_object)[,col])
    }
  }
  sce_object
}

sce_subset &lt;- sce[,sce$Protocol %in% c(control_protocol, test_protocol)]
sce_subset &lt;- refactorise(sce_subset, c(&quot;Protocol&quot;, &quot;Sample&quot;, &quot;manual_labels_coarse&quot;))

sce_muscat &lt;- prepSCE(sce_subset,
                      kid=&quot;manual_labels_coarse&quot;,
                      gid=&quot;Protocol&quot;,
                      sid=&quot;Sample&quot;,
                      drop=TRUE)</code></pre>
<p>We store number of clusters and number of sample IDs as variables</p>
<pre class="r"><code>nk &lt;- length(kids &lt;- levels(sce_muscat$cluster_id))
ns &lt;- length(sids &lt;- levels(sce_muscat$sample_id))
names(kids) &lt;- kids; names(sids) &lt;- sids

pb &lt;- aggregateData(sce_muscat, assay=&quot;counts&quot;, fun=&quot;sum&quot;, by=c(&quot;cluster_id&quot;, &quot;sample_id&quot;))

(pb_mds &lt;- pbMDS(pb))</code></pre>
<pre><code>Removing 1 rows with all zero counts</code></pre>
<p><img src="figure/HSP_analysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/docs/figure/HSP_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="run-ds-for-all-protocols" class="section level1 tabset">
<h1 class="tabset">Run DS for all Protocols</h1>
<pre class="r"><code>run_ds_topn &lt;- function(control_protocol, test_protocol, genes=hsps){
  sce_subset &lt;- sce[,sce$Protocol %in% c(control_protocol, test_protocol)]
  sce_subset &lt;- refactorise(sce_subset, c(&quot;Protocol&quot;, &quot;Sample&quot;, &quot;manual_labels_coarse&quot;))
  
  sce_muscat &lt;- prepSCE(sce_subset,
                        kid=&quot;manual_labels_coarse&quot;,
                        gid=&quot;Protocol&quot;,
                        sid=&quot;Sample&quot;,
                        drop=TRUE)
  
  nk &lt;- length(kids &lt;- levels(sce_muscat$cluster_id))
  ns &lt;- length(sids &lt;- levels(sce_muscat$sample_id))
  names(kids) &lt;- kids; names(sids) &lt;- sids
  
  pb &lt;- aggregateData(sce_muscat, assay=&quot;counts&quot;, fun=&quot;sum&quot;, by=c(&quot;cluster_id&quot;, &quot;sample_id&quot;), verbose=F)
  res &lt;- pbDS(pb, verbose=F)
  as.ggplot(pbHeatmap(sce_muscat, res, top_n=20)) +
    labs(title=paste0(control_protocol, &quot; vs. &quot;, test_protocol))
}

cat(&quot;\n\n## Fresh vs. Tabib et al \n\n&quot;)</code></pre>
<div id="fresh-vs.-tabib-et-al" class="section level2">
<h2>Fresh vs. Tabib et al</h2>
<pre class="r"><code>run_ds_topn(&quot;Fresh&quot;, &quot;Tabib_et_al&quot;)</code></pre>
<pre><code>Warning: `filter_()` was deprecated in dplyr 0.7.0.
Please use `filter()` instead.
See vignette(&#39;programming&#39;) for more help
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</code></pre>
<p><img src="figure/HSP_analysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/docs/figure/HSP_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n## Fresh vs. Sole-Boldo et al \n\n&quot;)</code></pre>
</div>
<div id="fresh-vs.-sole-boldo-et-al" class="section level2">
<h2>Fresh vs. Sole-Boldo et al</h2>
<pre class="r"><code>run_ds_topn(&quot;Fresh&quot;, &quot;Sole_Boldo_et_al&quot;)</code></pre>
<p><img src="figure/HSP_analysis.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/docs/figure/HSP_analysis.Rmd/unnamed-chunk-8-2.png" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n## Fresh vs. He et al \n\n&quot;)</code></pre>
</div>
<div id="fresh-vs.-he-et-al" class="section level2">
<h2>Fresh vs. He et al</h2>
<pre class="r"><code>run_ds_topn(&quot;Fresh&quot;, &quot;He_et_al&quot;)</code></pre>
<p><img src="figure/HSP_analysis.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-3">
Past versions of unnamed-chunk-8-3.png
</button>
</p>
<div id="fig-unnamed-chunk-8-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/docs/figure/HSP_analysis.Rmd/unnamed-chunk-8-3.png" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="run-ds-for-heatmap-genes" class="section level1 tabset">
<h1 class="tabset">Run DS for heatmap genes</h1>
<pre class="r"><code>run_ds &lt;- function(control_protocol, test_protocol, genes=hsps){
  sce_subset &lt;- sce[,sce$Protocol %in% c(control_protocol, test_protocol)]
  sce_subset &lt;- refactorise(sce_subset, c(&quot;Protocol&quot;, &quot;Sample&quot;, &quot;manual_labels_coarse&quot;))
  
  sce_muscat &lt;- prepSCE(sce_subset,
                        kid=&quot;manual_labels_coarse&quot;,
                        gid=&quot;Protocol&quot;,
                        sid=&quot;Sample&quot;,
                        drop=TRUE)
  
  nk &lt;- length(kids &lt;- levels(sce_muscat$cluster_id))
  ns &lt;- length(sids &lt;- levels(sce_muscat$sample_id))
  names(kids) &lt;- kids; names(sids) &lt;- sids
  
  pb &lt;- aggregateData(sce_muscat, assay=&quot;counts&quot;, fun=&quot;sum&quot;, by=c(&quot;cluster_id&quot;, &quot;sample_id&quot;), verbose=F)
  res &lt;- pbDS(pb, verbose=F)
  
  as.ggplot(pbHeatmap(sce_muscat, res, g = hsps, fdr=1, lfc=0)) +
    labs(title=paste0(control_protocol, &quot; vs. &quot;, test_protocol))
}

cat(&quot;\n\n## Fresh vs. Tabib et al \n\n&quot;)</code></pre>
<div id="fresh-vs.-tabib-et-al-1" class="section level2">
<h2>Fresh vs. Tabib et al</h2>
<pre class="r"><code>run_ds(&quot;Fresh&quot;, &quot;Tabib_et_al&quot;)</code></pre>
<p><img src="figure/HSP_analysis.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/docs/figure/HSP_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n## Fresh vs. Sole-Boldo et al \n\n&quot;)</code></pre>
</div>
<div id="fresh-vs.-sole-boldo-et-al-1" class="section level2">
<h2>Fresh vs. Sole-Boldo et al</h2>
<pre class="r"><code>run_ds(&quot;Fresh&quot;, &quot;Sole_Boldo_et_al&quot;)</code></pre>
<p><img src="figure/HSP_analysis.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/docs/figure/HSP_analysis.Rmd/unnamed-chunk-9-2.png" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n## Fresh vs. He et al \n\n&quot;)</code></pre>
</div>
<div id="fresh-vs.-he-et-al-1" class="section level2">
<h2>Fresh vs. He et al</h2>
<pre class="r"><code>run_ds(&quot;Fresh&quot;, &quot;He_et_al&quot;)</code></pre>
<p><img src="figure/HSP_analysis.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-3">
Past versions of unnamed-chunk-9-3.png
</button>
</p>
<div id="fig-unnamed-chunk-9-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Optimised-skin-cell-protocol-scRNA/blob/21876325449c12f0117d805b23a7e3e1fb99117f/docs/figure/HSP_analysis.Rmd/unnamed-chunk-9-3.png" target="_blank">2187632</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-03-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplotify_0.1.0             ggplot2_3.3.5              
 [3] muscat_1.8.1                SingleCellExperiment_1.16.0
 [5] SummarizedExperiment_1.24.0 Biobase_2.54.0             
 [7] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
 [9] IRanges_2.28.0              S4Vectors_0.32.3           
[11] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[13] matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] backports_1.4.1           circlize_0.4.14          
  [3] blme_1.0-5                workflowr_1.7.0          
  [5] plyr_1.8.6                TMB_1.7.22               
  [7] splines_4.1.0             listenv_0.8.0            
  [9] BiocParallel_1.28.3       scater_1.22.0            
 [11] digest_0.6.29             yulab.utils_0.0.4        
 [13] foreach_1.5.2             htmltools_0.5.2          
 [15] magick_2.7.3              viridis_0.6.2            
 [17] lmerTest_3.1-3            fansi_1.0.2              
 [19] magrittr_2.0.2            memoise_2.0.1            
 [21] ScaledMatrix_1.2.0        cluster_2.1.2            
 [23] doParallel_1.0.17         limma_3.50.1             
 [25] globals_0.14.0            ComplexHeatmap_2.11.1    
 [27] Biostrings_2.62.0         annotate_1.72.0          
 [29] prettyunits_1.1.1         colorspace_2.0-3         
 [31] ggrepel_0.9.1             blob_1.2.2               
 [33] xfun_0.30                 dplyr_1.0.8              
 [35] crayon_1.5.0              RCurl_1.98-1.6           
 [37] jsonlite_1.8.0            genefilter_1.76.0        
 [39] lme4_1.1-28               survival_3.2-13          
 [41] iterators_1.0.14          glue_1.6.2               
 [43] gtable_0.3.0              zlibbioc_1.40.0          
 [45] XVector_0.34.0            GetoptLong_1.0.5         
 [47] DelayedArray_0.20.0       BiocSingular_1.10.0      
 [49] future.apply_1.8.1        shape_1.4.6              
 [51] scales_1.1.1              DBI_1.1.2                
 [53] edgeR_3.36.0              Rcpp_1.0.8               
 [55] viridisLite_0.4.0         xtable_1.8-4             
 [57] progress_1.2.2            clue_0.3-60              
 [59] gridGraphics_0.5-1        bit_4.0.4                
 [61] rsvd_1.0.5                httr_1.4.2               
 [63] gplots_3.1.1              RColorBrewer_1.1-2       
 [65] ellipsis_0.3.2            farver_2.1.0             
 [67] pkgconfig_2.0.3           XML_3.99-0.9             
 [69] scuttle_1.4.0             sass_0.4.0               
 [71] locfit_1.5-9.5            utf8_1.2.2               
 [73] labeling_0.4.2            reshape2_1.4.4           
 [75] tidyselect_1.1.2          rlang_1.0.2              
 [77] later_1.3.0               AnnotationDbi_1.56.2     
 [79] munsell_0.5.0             tools_4.1.0              
 [81] cachem_1.0.6              cli_3.2.0                
 [83] generics_0.1.2            RSQLite_2.2.10           
 [85] broom_0.7.12              evaluate_0.15            
 [87] stringr_1.4.0             fastmap_1.1.0            
 [89] yaml_2.3.5                knitr_1.37               
 [91] bit64_4.0.5               fs_1.5.2                 
 [93] caTools_1.18.2            purrr_0.3.4              
 [95] KEGGREST_1.34.0           future_1.24.0            
 [97] nlme_3.1-155              sparseMatrixStats_1.6.0  
 [99] whisker_0.4               pbkrtest_0.5.1           
[101] compiler_4.1.0            rstudioapi_0.13          
[103] beeswarm_0.4.0            png_0.1-7                
[105] variancePartition_1.24.0  tibble_3.1.6             
[107] geneplotter_1.72.0        bslib_0.3.1              
[109] stringi_1.7.6             highr_0.9                
[111] lattice_0.20-45           Matrix_1.4-0             
[113] nloptr_2.0.0              vctrs_0.3.8              
[115] pillar_1.7.0              lifecycle_1.0.1          
[117] jquerylib_0.1.4           GlobalOptions_0.1.2      
[119] BiocNeighbors_1.12.0      data.table_1.14.2        
[121] bitops_1.0-7              irlba_2.3.5              
[123] httpuv_1.6.5              R6_2.5.1                 
[125] promises_1.2.0.1          KernSmooth_2.23-20       
[127] gridExtra_2.3             parallelly_1.30.0        
[129] vipor_0.4.5               codetools_0.2-18         
[131] gtools_3.9.2              boot_1.3-28              
[133] MASS_7.3-55               assertthat_0.2.1         
[135] DESeq2_1.34.0             rprojroot_2.0.2          
[137] rjson_0.2.21              withr_2.5.0              
[139] sctransform_0.3.3         GenomeInfoDbData_1.2.7   
[141] parallel_4.1.0            hms_1.1.1                
[143] grid_4.1.0                beachmat_2.10.0          
[145] tidyr_1.2.0               glmmTMB_1.1.2.3          
[147] minqa_1.2.4               rmarkdown_2.11           
[149] DelayedMatrixStats_1.16.0 Cairo_1.5-14             
[151] git2r_0.29.0              numDeriv_2016.8-1.1      
[153] ggbeeswarm_0.6.0         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggplotify_0.1.0             ggplot2_3.3.5              
 [3] muscat_1.8.1                SingleCellExperiment_1.16.0
 [5] SummarizedExperiment_1.24.0 Biobase_2.54.0             
 [7] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
 [9] IRanges_2.28.0              S4Vectors_0.32.3           
[11] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[13] matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] backports_1.4.1           circlize_0.4.14          
  [3] blme_1.0-5                workflowr_1.7.0          
  [5] plyr_1.8.6                TMB_1.7.22               
  [7] splines_4.1.0             listenv_0.8.0            
  [9] BiocParallel_1.28.3       scater_1.22.0            
 [11] digest_0.6.29             yulab.utils_0.0.4        
 [13] foreach_1.5.2             htmltools_0.5.2          
 [15] magick_2.7.3              viridis_0.6.2            
 [17] lmerTest_3.1-3            fansi_1.0.2              
 [19] magrittr_2.0.2            memoise_2.0.1            
 [21] ScaledMatrix_1.2.0        cluster_2.1.2            
 [23] doParallel_1.0.17         limma_3.50.1             
 [25] globals_0.14.0            ComplexHeatmap_2.11.1    
 [27] Biostrings_2.62.0         annotate_1.72.0          
 [29] prettyunits_1.1.1         colorspace_2.0-3         
 [31] ggrepel_0.9.1             blob_1.2.2               
 [33] xfun_0.30                 dplyr_1.0.8              
 [35] crayon_1.5.0              RCurl_1.98-1.6           
 [37] jsonlite_1.8.0            genefilter_1.76.0        
 [39] lme4_1.1-28               survival_3.2-13          
 [41] iterators_1.0.14          glue_1.6.2               
 [43] gtable_0.3.0              zlibbioc_1.40.0          
 [45] XVector_0.34.0            GetoptLong_1.0.5         
 [47] DelayedArray_0.20.0       BiocSingular_1.10.0      
 [49] future.apply_1.8.1        shape_1.4.6              
 [51] scales_1.1.1              DBI_1.1.2                
 [53] edgeR_3.36.0              Rcpp_1.0.8               
 [55] viridisLite_0.4.0         xtable_1.8-4             
 [57] progress_1.2.2            clue_0.3-60              
 [59] gridGraphics_0.5-1        bit_4.0.4                
 [61] rsvd_1.0.5                httr_1.4.2               
 [63] gplots_3.1.1              RColorBrewer_1.1-2       
 [65] ellipsis_0.3.2            farver_2.1.0             
 [67] pkgconfig_2.0.3           XML_3.99-0.9             
 [69] scuttle_1.4.0             sass_0.4.0               
 [71] locfit_1.5-9.5            utf8_1.2.2               
 [73] labeling_0.4.2            reshape2_1.4.4           
 [75] tidyselect_1.1.2          rlang_1.0.2              
 [77] later_1.3.0               AnnotationDbi_1.56.2     
 [79] munsell_0.5.0             tools_4.1.0              
 [81] cachem_1.0.6              cli_3.2.0                
 [83] generics_0.1.2            RSQLite_2.2.10           
 [85] broom_0.7.12              evaluate_0.15            
 [87] stringr_1.4.0             fastmap_1.1.0            
 [89] yaml_2.3.5                knitr_1.37               
 [91] bit64_4.0.5               fs_1.5.2                 
 [93] caTools_1.18.2            purrr_0.3.4              
 [95] KEGGREST_1.34.0           future_1.24.0            
 [97] nlme_3.1-155              sparseMatrixStats_1.6.0  
 [99] whisker_0.4               pbkrtest_0.5.1           
[101] compiler_4.1.0            rstudioapi_0.13          
[103] beeswarm_0.4.0            png_0.1-7                
[105] variancePartition_1.24.0  tibble_3.1.6             
[107] geneplotter_1.72.0        bslib_0.3.1              
[109] stringi_1.7.6             highr_0.9                
[111] lattice_0.20-45           Matrix_1.4-0             
[113] nloptr_2.0.0              vctrs_0.3.8              
[115] pillar_1.7.0              lifecycle_1.0.1          
[117] jquerylib_0.1.4           GlobalOptions_0.1.2      
[119] BiocNeighbors_1.12.0      data.table_1.14.2        
[121] bitops_1.0-7              irlba_2.3.5              
[123] httpuv_1.6.5              R6_2.5.1                 
[125] promises_1.2.0.1          KernSmooth_2.23-20       
[127] gridExtra_2.3             parallelly_1.30.0        
[129] vipor_0.4.5               codetools_0.2-18         
[131] gtools_3.9.2              boot_1.3-28              
[133] MASS_7.3-55               assertthat_0.2.1         
[135] DESeq2_1.34.0             rprojroot_2.0.2          
[137] rjson_0.2.21              withr_2.5.0              
[139] sctransform_0.3.3         GenomeInfoDbData_1.2.7   
[141] parallel_4.1.0            hms_1.1.1                
[143] grid_4.1.0                beachmat_2.10.0          
[145] tidyr_1.2.0               glmmTMB_1.1.2.3          
[147] minqa_1.2.4               rmarkdown_2.11           
[149] DelayedMatrixStats_1.16.0 Cairo_1.5-14             
[151] git2r_0.29.0              numDeriv_2016.8-1.1      
[153] ggbeeswarm_0.6.0         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
