<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Fresh vs. Culture: Gene Expression Plot and DA</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol Paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Fresh vs. Culture: Gene Expression Plot and DA</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>02_Protocol/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211228code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211228)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211228code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211228)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtree97b097bcd93105d275ebb47c066a8129eb61b5b2targetblank97b097ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/97b097bcd93105d275ebb47c066a8129eb61b5b2" target="_blank">97b097b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtree97b097bcd93105d275ebb47c066a8129eb61b5b2targetblank97b097ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/97b097bcd93105d275ebb47c066a8129eb61b5b2" target="_blank">97b097b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Metadata/.DS_Store
    Ignored:    Metadata/cell_annotation_markers/.DS_Store
    Ignored:    Paper/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/Unused/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/seeds/.DS_Store
    Ignored:    data/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-01-30-19h-37m-35s-err.txt
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-01-30-19h-37m-35s-out.txt
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-01-30-21h-05m-37s-err.txt
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-01-30-21h-05m-37s-out.txt
    Untracked:  analysis/logs/P2_03_Fresh_only_subclustering.Rmd-2022-01-30-19h-48m-49s-err.txt
    Untracked:  analysis/logs/P2_03_Fresh_only_subclustering.Rmd-2022-01-30-19h-48m-49s-out.txt
    Untracked:  analysis/logs/P2_03_FvC_Annotations.Rmd-2022-01-30-20h-02m-55s-err.txt
    Untracked:  analysis/logs/P2_03_FvC_Annotations.Rmd-2022-01-30-20h-02m-55s-out.txt
    Untracked:  analysis/logs/P2_03_FvC_Annotations.Rmd-2022-01-30-21h-05m-51s-err.txt
    Untracked:  analysis/logs/P2_03_FvC_Annotations.Rmd-2022-01-30-21h-05m-51s-out.txt
    Untracked:  analysis/logs/P2_04_FvC_gene_expression_plot.Rmd-2022-01-30-21h-15m-52s-err.txt
    Untracked:  analysis/logs/P2_04_FvC_gene_expression_plot.Rmd-2022-01-30-21h-15m-52s-out.txt
    Untracked:  output/plots/figure3.png
    Untracked:  output/plots/figure3_supplemental.png
    Untracked:  output/plots/figure4.png
    Untracked:  output/plots/figure5.png
    Untracked:  output/plots/figure6.png

Unstaged changes:
    Modified:   analysis/P2_03_FvC_Annotations.Rmd
    Modified:   analysis/admin.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/P2_04_FvC_gene_expression_plot.Rmd</code>) and HTML (<code>docs/P2_04_FvC_gene_expression_plot.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/analysis/P2_04_FvC_gene_expression_plot.Rmd" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
<td>
Added new clustering after meeting 29-Jan-2022
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<p>We load the data for the two ‘Fresh’ samples and four SSc skin samples that we have processed in the previous script. All six samples were jointly batch-corrected.</p>
<pre class="r"><code>sce &lt;- readRDS(file=&quot;../data/EOS_files/FvC_EOS3.rds&quot;)

sce$manual_labels_coarse &lt;- sub(&quot;Vascular endothelial cells&quot;, &quot;Vascular endothelial&quot;, sce$manual_labels_coarse)
sce$manual_labels_coarse &lt;- factor(sce$manual_labels_coarse)
unique(sce$manual_labels_coarse)</code></pre>
<pre><code>[1] Schwann cells         Keratinocytes         Fibroblasts          
[4] Macrophages           Pericytes/VSMC        Dendritic cells      
[7] Vascular endothelial  T cells               Lymphatic endothelial
9 Levels: Dendritic cells Fibroblasts Keratinocytes ... Vascular endothelial</code></pre>
</div>
<div id="plot-comparing-expression-values-for-selected-genes" class="section level1">
<h1>Plot comparing expression values for selected genes</h1>
<p>Frac_cells_expressed_in = Number of e.g. Keratinocyte cells in sample X with an expression value &gt; 0 for the given gene / Number of keratinocyte cells in sample X</p>
<p>Expr_avg_to_overall_avg = Average expression value of gene in e.g. keratinocytes of Sample X</p>
<pre class="r"><code>marker_overview &lt;- read.xlsx(xlsxFile=&quot;../Metadata/Special_plot_markers.xlsx&quot;)</code></pre>
<pre class="r"><code># create initial data frame
genes_ball_plot &lt;- marker_overview %&gt;%
  dplyr::rename(cell_type=Cell.type, gene=Marker)

cat(&quot;Genes not found in data: &quot;)</code></pre>
<pre><code>Genes not found in data: </code></pre>
<pre class="r"><code>cat( genes_ball_plot$gene[!genes_ball_plot$gene %in% rownames(sce)])
genes_ball_plot &lt;- genes_ball_plot[genes_ball_plot$gene %in% rownames(sce),]


genes_ball_plot_with_values &lt;- lapply(unique(sce$Sample), function(sample){
  df_intermediate_level1 &lt;- genes_ball_plot
  # add the sample name as a column
  df_intermediate_level1$sample &lt;- sample
  
  # for loop that uses only the subset of the sce for the cell type
  df_intermediate_level1 &lt;- lapply(unique(genes_ball_plot$cell_type), function(cell_type){
    cat(paste0(&quot;    &quot;, cell_type, &quot;\n\n&quot;))
    genes_for_cell_type &lt;- genes_ball_plot[genes_ball_plot$cell_type == cell_type, &quot;gene&quot;]
    df_intermediate_level2 &lt;- df_intermediate_level1[df_intermediate_level1$cell_type == cell_type,]
    sce_subset &lt;- sce[,sce$Sample == sample &amp; sce$manual_labels_coarse == cell_type]
    sce_anti_subset &lt;- sce[,sce$Sample == sample &amp; sce$manual_labels_coarse != cell_type]
    # check whether the genes really occur in the sample
    mask &lt;- genes_for_cell_type %in% rownames(sce_subset)
    df_intermediate_level2$mean_of_logs_cell_type_only &lt;- 0
    df_intermediate_level2$frac_cells_expressed_in_cell_type_only &lt;- 0
    
    if (dim(sce_subset)[2] &gt; 0){ # check that the sample is not empty
      if (length(genes_for_cell_type) &gt; 1){ # if there is only one gene, then the rowMeans function needs to be replaced by mean
        if (dim(sce_subset)[2] &gt; 1){ # if there is only one cell then we dont take averages
          df_intermediate_level2[mask, &quot;mean_of_logs_cell_type_only&quot;] &lt;- rowMeans(logcounts(sce_subset)[genes_for_cell_type[mask],])
          df_intermediate_level2[mask, &quot;frac_cells_expressed_in_cell_type_only&quot;] &lt;- rowMeans(assay(sce_subset)[genes_for_cell_type[mask],] &gt; 0)
        } else {
          cat(paste0(&quot;Warning: &quot;, sample, &quot; only has one cell of type &quot;, cell_type))
          df_intermediate_level2[mask, &quot;mean_of_logs_cell_type_only&quot;] &lt;- logcounts(sce_subset)[genes_for_cell_type[mask],]
          df_intermediate_level2[mask, &quot;frac_cells_expressed_in_cell_type_only&quot;] &lt;- as.numeric(assay(sce_subset)[genes_for_cell_type[mask],] &gt; 0)
        }
      } else {
        df_intermediate_level2[mask, &quot;mean_of_logs_cell_type_only&quot;] &lt;- mean(logcounts(sce_subset)[genes_for_cell_type[mask],])
        df_intermediate_level2[mask, &quot;frac_cells_expressed_in_cell_type_only&quot;] &lt;- mean(assay(sce_subset)[genes_for_cell_type[mask],] &gt; 0)
      }
    }
    
    if (length(genes_for_cell_type) &gt; 1){ # if there is only one gene, then the rowMeans function needs to be replaced by mean
      df_intermediate_level2$avg_expression_other_cells &lt;- rowMeans(logcounts(sce_anti_subset)[genes_for_cell_type,])
    } else {
      df_intermediate_level2$avg_expression_other_cells &lt;- mean(logcounts(sce_anti_subset)[genes_for_cell_type,])
    }
    
    df_intermediate_level2
  }) %&gt;% ldply(data.frame)

  df_intermediate_level1
  
}) %&gt;% ldply(data.frame)</code></pre>
<pre><code>    Fibroblasts

    Keratinocytes

    Vascular endothelial

    Pericytes/VSMC

    T cells

    Lymphatic endothelial

    Dendritic cells

    Macrophages

    Mast cells

    Melanocytes

    Schwann cells

    Sweat gland cells

    Fibroblasts

    Keratinocytes

    Vascular endothelial

    Pericytes/VSMC

    T cells

Warning: Culture_S1 only has one cell of type T cells    Lymphatic endothelial

    Dendritic cells

    Macrophages

    Mast cells

    Melanocytes

    Schwann cells

    Sweat gland cells

    Fibroblasts

    Keratinocytes

    Vascular endothelial

    Pericytes/VSMC

    T cells

    Lymphatic endothelial

    Dendritic cells

    Macrophages

    Mast cells

    Melanocytes

    Schwann cells

    Sweat gland cells

    Fibroblasts

    Keratinocytes

    Vascular endothelial

    Pericytes/VSMC

    T cells

    Lymphatic endothelial

    Dendritic cells

    Macrophages

    Mast cells

    Melanocytes

    Schwann cells

    Sweat gland cells

    Fibroblasts

    Keratinocytes

    Vascular endothelial

    Pericytes/VSMC

    T cells

    Lymphatic endothelial

    Dendritic cells

    Macrophages

    Mast cells

    Melanocytes

    Schwann cells

    Sweat gland cells

    Fibroblasts

    Keratinocytes

    Vascular endothelial

    Pericytes/VSMC

    T cells

    Lymphatic endothelial

    Dendritic cells

    Macrophages

    Mast cells

    Melanocytes

    Schwann cells

    Sweat gland cells</code></pre>
<pre class="r"><code># add relative column
genes_ball_plot_with_values$expr_avg_to_overall_avg &lt;- genes_ball_plot_with_values$mean_of_logs_cell_type_only / genes_ball_plot_with_values$avg_expression_other_cells
genes_ball_plot_with_values[is.na(genes_ball_plot_with_values$expr_avg_to_overall_avg), &quot;expr_avg_to_overall_avg&quot;] &lt;- 0
genes_ball_plot_with_values[genes_ball_plot_with_values$expr_avg_to_overall_avg &gt; 50, &quot;expr_avg_to_overall_avg&quot;] &lt;- 50

# add protocol as a column
sample2protocol &lt;- data.frame(&quot;sample&quot;=levels(colData(sce)$Sample)) %&gt;%
  mutate(&quot;Protocol&quot;=sub(&quot;_S[0-9]*&quot;, &quot;&quot;, sample)) %&gt;%
  mutate(&quot;Protocol&quot;=sub(&quot;Skin&quot;,&quot;SSc&quot;, Protocol))
  
genes_ball_plot_with_values &lt;- genes_ball_plot_with_values %&gt;%
  left_join(sample2protocol)</code></pre>
<pre><code>Joining, by = &quot;sample&quot;</code></pre>
<pre class="r"><code># use this to normalise the colour scale
# col_norm_const &lt;- max(genes_ball_plot_with_values$mean_of_logs_cell_type_only )

# fix cell type column as factor to ensure correct ordering
genes_ball_plot_with_values$cell_type &lt;- factor(genes_ball_plot_with_values$cell_type, levels=unique(genes_ball_plot$cell_type))</code></pre>
<pre class="r"><code>folder_path &lt;- &quot;/Users/dominiquepaul/xRobinson Group/01_Systemic Sclerosis/output/plots/&quot;

# Change order of Sample levels
genes_ball_plot_with_values$sample &lt;- factor(genes_ball_plot_with_values$sample, levels=c(&quot;Culture_S5&quot;, &quot;Culture_S4&quot;, &quot;Culture_S2&quot;, &quot;Culture_S1&quot;,&quot;Fresh_S2&quot;, &quot;Fresh_S1&quot;))
genes_ball_plot_with_values$Protocol &lt;- factor(genes_ball_plot_with_values$Protocol, levels=c(&quot;Fresh&quot;, &quot;Culture&quot;))

# rename the levels to have shorter plot panel titles
levels(genes_ball_plot_with_values$cell_type) &lt;- sub(&quot;Vascular endothelial&quot;, &quot;Vasc. Endo.&quot;, levels(genes_ball_plot_with_values$cell_type))
levels(genes_ball_plot_with_values$cell_type) &lt;- sub(&quot;Lymphatic endothelial&quot;, &quot;Lym. Endo.&quot;, levels(genes_ball_plot_with_values$cell_type))
levels(genes_ball_plot_with_values$cell_type) &lt;- sub(&quot;Sweat gland cells&quot;, &quot;Sweat gland&quot;, levels(genes_ball_plot_with_values$cell_type))
levels(genes_ball_plot_with_values$cell_type) &lt;- sub(&quot;Dendritic cells&quot;, &quot;Dendritic&quot;, levels(genes_ball_plot_with_values$cell_type))
levels(genes_ball_plot_with_values$cell_type) &lt;- sub(&quot;Schwann cells&quot;, &quot;Schwann&quot;, levels(genes_ball_plot_with_values$cell_type))
#levels(genes_ball_plot_with_values$cell_type) &lt;- sub(&quot;Melanocytes&quot;, &quot;Sweat gland.&quot;, levels(genes_ball_plot_with_values$cell_type))


special_plot &lt;- ggplot(genes_ball_plot_with_values, aes(x=gene, y=sample, size=frac_cells_expressed_in_cell_type_only, color=mean_of_logs_cell_type_only), otherInfo(Protocol)) +
  geom_point() +
  facet_grid(Protocol~cell_type, scales=&quot;free&quot;, space=&quot;free&quot;) +
  theme_ipsum_rc() +
  theme(axis.text.x = element_text(angle = 90, hjust=1), 
        axis.ticks.x=element_blank(),
        panel.spacing = unit(3, &quot;mm&quot;),
        strip.text.y = element_blank(),
        strip.text.x = element_text(size=11)
        ) +
  labs(title=&quot;Comparison of selected genes&quot;,
       subtitle=&quot;Size: proportion of cells of each sample expressing the marker \nColour:mean of log expression&quot;,
       y=&quot;&quot;, x=&quot;&quot;) +
  scale_colour_viridis() +
  guides(color=guide_colorbar(title=c(&quot;Mean of log counts&quot;)),
         size=guide_legend(title=c(&quot;% of cell expressed in cell type&quot;))) 
  
special_plot</code></pre>
<p><img src="figure/P2_04_FvC_gene_expression_plot.Rmd/plot%20and%20save-1.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(paste0(folder_path, &quot;special_plot.png&quot;),
       plot=special_plot,
       width=20*0.9,
       height=7*0.9,
       units=&quot;in&quot;,
       bg=&quot;white&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] stringr_1.4.0               viridis_0.6.2              
 [3] viridisLite_0.4.0           RColorBrewer_1.1-2         
 [5] edgeR_3.36.0                limma_3.50.0               
 [7] scater_1.22.0               scuttle_1.4.0              
 [9] openxlsx_4.2.5              dplyr_1.0.7                
[11] plyr_1.8.6                  SingleCellExperiment_1.16.0
[13] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[15] GenomicRanges_1.46.1        GenomeInfoDb_1.30.0        
[17] IRanges_2.28.0              S4Vectors_0.32.3           
[19] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[21] matrixStats_0.61.0          ggplot2_3.3.5              
[23] hrbrthemes_0.8.6           

loaded via a namespace (and not attached):
 [1] bitops_1.0-7              fs_1.5.2                 
 [3] rprojroot_2.0.2           tools_4.1.0              
 [5] bslib_0.3.1               utf8_1.2.2               
 [7] R6_2.5.1                  irlba_2.3.5              
 [9] vipor_0.4.5               DBI_1.1.2                
[11] colorspace_2.0-2          withr_2.4.3              
[13] tidyselect_1.1.1          gridExtra_2.3            
[15] compiler_4.1.0            git2r_0.29.0             
[17] extrafontdb_1.0           textshaping_0.3.6        
[19] BiocNeighbors_1.12.0      DelayedArray_0.20.0      
[21] labeling_0.4.2            sass_0.4.0               
[23] scales_1.1.1              systemfonts_1.0.3        
[25] digest_0.6.29             rmarkdown_2.11           
[27] XVector_0.34.0            pkgconfig_2.0.3          
[29] htmltools_0.5.2           extrafont_0.17           
[31] sparseMatrixStats_1.6.0   highr_0.9                
[33] fastmap_1.1.0             rlang_0.4.12             
[35] DelayedMatrixStats_1.16.0 farver_2.1.0             
[37] jquerylib_0.1.4           generics_0.1.1           
[39] jsonlite_1.7.3            BiocParallel_1.28.3      
[41] zip_2.2.0                 RCurl_1.98-1.5           
[43] magrittr_2.0.1            BiocSingular_1.10.0      
[45] GenomeInfoDbData_1.2.7    Matrix_1.4-0             
[47] Rcpp_1.0.8                ggbeeswarm_0.6.0         
[49] munsell_0.5.0             fansi_1.0.2              
[51] gdtools_0.2.3             lifecycle_1.0.1          
[53] stringi_1.7.6             whisker_0.4              
[55] yaml_2.2.1                zlibbioc_1.40.0          
[57] grid_4.1.0                parallel_4.1.0           
[59] promises_1.2.0.1          ggrepel_0.9.1            
[61] crayon_1.4.2              lattice_0.20-45          
[63] beachmat_2.10.0           locfit_1.5-9.4           
[65] knitr_1.37                pillar_1.6.4             
[67] ScaledMatrix_1.2.0        glue_1.6.0               
[69] evaluate_0.14             vctrs_0.3.8              
[71] httpuv_1.6.4              Rttf2pt1_1.3.8           
[73] gtable_0.3.0              purrr_0.3.4              
[75] assertthat_0.2.1          xfun_0.29                
[77] rsvd_1.0.5                later_1.3.0              
[79] ragg_1.2.1                tibble_3.1.6             
[81] beeswarm_0.4.0            workflowr_1.7.0          
[83] ellipsis_0.3.2           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
