<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2022-02-01" />

<title>Comparison of the ‘new’ skin tissue protocol with skin samples 1,2,4 &amp; 5</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol Paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Comparison of the ‘new’ skin tissue protocol with skin samples 1,2,4 &amp; 5</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2022-02-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>02_Protocol/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211228code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211228)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211228code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211228)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtreec4b210d95296965afb4378ce3e6c9dbef53901f5targetblankc4b210da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/c4b210d95296965afb4378ce3e6c9dbef53901f5" target="_blank">c4b210d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomDominiquePaulSkincellprotocolscRNAtreec4b210d95296965afb4378ce3e6c9dbef53901f5targetblankc4b210da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/tree/c4b210d95296965afb4378ce3e6c9dbef53901f5" target="_blank">c4b210d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    Metadata/.DS_Store
    Ignored:    Metadata/cell_annotation_markers/.DS_Store
    Ignored:    Paper/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/Supple_table_1.png
    Ignored:    analysis/Unused/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    code/seeds/.DS_Store
    Ignored:    data/
    Ignored:    output/.DS_Store
    Ignored:    output/plots/.DS_Store
    Ignored:    output/plots_31Jan/.DS_Store

Untracked files:
    Untracked:  analysis/P2_03_Fresh_only_subclustering_v2.Rmd
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-16h-27m-18s-err.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-16h-27m-18s-out.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-16h-28m-27s-err.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-16h-28m-27s-out.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-19h-12m-34s-err.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-19h-12m-34s-out.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-20h-17m-22s-err.txt
    Untracked:  analysis/logs/P2_01_data_comparison.Rmd-2022-02-01-20h-17m-22s-out.txt
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-02-01-19h-54m-36s-err.txt
    Untracked:  analysis/logs/P2_02_Cell_Cycle.Rmd-2022-02-01-19h-54m-36s-out.txt
    Untracked:  analysis/logs/P2_03_Fresh_only_subclustering.Rmd-2022-02-01-19h-58m-19s-err.txt
    Untracked:  analysis/logs/P2_03_Fresh_only_subclustering.Rmd-2022-02-01-19h-58m-19s-out.txt

Unstaged changes:
    Modified:   analysis/P2_01_data_comparison.Rmd
    Modified:   analysis/P2_03_Fresh_only_subclustering.Rmd
    Modified:   analysis/admin.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/P2_01_data_comparison.Rmd</code>) and HTML (<code>docs/P2_01_data_comparison.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/c4b210d95296965afb4378ce3e6c9dbef53901f5/analysis/P2_01_data_comparison.Rmd" target="_blank">c4b210d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-02-01
</td>
<td>
Annotated clusters, next step is subclustering
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Skin-cell-protocol-scRNA/c4b210d95296965afb4378ce3e6c9dbef53901f5/docs/P2_01_data_comparison.html" target="_blank">c4b210d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-02-01
</td>
<td>
Annotated clusters, next step is subclustering
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Skin-cell-protocol-scRNA/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/P2_01_data_comparison.html" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
<td>
Added new clustering after meeting 29-Jan-2022
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/239a294b77f8417881912b9058abc84dad2da0ac/analysis/P2_01_data_comparison.Rmd" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
<td>
Added new clustering after meeting 29-Jan-2022
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Skin-cell-protocol-scRNA/239a294b77f8417881912b9058abc84dad2da0ac/docs/P2_01_data_comparison.html" target="_blank">239a294</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
<td>
Added new clustering after meeting 29-Jan-2022
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/analysis/P2_01_data_comparison.Rmd" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
<td>
first commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/DominiquePaul/Skin-cell-protocol-scRNA/1923ae36314462857191f55817eed51512a6c060/docs/P2_01_data_comparison.html" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
<td>
first commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="loading-and-preprocess-the-data" class="section level1">
<h1>Loading and preprocess the data</h1>
<div id="identify-cultured-samples" class="section level2">
<h2>Identify cultured samples</h2>
<pre class="r"><code># Skin control data loaded as one group
samples_skin &lt;- list.files(path=&quot;../data/skin_tissue/scRNA_datasets&quot;,
                        pattern=&quot;matrix.mtx&quot;,
                        full.names=TRUE,
                        recursive=TRUE)

samples_skin &lt;- dirname(samples_skin)
samples_skin &lt;- grep(&quot;control&quot;, samples_skin, value=TRUE) # Only use control data
samples_skin &lt;- grep(&quot;filtered&quot;, samples_skin, value=TRUE)

names(samples_skin) &lt;- sapply(strsplit(samples_skin, &quot;/&quot;), .subset, 5) # This number might have to be adapted for other file paths or folder structures

names(samples_skin) &lt;- gsub(&quot;o24300_1_13-control1_BB&quot;, &quot;Culture_S1&quot;, names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o24300_1_09-control2_MFB&quot;, &quot;Culture_S2&quot;, names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o24793_1_09-BB_skin_control&quot;, &quot;Culture_S3&quot;,names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o25154_1_1-SS08_control&quot;, &quot;Culture_S4&quot;, names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o25154_1_3-SS18_control&quot;, &quot;Culture_S5&quot;, names(samples_skin))

samples_skin &lt;- samples_skin[!names(samples_skin) %in% &quot;Culture_S3&quot;]</code></pre>
</div>
<div id="load-fresh-samples" class="section level2">
<h2>Load fresh samples</h2>
<pre class="r"><code># Fresh data loaded together
samples_fresh &lt;- list.files(path=&quot;../data/fresh_data&quot;,
                      pattern=&quot;matrix.mtx&quot;,
                      full.names=TRUE,
                      recursive=TRUE)

samples_fresh &lt;- dirname(samples_fresh)
samples_fresh &lt;- grep(&quot;filtered&quot;, samples_fresh, value=TRUE)

names(samples_fresh) &lt;- sapply(strsplit(samples_fresh, &quot;/&quot;), .subset, 4) # This number might have to be adapted for other file paths or folder structures

names(samples_fresh) &lt;- gsub(&quot;01_o25154_1_5-HC_fresh&quot;, &quot;Fresh_S1&quot;, names(samples_fresh))
names(samples_fresh) &lt;- gsub(&quot;02_o25459_1_6-HS_Blaz&quot;, &quot;Fresh_S2&quot;, names(samples_fresh))

# Load data
sce_fresh &lt;- DropletUtils::read10xCounts(samples=samples_fresh)

# Factoring
sce_fresh$Sample &lt;- factor(sce_fresh$Sample) # Factor the sample name to save space

# Renaming
rownames(sce_fresh) &lt;- rowData(sce_fresh)$Symbol # Rename the rows
sce_fresh &lt;- sce_fresh[duplicated(rownames(sce_fresh)) == FALSE,] # remove duplicate rownames
colnames(sce_fresh) &lt;- paste0(sce_fresh$Sample, &quot;.&quot;, sce_fresh$Barcode) # Rename the columns


# add protocol name
sce_fresh$Protocol &lt;- factor(&quot;Fresh&quot;)</code></pre>
</div>
<div id="load-fresh-and-cultured-samples-as-one-dataset" class="section level2">
<h2>Load fresh and cultured samples as one dataset</h2>
<pre class="r"><code>samples_all &lt;- c(samples_skin, samples_fresh)

# Load data
sce_all &lt;- DropletUtils::read10xCounts(samples=samples_all)

# Factoring
sce_all$Sample &lt;- factor(sce_all$Sample) # Factor the sample name to save space

# Renaming
rownames(sce_all) &lt;- rowData(sce_all)$Symbol # Rename the rows
sce_all &lt;- sce_all[duplicated(rownames(sce_all)) == FALSE,] # remove duplicate rownames
colnames(sce_all) &lt;- paste0(sce_all$Sample, &quot;.&quot;, sce_all$Barcode) # Rename the columns

# Add column for protocol
sce_all$Protocol &lt;- factor(ifelse(grepl(&quot;Fresh&quot;, sce_all$Sample), &quot;Fresh&quot;, &quot;Culture&quot;))</code></pre>
</div>
<div id="basic-filtering" class="section level2 tabset">
<h2 class="tabset">Basic filtering</h2>
<p>We run a crude filtering on the cells to eliminate the majority of bad reads and speed up the runtime of the code below.</p>
<pre class="r"><code>crude_filtering &lt;- function(sce, sce_name){
  
  original_sce_dimensions &lt;- dim(sce)
  
  sce &lt;- sce[rowSums(counts(sce)&gt;0) &gt; 10,  # We only keep genes expressed in more than ten cells
             colSums(counts(sce) &gt;0) &gt; 150] # We only keep cells with more than 150 genes expressed

  new_sce_dimensions &lt;- dim(sce)

  absolute_changes &lt;- original_sce_dimensions - new_sce_dimensions
  relative_changes &lt;- absolute_changes / original_sce_dimensions

  cat(paste0(&quot;\n\n### &quot;, sce_name))
  cat(paste0(&quot;\n\n#### Dropped &quot;, absolute_changes[1], &quot; genes out of &quot;, original_sce_dimensions[1], &quot; (~&quot;, round(100*relative_changes[1],2), &quot;%). &quot;, new_sce_dimensions[1], &quot; genes remaining.\n\n&quot;))
  cat(paste0(&quot;\n#### Dropped &quot;, absolute_changes[2], &quot; cells out of &quot;, original_sce_dimensions[2], &quot; (~&quot;, round(100*relative_changes[2],2), &quot;%). &quot;, new_sce_dimensions[2], &quot; cells remaining.\n\n&quot; ))
  
  sce
}

sce_fresh &lt;- crude_filtering(sce_fresh, &quot;Fresh&quot;)</code></pre>
<div id="fresh" class="section level3">
<h3>Fresh</h3>
<div id="dropped-5236-genes-out-of-21544-24.3.-16308-genes-remaining." class="section level4">
<h4>Dropped 5236 genes out of 21544 (~24.3%). 16308 genes remaining.</h4>
</div>
<div id="dropped-189-cells-out-of-9941-1.9.-9752-cells-remaining." class="section level4">
<h4>Dropped 189 cells out of 9941 (~1.9%). 9752 cells remaining.</h4>
<pre class="r"><code>sce_all &lt;- crude_filtering(sce_all, &quot;All&quot;)</code></pre>
</div>
</div>
<div id="all" class="section level3">
<h3>All</h3>
<div id="dropped-4063-genes-out-of-21544-18.86.-17481-genes-remaining." class="section level4">
<h4>Dropped 4063 genes out of 21544 (~18.86%). 17481 genes remaining.</h4>
</div>
<div id="dropped-456-cells-out-of-28171-1.62.-27715-cells-remaining." class="section level4">
<h4>Dropped 456 cells out of 28171 (~1.62%). 27715 cells remaining.</h4>
</div>
</div>
</div>
</div>
<div id="overview-of-doublets-dropped-comparison" class="section level1">
<h1>Overview of doublets dropped comparison</h1>
<pre class="r"><code># identify doublets
# seed_to_use &lt;- readRDS(&quot;../code/seeds/seed8_fresh.rds&quot;)
# set.seed(seed_to_use)
set.seed(100)
sce_fresh &lt;- scDblFinder(sce_fresh, samples=&quot;Sample&quot;)

# seed_to_use &lt;- readRDS(&quot;../code/seeds/seed9_all.rds&quot;)
# set.seed(seed_to_use)
set.seed(100)
sce_all &lt;- scDblFinder(sce_all, samples=&quot;Sample&quot;)

# plot the number of doublets per sample

# Fresh
cat(paste0(&quot;\n\n## Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only" class="section level2">
<h2>Fresh samples only</h2>
<pre class="r"><code>table(sce_fresh$scDblFinder.class, sce_fresh$Sample) %&gt;%
  as.data.frame() %&gt;%
  dplyr::rename(&quot;Type&quot;=&quot;Var1&quot;, &quot;Sample&quot;=&quot;Var2&quot;) %&gt;%
  ggplot(aes(x=Sample, y=Freq, fill = Type, label=Freq)) +
  geom_bar(stat=&quot;identity&quot;,
           position = position_stack(reverse = TRUE)) +
  labs(title = &quot;Doublets identified&quot;,
       x=&quot;&quot;,
       y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/remove%20doublets-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-remove-doublets-1">
Past versions of “remove doublets-1.png”
</button>
</p>
<div id="fig-no-spaces-remove-doublets-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/remove doublets-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Fresh and culture
# plot the number of doublets per sample
cat(paste0(&quot;\n\n## Fresh and cultured samples \n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples" class="section level2">
<h2>Fresh and cultured samples</h2>
<pre class="r"><code>table(sce_all$scDblFinder.class, sce_all$Sample) %&gt;%
  as.data.frame() %&gt;%
  dplyr::rename(&quot;Type&quot;=&quot;Var1&quot;, &quot;Sample&quot;=&quot;Var2&quot;) %&gt;%
  ggplot(aes(x=Sample, y=Freq, fill = Type, label=Freq)) +
  geom_bar(stat=&quot;identity&quot;,
           position = position_stack(reverse = TRUE)) +
  labs(title = &quot;Doublets identified&quot;,
       x=&quot;&quot;,
       y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/remove%20doublets-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-remove-doublets-2">
Past versions of “remove doublets-2.png”
</button>
</p>
<div id="fig-no-spaces-remove-doublets-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/remove doublets-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># finally remove droplets
sce_fresh &lt;- sce_fresh[,colData(sce_fresh)$scDblFinder.class==&quot;singlet&quot;]
sce_all &lt;- sce_all[,colData(sce_all)$scDblFinder.class==&quot;singlet&quot;]</code></pre>
</div>
</div>
<div id="quality-control" class="section level1">
<h1>Quality Control</h1>
<div id="mitochondria-check" class="section level2 tabset">
<h2 class="tabset">Mitochondria check</h2>
<p>Do we have mitochondrial genes in the data?</p>
<pre class="r"><code>identify_mitochondrial_genes &lt;- function(sce, sce_name){
  rowData(sce)$is_mitochondrial &lt;- grepl(&quot;^(MT-)&quot;, rowData(sce)$Symbol, ignore.case=TRUE)
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; \n\n&quot;))
  print(table(rowData(sce)$is_mitochondrial) %&gt;%
          as.data.frame() %&gt;%
          rename(is.mito.gene=Var1) %&gt;%
          kable() %&gt;%
          kable_styling())
  
  sce
}

sce_fresh &lt;- identify_mitochondrial_genes(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-1" class="section level3">
<h3>Fresh samples only</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
is.mito.gene
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
16273
</td>
</tr>
<tr>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
35
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>sce_all &lt;- identify_mitochondrial_genes(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-1" class="section level3">
<h3>Fresh and cultured samples</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
is.mito.gene
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
17444
</td>
</tr>
<tr>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
37
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="add-quality-control-metrics" class="section level2">
<h2>Add quality control metrics</h2>
<pre class="r"><code># Fresh only
sce_fresh &lt;- scater::addPerCellQC(sce_fresh, list(Mt = rowData(sce_fresh)$is_mitochondrial,
                                                  genes= !rowData(sce_fresh)$is_mitochondrial),
                                  percent_top=c(50, 100, 200, 500))

sce_fresh$log_sum &lt;- log10(sce_fresh$sum)
sce_fresh$log_total &lt;- log10(sce_fresh$total)
  
# Fresh and culture
sce_all &lt;- scater::addPerCellQC(sce_all, list(Mt = rowData(sce_all)$is_mitochondrial,
                                                  genes= !rowData(sce_all)$is_mitochondrial),
                                  percent_top=c(50, 100, 200, 500))

sce_all$log_sum &lt;- log10(sce_all$sum)
sce_all$log_total &lt;- log10(sce_all$total)</code></pre>
</div>
<div id="first-qc-charts" class="section level2">
<h2>First QC charts</h2>
<pre class="r"><code>plot_qc_charts &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; \n\n&quot;))

  # cat(&quot;\n### Number of genes \n\n&quot;)
  plot1 &lt;- ggplot(as.data.frame(colData(sce)), aes(x=Sample, y=log10(detected))) +
    geom_violin(scale=&quot;count&quot;) +
    labs(title=&quot;Number of detected genes&quot;,
         subtitle=sce_name,
         x=&quot;&quot;,
         y=&quot;Number of detected genes (log10)&quot;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))

  # cat(&quot;\n### Number of genes \n\n&quot;)
  plot2 &lt;- ggplot(as.data.frame(colData(sce)), aes(x=Sample, y=log_total)) +
    geom_violin(scale=&quot;count&quot;) +
    labs(title=&quot;Total gene expression counts&quot;,
         subtitle=sce_name,
         x=&quot;&quot;,
         y=&quot;Total gene expression counts (log10)&quot;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))

  # plot3 &lt;- ggplot(as.data.frame(qcs), aes(x=Sample, y=subsets_Mt_percent)) +
  plot3 &lt;- ggplot(as.data.frame(colData(sce)), aes(x=Sample, y=subsets_Mt_percent)) +
    geom_violin(scale=&quot;count&quot;) +
    labs(title=&quot;Share of mitochondrial genes&quot;,
         subtitle=sce_name,
         x=&quot;&quot;,
         y=&quot;Mitochodrial share (percent)&quot;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))

  gridExtra::grid.arrange(plot1, plot2, plot3, nrow=1)
}

plot_qc_charts(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-2" class="section level3">
<h3>Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/First%20QC%20charts-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-First-QC-charts-1">
Past versions of “First QC charts-1.png”
</button>
</p>
<div id="fig-no-spaces-First-QC-charts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/First QC charts-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_qc_charts(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-2" class="section level3">
<h3>Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/First%20QC%20charts-2.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-First-QC-charts-2">
Past versions of “First QC charts-2.png”
</button>
</p>
<div id="fig-no-spaces-First-QC-charts-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/First QC charts-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="mitochondria-qc-charts" class="section level2 tabset">
<h2 class="tabset">Mitochondria QC charts</h2>
<pre class="r"><code>plot_mitochondrial_qc_charts &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; \n\n&quot;))
  
  mito1 &lt;- ggplot(as.data.frame(colData(sce)),
         aes(x=log_total, y=subsets_genes_detected, color=subsets_Mt_percent)) +
    geom_point(size=0.25) +
    labs(title=&quot;Library size vs. gene count&quot;,
         subtitle=&quot;percentual share of mitochondrial genes in blue&quot;,
         x=&quot;Sum of expression values by cell&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    geom_density_2d() +
    scale_y_log10()

  mito2 &lt;- ggplot(as.data.frame(colData(sce)),
         aes(x=subsets_Mt_percent,
             y=subsets_genes_detected,
             color=log_sum)) +
    geom_point(size=0.25) +
    labs(title=&quot;Mito. content vs. genes detected&quot;,
         subtitle=&quot;coloured by log library size&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d()

  gridExtra::grid.arrange(mito1, mito2, nrow=1)
  
}

plot_mitochondrial_qc_charts(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-3" class="section level3">
<h3>Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%202-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-QC:-Mitochondrial-content-2-1">
Past versions of “plots for QC: Mitochondrial content 2-1.png”
</button>
</p>
<div id="fig-no-spaces-plots-for-QC:-Mitochondrial-content-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/plots for QC: Mitochondrial content 2-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_mitochondrial_qc_charts(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-3" class="section level3">
<h3>Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%202-2.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-QC:-Mitochondrial-content-2-2">
Past versions of “plots for QC: Mitochondrial content 2-2.png”
</button>
</p>
<div id="fig-no-spaces-plots-for-QC:-Mitochondrial-content-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/plots for QC: Mitochondrial content 2-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="library-size-vs.-mito-share-by-sample" class="section level2 tabset">
<h2 class="tabset">Library size vs. mito share by sample</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-4" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
       aes(x=total, y=subsets_genes_detected, color=subsets_Mt_percent)) +
  geom_point(size=0.25) +
  labs(title=&quot;Library size vs. gene count faceted by sample&quot;,
       subtitle=&quot;percentual share of mitochondrial genes in blue&quot;,
       x=&quot;Sum of expression values by cell&quot;,
       y=&quot;# of non-mitochondrial genes detected&quot;) +
  scale_x_log10() +
  geom_density_2d() +
  scale_y_log10() +
  facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/lib%20size%20vs.%20mito-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-lib-size-vs.-mito-1">
Past versions of “lib size vs. mito-1.png”
</button>
</p>
<div id="fig-no-spaces-lib-size-vs.-mito-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/lib size vs. mito-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh and cultured samples \n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-4" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
       aes(x=total, y=subsets_genes_detected, color=subsets_Mt_percent)) +
  geom_point(size=0.25) +
  labs(title=&quot;Library size vs. gene count faceted by sample&quot;,
       subtitle=&quot;percentual share of mitochondrial genes in blue&quot;,
       x=&quot;Sum of expression values by cell&quot;,
       y=&quot;# of non-mitochondrial genes detected&quot;) +
  scale_x_log10() +
  geom_density_2d() +
  scale_y_log10() +
  facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/lib%20size%20vs.%20mito-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-lib-size-vs.-mito-2">
Past versions of “lib size vs. mito-2.png”
</button>
</p>
<div id="fig-no-spaces-lib-size-vs.-mito-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/lib size vs. mito-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="mito.-percent-vs.-genes-by-sample" class="section level2 tabset">
<h2 class="tabset">Mito. percent vs. genes by sample</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-5" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
         aes(x=subsets_Mt_percent,
             y=subsets_genes_detected,
             color=log_sum)) +
    geom_point(size=0.25) +
    labs(title=&quot;Mito. content vs. genes detected faceted by sample&quot;,
         subtitle=&quot;Coloured by log library size&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d() +
    facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/unnamed-chunk-1-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/unnamed-chunk-1-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh and cultured samples \n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-5" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
         aes(x=subsets_Mt_percent,
             y=subsets_genes_detected,
             color=log_sum)) +
    geom_point(size=0.25) +
    labs(title=&quot;Mito. content vs. genes detected faceted by sample&quot;,
         subtitle=&quot;Coloured by log library size&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d() +
    facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/unnamed-chunk-1-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-2">
Past versions of unnamed-chunk-1-2.png
</button>
</p>
<div id="fig-unnamed-chunk-1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/unnamed-chunk-1-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="filtering" class="section level2">
<h2>Filtering</h2>
<p>Cells are marked to be dropped if:</p>
<ul>
<li>Sum of all expression values in a given cell is lower than 2 median absolute deviations from the sample mean</li>
<li>Number of genes of a given cell is lower than 2 median absolute deviations from the average number of genes in the given sample</li>
<li>The percentage of mitochondrial gene expression values is higher than 2 median absolute deviations from the sample mean</li>
<li>More than 25% percent of a genes expression values are from mitochondrial genes</li>
<li>The sum of all expression values is less than 1500</li>
<li>If they express less than 500 genes</li>
</ul>
<pre class="r"><code>identify_lowq_cells &lt;- function(sce){
  
  # Relative criteria for filtering
  sce$total_expression_counts_drop &lt;- isOutlier(sce$sum, nmads=2, type=&quot;lower&quot;, batch=sce$Sample, log=TRUE)

  sce$genes_detected_drop &lt;- isOutlier(sce$detected, nmads=2, type=&quot;lower&quot;, batch=sce$Sample, log=TRUE)

  sce$mito_drop &lt;- isOutlier(sce$subsets_Mt_percent, nmads=2, type=&quot;higher&quot;, batch=sce$Sample)

  # Absolute thresholds for filtering

  # Drop cells with a sum expression values lower than 1500
  sce$total_expression_counts_drop_manual_thresh &lt;- sce$sum &lt; 1500
  
  # A manual threshold: we require cells to have at least 500 genes expressed
  sce$genes_detected_manual_thresh &lt;- sce$detected &lt; 500

  # Drop cells with a percentage of mitochondrial expression values above a certain value
  sce$mito_drop_manual_thresh_25 &lt;- sce$subsets_Mt_percent &gt; 25
  
  sce$to_be_dropped &lt;- sce$total_expression_counts_drop |
    sce$genes_detected_drop |
    sce$mito_drop |
    sce$total_expression_counts_drop_manual_thresh | 
    sce$genes_detected_manual_thresh |
    sce$mito_drop_manual_thresh_25
  

 sce
}

sce_fresh &lt;- identify_lowq_cells(sce_fresh)
sce_all &lt;- identify_lowq_cells(sce_all)</code></pre>
<div id="how-many-cells-are-captured-by-each-filter" class="section level3 tabset">
<h3 class="tabset">How many cells are captured by each filter?</h3>
<pre class="r"><code>print_filter_overview &lt;- function(sce, sce_name){
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; \n\n&quot;))
  
  print(t(data.frame(Drop_via_lib_size = sum(sce$total_expression_counts_drop),
           Drop_via_lib_size_manual = sum(sce$total_expression_counts_drop_manual_thresh),
           Drop_via_detected_genes = sum(sce$genes_detected_drop),
           Drop_via_manual_genes_thresh = sum(sce$genes_detected_manual_thresh),
           Drop_via_mito_percentage = sum(sce$mito_drop),
           Drop_via_mito_thresh_25 = sum(sce$mito_drop_manual_thresh_25),
           Drop_via_joint_criteria = sum(sce$total_expression_counts_drop |
           sce$genes_detected_drop |
           sce$mito_drop |
           sce$total_expression_counts_drop_manual_thresh |
           sce$mito_drop_manual_thresh_25 | 
           sce$genes_detected_manual_thresh),
           Total_cells = nrow(colData(sce)))) %&gt;%
  kable() %&gt;%
  kable_styling())
}

print_filter_overview(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
</div>
<div id="fresh-samples-only-6" class="section level3">
<h3>Fresh samples only</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Drop_via_lib_size
</td>
<td style="text-align:right;">
676
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_lib_size_manual
</td>
<td style="text-align:right;">
807
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_detected_genes
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_manual_genes_thresh
</td>
<td style="text-align:right;">
533
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_percentage
</td>
<td style="text-align:right;">
835
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_thresh_25
</td>
<td style="text-align:right;">
2272
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_joint_criteria
</td>
<td style="text-align:right;">
2915
</td>
</tr>
<tr>
<td style="text-align:left;">
Total_cells
</td>
<td style="text-align:right;">
9107
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>print_filter_overview(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-6" class="section level3">
<h3>Fresh and cultured samples</h3>
<table class="table" style="margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Drop_via_lib_size
</td>
<td style="text-align:right;">
1881
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_lib_size_manual
</td>
<td style="text-align:right;">
3836
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_detected_genes
</td>
<td style="text-align:right;">
2863
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_manual_genes_thresh
</td>
<td style="text-align:right;">
2565
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_percentage
</td>
<td style="text-align:right;">
2788
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_thresh_25
</td>
<td style="text-align:right;">
4233
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_joint_criteria
</td>
<td style="text-align:right;">
7546
</td>
</tr>
<tr>
<td style="text-align:left;">
Total_cells
</td>
<td style="text-align:right;">
25383
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="how-many-cells-are-affected-by-each-filter" class="section level2 tabset">
<h2 class="tabset">How many cells are affected by each filter</h2>
<pre class="r"><code>show_panel_overview_filters &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; {.tabset }\n\n&quot;))
  
  ### Absolute numbers ### 
  
  absolute_count &lt;- rbind(
    table(sce$Sample),
    table(sce$total_expression_counts_drop, sce$Sample)[2,],
    table(sce$total_expression_counts_drop_manual_thresh, sce$Sample)[2,],
    table(sce$genes_detected_drop, sce$Sample)[2,],
    table(sce$genes_detected_drop, sce$Sample)[2,],
    table(sce$mito_drop, sce$Sample)[2,],
    table(sce$mito_drop_manual_thresh_25, sce$Sample)[2,],
    table(sce$to_be_dropped, sce$Sample)[2,],
    table(sce$to_be_dropped, sce$Sample)[1,]) %&gt;%
    as.data.frame()

  rownames(absolute_count) &lt;- c(&quot;Number of cells&quot;,
               &quot;Library size relative filter&quot;,
               &quot;Library size manual filter&quot;,
               &quot;Number of genes relative filter&quot;,
               &quot;Number of genes manual filter&quot;,
               &quot;Mito. percentage relative threshold&quot;,
               &quot;Mito. percentage manual threshold&quot;,
               &quot;Total dropped&quot;,
               &quot;Cells remaining&quot;)
  
  absolute_count[, &quot;Total&quot;] &lt;- rowSums(absolute_count)
  
  cat(&quot;\n\n#### Table (Absolute numbers) \n\n&quot;)
  print(absolute_count %&gt;% 
    kable() %&gt;%
    kable_styling())
  
  cat(&quot;\n\n#### Panel (Absolute numbers) \n\n&quot;)
  print(absolute_count[2:7,-dim(absolute_count)[2]] %&gt;%
    tibble::rownames_to_column() %&gt;%
    tidyr::gather(colname, value, -rowname) %&gt;%
    ggplot(aes(x = colname, y = rowname, fill = value)) +
    geom_tile() +
    geom_text(aes(label=value), color=&quot;white&quot;) +
    scale_x_discrete(position=&quot;top&quot;) +
    labs(title=&quot;Cells filtered out by respective filter&quot;,
         subtitle=&quot;Number of cells&quot;,
         x=&quot;&quot;,
         y=&quot;&quot;) +
    guides(fill=&quot;none&quot;))
  
  
  ### Relative numbers ### 
  
  # divide by number of cells to give a relative count
  relative_count &lt;- mapply(&#39;/&#39;, absolute_count, absolute_count[1,]) %&gt;%
    round(3) %&gt;%
    as.data.frame()
  
  rownames(relative_count) &lt;- c(&quot;Number of cells&quot;,
                                         &quot;Library size relative filter&quot;,
                                         &quot;Library size manual filter&quot;,
                                         &quot;Number of genes relative filter&quot;,
                                         &quot;Number of genes manual filter&quot;,
                                         &quot;Mito. percentage relative threshold&quot;,
                                         &quot;Mito. percentage manual threshold&quot;,
                                         &quot;Total dropped&quot;,
                                         &quot;Cells remaining&quot;)
  
  cat(&quot;\n\n#### Table (Relative numbers by sample) \n\n&quot;)
  print(relative_count %&gt;% 
          mutate_all(list(~paste0(.*100, &quot;%&quot;))) %&gt;%
    kable() %&gt;%
    kable_styling())
  
  
  cat(&quot;\n\n#### Panel (Relative numbers by sample) \n\n&quot;)
  print(relative_count[2:7,-dim(relative_count)[2]] %&gt;%
    tibble::rownames_to_column() %&gt;%
    tidyr::gather(colname, value, -rowname) %&gt;%
    ggplot(aes(x = colname, y = rowname, fill = value)) +
    geom_tile() +
    geom_text(aes(label=scales::percent(value, accuracy = 0.1)), color=&quot;white&quot;) +
    scale_x_discrete(position=&quot;top&quot;) +
    labs(title=&quot;Cells filtered out by respective filter&quot;,
         subtitle=&quot;Percentage of all cells in given sample before filtering&quot;,
         x=&quot;&quot;,
         y=&quot;&quot;) +
    guides(fill=&quot;none&quot;))
}

show_panel_overview_filters(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-7" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<div id="table-absolute-numbers" class="section level4">
<h4>Table (Absolute numbers)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Fresh_S1
</th>
<th style="text-align:right;">
Fresh_S2
</th>
<th style="text-align:right;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
4392
</td>
<td style="text-align:right;">
4715
</td>
<td style="text-align:right;">
9107
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:right;">
287
</td>
<td style="text-align:right;">
389
</td>
<td style="text-align:right;">
676
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:right;">
147
</td>
<td style="text-align:right;">
660
</td>
<td style="text-align:right;">
807
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:right;">
417
</td>
<td style="text-align:right;">
502
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:right;">
417
</td>
<td style="text-align:right;">
502
</td>
<td style="text-align:right;">
919
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:right;">
361
</td>
<td style="text-align:right;">
474
</td>
<td style="text-align:right;">
835
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:right;">
827
</td>
<td style="text-align:right;">
1445
</td>
<td style="text-align:right;">
2272
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:right;">
1080
</td>
<td style="text-align:right;">
1835
</td>
<td style="text-align:right;">
2915
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:right;">
3312
</td>
<td style="text-align:right;">
2880
</td>
<td style="text-align:right;">
6192
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-absolute-numbers" class="section level4">
<h4>Panel (Absolute numbers)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Overview-of-cells-being-dropped-1">
Past versions of “Overview of cells being dropped-1.png”
</button>
</p>
<div id="fig-no-spaces-Overview-of-cells-being-dropped-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Overview of cells being dropped-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-relative-numbers-by-sample" class="section level4">
<h4>Table (Relative numbers by sample)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Fresh_S1
</th>
<th style="text-align:left;">
Fresh_S2
</th>
<th style="text-align:left;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:left;">
6.5%
</td>
<td style="text-align:left;">
8.3%
</td>
<td style="text-align:left;">
7.4%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:left;">
3.3%
</td>
<td style="text-align:left;">
14%
</td>
<td style="text-align:left;">
8.9%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:left;">
9.5%
</td>
<td style="text-align:left;">
10.6%
</td>
<td style="text-align:left;">
10.1%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:left;">
9.5%
</td>
<td style="text-align:left;">
10.6%
</td>
<td style="text-align:left;">
10.1%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:left;">
8.2%
</td>
<td style="text-align:left;">
10.1%
</td>
<td style="text-align:left;">
9.2%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:left;">
18.8%
</td>
<td style="text-align:left;">
30.6%
</td>
<td style="text-align:left;">
24.9%
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:left;">
24.6%
</td>
<td style="text-align:left;">
38.9%
</td>
<td style="text-align:left;">
32%
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:left;">
75.4%
</td>
<td style="text-align:left;">
61.1%
</td>
<td style="text-align:left;">
68%
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-relative-numbers-by-sample" class="section level4">
<h4>Panel (Relative numbers by sample)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Overview-of-cells-being-dropped-2">
Past versions of “Overview of cells being dropped-2.png”
</button>
</p>
<div id="fig-no-spaces-Overview-of-cells-being-dropped-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Overview of cells being dropped-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>show_panel_overview_filters(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
</div>
<div id="fresh-and-cultured-samples-7" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<div id="table-absolute-numbers-1" class="section level4">
<h4>Table (Absolute numbers)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Culture_S1
</th>
<th style="text-align:right;">
Culture_S2
</th>
<th style="text-align:right;">
Culture_S4
</th>
<th style="text-align:right;">
Culture_S5
</th>
<th style="text-align:right;">
Fresh_S1
</th>
<th style="text-align:right;">
Fresh_S2
</th>
<th style="text-align:right;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
976
</td>
<td style="text-align:right;">
4473
</td>
<td style="text-align:right;">
7986
</td>
<td style="text-align:right;">
2853
</td>
<td style="text-align:right;">
4380
</td>
<td style="text-align:right;">
4715
</td>
<td style="text-align:right;">
25383
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
429
</td>
<td style="text-align:right;">
549
</td>
<td style="text-align:right;">
224
</td>
<td style="text-align:right;">
290
</td>
<td style="text-align:right;">
389
</td>
<td style="text-align:right;">
1881
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:right;">
464
</td>
<td style="text-align:right;">
640
</td>
<td style="text-align:right;">
1674
</td>
<td style="text-align:right;">
250
</td>
<td style="text-align:right;">
146
</td>
<td style="text-align:right;">
662
</td>
<td style="text-align:right;">
3836
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
685
</td>
<td style="text-align:right;">
908
</td>
<td style="text-align:right;">
353
</td>
<td style="text-align:right;">
417
</td>
<td style="text-align:right;">
500
</td>
<td style="text-align:right;">
2863
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
685
</td>
<td style="text-align:right;">
908
</td>
<td style="text-align:right;">
353
</td>
<td style="text-align:right;">
417
</td>
<td style="text-align:right;">
500
</td>
<td style="text-align:right;">
2863
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
581
</td>
<td style="text-align:right;">
915
</td>
<td style="text-align:right;">
397
</td>
<td style="text-align:right;">
358
</td>
<td style="text-align:right;">
479
</td>
<td style="text-align:right;">
2788
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:right;">
263
</td>
<td style="text-align:right;">
329
</td>
<td style="text-align:right;">
1017
</td>
<td style="text-align:right;">
349
</td>
<td style="text-align:right;">
831
</td>
<td style="text-align:right;">
1444
</td>
<td style="text-align:right;">
4233
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:right;">
558
</td>
<td style="text-align:right;">
1090
</td>
<td style="text-align:right;">
2353
</td>
<td style="text-align:right;">
628
</td>
<td style="text-align:right;">
1083
</td>
<td style="text-align:right;">
1834
</td>
<td style="text-align:right;">
7546
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:right;">
418
</td>
<td style="text-align:right;">
3383
</td>
<td style="text-align:right;">
5633
</td>
<td style="text-align:right;">
2225
</td>
<td style="text-align:right;">
3297
</td>
<td style="text-align:right;">
2881
</td>
<td style="text-align:right;">
17837
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-absolute-numbers-1" class="section level4">
<h4>Panel (Absolute numbers)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-3.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Overview-of-cells-being-dropped-3">
Past versions of “Overview of cells being dropped-3.png”
</button>
</p>
<div id="fig-no-spaces-Overview-of-cells-being-dropped-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Overview of cells being dropped-3.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="table-relative-numbers-by-sample-1" class="section level4">
<h4>Table (Relative numbers by sample)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Culture_S1
</th>
<th style="text-align:left;">
Culture_S2
</th>
<th style="text-align:left;">
Culture_S4
</th>
<th style="text-align:left;">
Culture_S5
</th>
<th style="text-align:left;">
Fresh_S1
</th>
<th style="text-align:left;">
Fresh_S2
</th>
<th style="text-align:left;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:left;">
0%
</td>
<td style="text-align:left;">
9.6%
</td>
<td style="text-align:left;">
6.9%
</td>
<td style="text-align:left;">
7.9%
</td>
<td style="text-align:left;">
6.6%
</td>
<td style="text-align:left;">
8.3%
</td>
<td style="text-align:left;">
7.4%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:left;">
47.5%
</td>
<td style="text-align:left;">
14.3%
</td>
<td style="text-align:left;">
21%
</td>
<td style="text-align:left;">
8.8%
</td>
<td style="text-align:left;">
3.3%
</td>
<td style="text-align:left;">
14%
</td>
<td style="text-align:left;">
15.1%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:left;">
0%
</td>
<td style="text-align:left;">
15.3%
</td>
<td style="text-align:left;">
11.4%
</td>
<td style="text-align:left;">
12.4%
</td>
<td style="text-align:left;">
9.5%
</td>
<td style="text-align:left;">
10.6%
</td>
<td style="text-align:left;">
11.3%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:left;">
0%
</td>
<td style="text-align:left;">
15.3%
</td>
<td style="text-align:left;">
11.4%
</td>
<td style="text-align:left;">
12.4%
</td>
<td style="text-align:left;">
9.5%
</td>
<td style="text-align:left;">
10.6%
</td>
<td style="text-align:left;">
11.3%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:left;">
5.9%
</td>
<td style="text-align:left;">
13%
</td>
<td style="text-align:left;">
11.5%
</td>
<td style="text-align:left;">
13.9%
</td>
<td style="text-align:left;">
8.2%
</td>
<td style="text-align:left;">
10.2%
</td>
<td style="text-align:left;">
11%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:left;">
26.9%
</td>
<td style="text-align:left;">
7.4%
</td>
<td style="text-align:left;">
12.7%
</td>
<td style="text-align:left;">
12.2%
</td>
<td style="text-align:left;">
19%
</td>
<td style="text-align:left;">
30.6%
</td>
<td style="text-align:left;">
16.7%
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:left;">
57.2%
</td>
<td style="text-align:left;">
24.4%
</td>
<td style="text-align:left;">
29.5%
</td>
<td style="text-align:left;">
22%
</td>
<td style="text-align:left;">
24.7%
</td>
<td style="text-align:left;">
38.9%
</td>
<td style="text-align:left;">
29.7%
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:left;">
42.8%
</td>
<td style="text-align:left;">
75.6%
</td>
<td style="text-align:left;">
70.5%
</td>
<td style="text-align:left;">
78%
</td>
<td style="text-align:left;">
75.3%
</td>
<td style="text-align:left;">
61.1%
</td>
<td style="text-align:left;">
70.3%
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-relative-numbers-by-sample-1" class="section level4">
<h4>Panel (Relative numbers by sample)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-4.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Overview-of-cells-being-dropped-4">
Past versions of “Overview of cells being dropped-4.png”
</button>
</p>
<div id="fig-no-spaces-Overview-of-cells-being-dropped-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Overview of cells being dropped-4.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="plots-cells-that-will-be-dropped" class="section level2 tabset">
<h2 class="tabset">Plots cells that will be dropped</h2>
<pre class="r"><code>show_violin_plots_of_dropped &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; {.tabset}\n\n&quot;))
  
  # for the right colouring we order the levels manually
  sce$to_be_dropped &lt;- factor(sce$to_be_dropped, levels=c(FALSE, TRUE))

  plot1 &lt;- plotColData(sce,
                       x=&quot;Sample&quot;,
                       y=&quot;log_total&quot;,
                       colour_by=&quot;to_be_dropped&quot;,
                       point_size=0.5) +
    labs(title=&quot;Library size&quot;,
         subtitle=&quot;Dropped cells in orange&quot;,
         x=&quot;&quot;,
         y=&quot;Library size (log-scale)&quot;) +
    guides(colour=&quot;none&quot;) +
    theme_ipsum_rc() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))
  
  plot2 &lt;- plotColData(sce,
                       x=&quot;Sample&quot;,
                       y=&quot;detected&quot;,
                       colour_by=&quot;to_be_dropped&quot;,
                       point_size=0.5) +
    labs(title=&quot;Genes detected&quot;,
         subtitle=&quot;Dropped cells in orange&quot;,
         x=&quot;&quot;,
         y=&quot;Genes detected&quot;) +
    guides(colour=&quot;none&quot;) +
    theme_ipsum_rc() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))
  
  plot3 &lt;- plotColData(sce,
                       x=&quot;Sample&quot;,
                       y=&quot;subsets_Mt_percent&quot;,
                       colour_by=&quot;to_be_dropped&quot;,
                       point_size=0.5) +
    labs(title=&quot;Mitochondrial counts as % library size&quot;,
         x=&quot;&quot;,
         subtitle=&quot;Dropped cells in orange&quot;,
         y=&quot;Mito. gene expression values as % of total&quot;) +
    guides(colour=&quot;none&quot;) +
    theme_ipsum_rc() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))

  print(gridExtra::grid.arrange(plot1, plot2, plot3, nrow=1))
}

show_violin_plots_of_dropped(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-8" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/Violin%20plot%20of%20cells%20to%20be%20dropped-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Violin-plot-of-cells-to-be-dropped-1">
Past versions of “Violin plot of cells to be dropped-1.png”
</button>
</p>
<div id="fig-no-spaces-Violin-plot-of-cells-to-be-dropped-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/Violin plot of cells to be dropped-1.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Violin plot of cells to be dropped-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>TableGrob (1 x 3) “arrange”: 3 grobs z cells name grob 1 1 (1-1,1-1) arrange gtable[layout] 2 2 (1-1,2-2) arrange gtable[layout] 3 3 (1-1,3-3) arrange gtable[layout]</p>
<pre class="r"><code>show_violin_plots_of_dropped(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-8" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/Violin%20plot%20of%20cells%20to%20be%20dropped-2.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Violin-plot-of-cells-to-be-dropped-2">
Past versions of “Violin plot of cells to be dropped-2.png”
</button>
</p>
<div id="fig-no-spaces-Violin-plot-of-cells-to-be-dropped-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/Violin plot of cells to be dropped-2.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Violin plot of cells to be dropped-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>TableGrob (1 x 3) “arrange”: 3 grobs z cells name grob 1 1 (1-1,1-1) arrange gtable[layout] 2 2 (1-1,2-2) arrange gtable[layout] 3 3 (1-1,3-3) arrange gtable[layout]</p>
</div>
</div>
<div id="mitochondria-plots-after-qc-filtering" class="section level2 tabset">
<h2 class="tabset">Mitochondria plots after QC filtering</h2>
<pre class="r"><code>show_mito_plots &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; {.tabset}\n\n&quot;))
  
  # for the right colouring we order the levels manually
  sce$to_be_dropped &lt;- factor(sce$to_be_dropped, levels=c(TRUE, FALSE))

  mito1 &lt;- ggplot(as.data.frame(colData(sce)), 
                  aes(x=log_total,
                      y=subsets_genes_detected,
                      color=to_be_dropped)) +
    geom_point(size=0.5) +
    labs(title=&quot;Library size vs. genes detected&quot;,
         subtitle=&quot;Cells dropped in red&quot;,
         x=&quot;Sum of expression values by cell&quot;,
         y=&quot;# of non-mitochondrial genes&quot;) +
    geom_density_2d(colour=&quot;black&quot;, alpha=0.6) +
    get_boolean_colour_scale() +
    scale_y_log10() +
    guides(colour=&quot;none&quot;)


  mito2 &lt;- ggplot(as.data.frame(colData(sce)),
                  aes(x=subsets_Mt_percent,
                      y=subsets_genes_detected,
                      color=to_be_dropped)) +
    geom_point(size=0.5) +
    labs(title=&quot;Mito. content v. genes detected&quot;,
         subtitle=&quot;Cells dropped in red&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;Genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d(colour=&quot;black&quot;, alpha=0.6) +
    get_boolean_colour_scale() +
    guides(colour=&quot;none&quot;)

  gridExtra::grid.arrange(mito1, mito2, nrow=1)
}

show_mito_plots(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-9" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%20after%20QC-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-QC:-Mitochondrial-content-after-QC-1">
Past versions of “plots for QC: Mitochondrial content after QC-1.png”
</button>
</p>
<div id="fig-no-spaces-plots-for-QC:-Mitochondrial-content-after-QC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/plots for QC: Mitochondrial content after QC-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>show_mito_plots(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-9" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%20after%20QC-2.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plots-for-QC:-Mitochondrial-content-after-QC-2">
Past versions of “plots for QC: Mitochondrial content after QC-2.png”
</button>
</p>
<div id="fig-no-spaces-plots-for-QC:-Mitochondrial-content-after-QC-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/plots for QC: Mitochondrial content after QC-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="gene-and-expression-counts-stratified-by-sample" class="section level1">
<h1>Gene and expression counts stratified by sample</h1>
<div id="coloured-by-to_be_dropped" class="section level2 tabset">
<h2 class="tabset">Coloured by to_be_dropped</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-10" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=to_be_dropped)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Cells dropped in red&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  get_boolean_colour_scale() +
  guides(colour=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20samples-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-stratified-samples-1">
Past versions of “stratified samples-1.png”
</button>
</p>
<div id="fig-no-spaces-stratified-samples-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/stratified samples-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh and cultured samples\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-10" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=to_be_dropped)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Cells dropped in red&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  get_boolean_colour_scale() +
  guides(colour=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20samples-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-stratified-samples-2">
Past versions of “stratified samples-2.png”
</button>
</p>
<div id="fig-no-spaces-stratified-samples-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/stratified samples-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="coloured-by-mitochondrial-content" class="section level2 tabset">
<h2 class="tabset">Coloured by mitochondrial content</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-11" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=subsets_Mt_percent)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Coloured by fraction of mitochondrial counts&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  scale_colour_viridis_c(option = &quot;inferno&quot;, limits=c(0,50), oob=scales::squish)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20coloured%20by%20mt%20percent-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-stratified-coloured-by-mt-percent-1">
Past versions of “stratified coloured by mt percent-1.png”
</button>
</p>
<div id="fig-no-spaces-stratified-coloured-by-mt-percent-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/stratified coloured by mt percent-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh and cultured samples\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-11" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=subsets_Mt_percent)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Coloured by fraction of mitochondrial counts&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  scale_colour_viridis_c(option = &quot;inferno&quot;, limits=c(0,50), oob=scales::squish)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20coloured%20by%20mt%20percent-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-stratified-coloured-by-mt-percent-2">
Past versions of “stratified coloured by mt percent-2.png”
</button>
</p>
<div id="fig-no-spaces-stratified-coloured-by-mt-percent-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/stratified coloured by mt percent-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="library-size-distributions-before-and-after-deleting-cells-marked-in-qc" class="section level1 tabset">
<h1 class="tabset">Library size distributions before and after deleting cells marked in QC</h1>
<div id="by-sample" class="section level2 tabset">
<h2 class="tabset">By sample</h2>
<pre class="r"><code># Fresh only before QC
cat(paste0(&quot;\n\n### Fresh samples only - before QC \n\n&quot;))</code></pre>
<div id="fresh-samples-only---before-qc" class="section level3">
<h3>Fresh samples only - before QC</h3>
<pre class="r"><code>sce_fresh %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_fresh)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_fresh)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-log10-counts-by-sample-1">
Past versions of “log10 counts by sample-1.png”
</button>
</p>
<div id="fig-no-spaces-log10-counts-by-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/log10 counts by sample-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Fresh only after QC
cat(paste0(&quot;\n\n### Fresh samples only - after QC \n\n&quot;))</code></pre>
</div>
<div id="fresh-samples-only---after-qc" class="section level3">
<h3>Fresh samples only - after QC</h3>
<pre class="r"><code>sce_fresh[, !sce_fresh$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_fresh)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_fresh)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-log10-counts-by-sample-2">
Past versions of “log10 counts by sample-2.png”
</button>
</p>
<div id="fig-no-spaces-log10-counts-by-sample-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/log10 counts by sample-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Fresh and culture before QC
cat(paste0(&quot;\n\n### Fresh and cultured samples - before QC\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples---before-qc" class="section level3">
<h3>Fresh and cultured samples - before QC</h3>
<pre class="r"><code>sce_all %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_all)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_all)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-3.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-log10-counts-by-sample-3">
Past versions of “log10 counts by sample-3.png”
</button>
</p>
<div id="fig-no-spaces-log10-counts-by-sample-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/log10 counts by sample-3.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Fresh and culture after QC
cat(paste0(&quot;\n\n### Fresh and cultured samples - after QC\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples---after-qc" class="section level3">
<h3>Fresh and cultured samples - after QC</h3>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_all)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_all)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-4.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-log10-counts-by-sample-4">
Past versions of “log10 counts by sample-4.png”
</button>
</p>
<div id="fig-no-spaces-log10-counts-by-sample-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/log10 counts by sample-4.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="by-protocol-for-fresh-and-cultured-samples" class="section level2 tabset">
<h2 class="tabset">By protocol for fresh and cultured samples</h2>
<pre class="r"><code>cat(&quot;\n\n### Before QC\n\n&quot;)</code></pre>
<div id="before-qc" class="section level3">
<h3>Before QC</h3>
<pre class="r"><code>sce_all %&gt;% 
  ggplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  geom_density(aes(log10(total), fill=Protocol, color=Protocol), alpha=0.3) +
  get_protocol_colours(levels(colData(sce_all)$Protocol), type=&quot;fill&quot;) +
  get_protocol_colours(levels(colData(sce_all)$Protocol))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20protocol-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-log10-counts-by-protocol-1">
Past versions of “log10 counts by protocol-1.png”
</button>
</p>
<div id="fig-no-spaces-log10-counts-by-protocol-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/log10 counts by protocol-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n### After QC\n\n&quot;)</code></pre>
</div>
<div id="after-qc" class="section level3">
<h3>After QC</h3>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  geom_density(aes(log10(total), fill=Protocol, color=Protocol), alpha=0.3) +
  get_protocol_colours(levels(colData(sce_all)$Protocol), type=&quot;fill&quot;) +
  get_protocol_colours(levels(colData(sce_all)$Protocol))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20protocol-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-log10-counts-by-protocol-2">
Past versions of “log10 counts by protocol-2.png”
</button>
</p>
<div id="fig-no-spaces-log10-counts-by-protocol-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/log10 counts by protocol-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="gene-count-distributions-after-deleting-cells-marked-in-qc" class="section level1">
<h1>Gene count distributions after deleting cells marked in QC</h1>
<div id="by-sample-1" class="section level2 tabset">
<h2 class="tabset">By sample</h2>
<pre class="r"><code># Fresh only after QC
cat(paste0(&quot;\n\n### Fresh samples only - after QC \n\n&quot;))</code></pre>
<div id="fresh-samples-only---after-qc-1" class="section level3">
<h3>Fresh samples only - after QC</h3>
<pre class="r"><code>sce_fresh[, !sce_fresh$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(detected, fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of gene counts&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_fresh)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_fresh)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/gene%20counts%20by%20sample-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-gene-counts-by-sample-1">
Past versions of “gene counts by sample-1.png”
</button>
</p>
<div id="fig-no-spaces-gene-counts-by-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/gene counts by sample-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Fresh and culture after QC
cat(paste0(&quot;\n\n### Fresh and cultured samples - after QC\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples---after-qc-1" class="section level3">
<h3>Fresh and cultured samples - after QC</h3>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(detected, fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of gene counts&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_all)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_all)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/gene%20counts%20by%20sample-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-gene-counts-by-sample-2">
Past versions of “gene counts by sample-2.png”
</button>
</p>
<div id="fig-no-spaces-gene-counts-by-sample-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/gene counts by sample-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="by-protocol-for-fresh-and-cultured-samples-1" class="section level2">
<h2>By protocol for fresh and cultured samples</h2>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of gene counts&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  geom_density(aes(detected, fill=Protocol, color=Protocol), alpha=0.3) +
  get_protocol_colours(levels(colData(sce_all)$Protocol), type=&quot;fill&quot;) +
  get_protocol_colours(levels(colData(sce_all)$Protocol))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/geme%20counts%20by%20protocol-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-geme-counts-by-protocol-1">
Past versions of “geme counts by protocol-1.png”
</button>
</p>
<div id="fig-no-spaces-geme-counts-by-protocol-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/geme counts by protocol-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section" class="section level1 unnumbered">
<h1 class="unnumbered"></h1>
<div id="number-of-cells-dropped-and-kept-per-sample" class="section level2">
<h2>Number of cells dropped and kept per sample</h2>
<pre class="r"><code>### Fresh only ###
cat(paste0(&quot;\n\n### Fresh samples only {.tabset}\n\n&quot;))</code></pre>
<div id="fresh-samples-only-12" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<pre class="r"><code># Absolute values
cat(paste0(&quot;\n\n#### Absolute \n\n&quot;))</code></pre>
<div id="absolute" class="section level4">
<h4>Absolute</h4>
<pre class="r"><code>table(sce_fresh$to_be_dropped, sce_fresh$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;stack&quot;) +
  labs(title=&quot;Absolute number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Barplot-of-cells-to-be-dropped-1">
Past versions of “Barplot of cells to be dropped-1.png”
</button>
</p>
<div id="fig-no-spaces-Barplot-of-cells-to-be-dropped-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Barplot of cells to be dropped-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Relative values
cat(paste0(&quot;\n\n#### Relative \n\n&quot;))</code></pre>
</div>
<div id="relative" class="section level4">
<h4>Relative</h4>
<pre class="r"><code>table(sce_fresh$to_be_dropped, sce_fresh$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;fill&quot;) +
  labs(title=&quot;Relative number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Barplot-of-cells-to-be-dropped-2">
Past versions of “Barplot of cells to be dropped-2.png”
</button>
</p>
<div id="fig-no-spaces-Barplot-of-cells-to-be-dropped-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Barplot of cells to be dropped-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>### Fresh v culture ###
cat(paste0(&quot;\n\n### Fresh and cultured samples {.tabset}\n\n&quot;))</code></pre>
</div>
</div>
<div id="fresh-and-cultured-samples-12" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<pre class="r"><code># Absolute values
cat(paste0(&quot;\n\n#### Absolute \n\n&quot;))</code></pre>
<div id="absolute-1" class="section level4">
<h4>Absolute</h4>
<pre class="r"><code>table(sce_all$to_be_dropped, sce_all$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;stack&quot;) +
  labs(title=&quot;Absolute number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Barplot-of-cells-to-be-dropped-3">
Past versions of “Barplot of cells to be dropped-3.png”
</button>
</p>
<div id="fig-no-spaces-Barplot-of-cells-to-be-dropped-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Barplot of cells to be dropped-3.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Relative values
cat(paste0(&quot;\n\n#### Relative \n\n&quot;))</code></pre>
</div>
<div id="relative-1" class="section level4">
<h4>Relative</h4>
<pre class="r"><code>table(sce_all$to_be_dropped, sce_all$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;fill&quot;) +
  labs(title=&quot;Relative number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Barplot-of-cells-to-be-dropped-4">
Past versions of “Barplot of cells to be dropped-4.png”
</button>
</p>
<div id="fig-no-spaces-Barplot-of-cells-to-be-dropped-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Barplot of cells to be dropped-4.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="finally-drop-cells-from-dataset" class="section level2">
<h2>Finally drop cells from dataset</h2>
<pre class="r fold-show"><code>sce_fresh_filtered &lt;- sce_fresh[, !sce_fresh$to_be_dropped]
sce_all_filtered &lt;- sce_all[, !sce_all$to_be_dropped]

# Drop genes which are expressed in only one or no cells at all
# These genes do not have an influence on the remaining analysis and 
# can cause issues with normalisation

# fresh cells 
mask &lt;- rowSums(counts(sce_fresh_filtered) &gt; 0) &gt; 1
print(sum(mask==FALSE))</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r fold-show"><code>sce_fresh_filtered &lt;- sce_fresh_filtered[mask,]

# fresh and cultured cells 
mask &lt;- rowSums(counts(sce_all_filtered) &gt; 0) &gt; 1
print(sum(mask==FALSE))</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r fold-show"><code>sce_all_filtered &lt;- sce_all_filtered[mask,]</code></pre>
</div>
</div>
<div id="normalisation" class="section level1 tabset">
<h1 class="tabset">Normalisation</h1>
<p>We normalise the data with scTransform</p>
<!-- ```{r normalisation, results='asis', eval=F} -->
<!-- cat(paste0("\n\n## Fresh samples only \n\n")) -->
<!-- lib.sce_fresh <- librarySizeFactors(sce_fresh_filtered) -->
<!-- ggplot(data.frame("x"=log10(lib.sce_fresh)), aes(x)) + -->
<!--   geom_histogram(color="black", fill="grey80") + -->
<!--   labs(title="Distribution of normalisation factors", -->
<!--        subtitle="Each normalisation factors are multiplied with the expression values of a given cell", -->
<!--        x="Log10 of normalisation factor", -->
<!--        y="Frequency") -->
<!-- sce_fresh_filtered <- logNormCounts(sce_fresh_filtered) -->
<!-- cat(paste0("\n\n## Fresh and cultured samples \n\n")) -->
<!-- lib.sce_all <- librarySizeFactors(sce_all_filtered) -->
<!-- ggplot(data.frame("x"=log10(lib.sce_all)), aes(x)) + -->
<!--   geom_histogram(color="black", fill="grey80") + -->
<!--   labs(title="Distribution of normalisation factors", -->
<!--        subtitle="Each normalisation factors are multiplied with the expression values of a given cell", -->
<!--        x="Log10 of normalisation factor", -->
<!--        y="Frequency") -->
<!-- sce_all_filtered <- logNormCounts(sce_all_filtered) -->
<!-- ``` -->
<pre class="r"><code>normalise_with_scTransform &lt;- function(sce){
  
  sce &lt;- logNormCounts(sce)

  sce_seurat &lt;- as.Seurat(sce, counts=&quot;counts&quot;, data=&quot;logcounts&quot;)
  
  seurat_normalized = sce_seurat %&gt;%
    Seurat::SCTransform(verbose = FALSE, assay=&quot;originalexp&quot;,  method=&quot;glmGamPoi&quot;, min_cells=1)
  
  sct_HVGs &lt;- seurat_normalized@assays$SCT@var.features
  
  assay(sce, &quot;corrected&quot;) &lt;- seurat_normalized@assays$SCT@counts
  assay(sce, &quot;logcounts&quot;) &lt;- seurat_normalized@assays$SCT@data
  rowData(sce)[, &quot;sct_HVG&quot;] &lt;- FALSE
  rowData(sce)[sct_HVGs, &quot;sct_HVG&quot;] &lt;- TRUE
  
  sce
}


sce_fresh_filtered &lt;- normalise_with_scTransform(sce_fresh_filtered)</code></pre>
<pre><code>Warning: Invalid name supplied, making object name syntactically valid. New
object name is Seurat..SCTransform.originalexp; see ?make.names for more details
on syntax validity</code></pre>
<pre class="r"><code>sce_all_filtered &lt;- normalise_with_scTransform(sce_all_filtered)</code></pre>
<pre><code>Warning: Invalid name supplied, making object name syntactically valid. New
object name is Seurat..SCTransform.originalexp; see ?make.names for more details
on syntax validity</code></pre>
</div>
<div id="feature-selection---hvgs" class="section level1 tabset">
<h1 class="tabset">Feature Selection - HVGs</h1>
<p>We select the 2000 most variable genes</p>
<pre class="r"><code>num_hvg=2000

select_hvgs &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n## &quot;, sce_name, &quot; \n\n&quot;))

  # model the gene variance
  dec.sce &lt;- scran::modelGeneVar(sce, block=sce$Sample)
  hvg_sub &lt;- getTopHVGs(dec.sce, n=num_hvg)
  
  mean_var_comb_sub &lt;- purrr::map(unique(sce$Sample), ~ {
   dec.sce$per.block[[.x]] %&gt;% 
    as_tibble %&gt;% 
    mutate(row_names = rownames(dec.sce$per.block[[.x]]),
           is_hvg = row_names %in% hvg_sub,
           Sample = .x)
    }) %&gt;% 
    purrr::reduce(rbind)
  
  print(mean_var_comb_sub %&gt;% 
    ggplot() +
      geom_point(aes(x = mean, y = total, color= is_hvg)) +
      geom_line(aes(x=mean, y= tech)) +
      labs(y=&quot;Variance&quot;,x=&quot;Mean expression&quot;) +
      facet_wrap(~Sample) +
      scale_color_manual(values=c(&quot;orange&quot;,&quot;black&quot;)) +
      labs(x=&quot;Mean of log-expression&quot;, 
           y=&quot;Variance of log-expression&quot;,
           title=&quot;Variance of the expression values as a function of the mean&quot;,
           subtitle=paste0(&quot;Top &quot;, num_hvg, &quot; highly variable genes&quot;)))
    
  list(&quot;sco&quot;=dec.sce, &quot;hvg&quot;=hvg_sub)
}

res.fresh &lt;- select_hvgs(sce_fresh_filtered, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-13" class="section level2">
<h2>Fresh samples only</h2>
<pre><code>Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):
collapsing to unique &#39;x&#39; values</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/feature%20selection-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-selection-1">
Past versions of “feature selection-1.png”
</button>
</p>
<div id="fig-no-spaces-feature-selection-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/feature selection-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dec.sce.fresh &lt;- res.fresh$sco
hvg.fresh &lt;- res.fresh$hvg

res.all &lt;- select_hvgs(sce_all_filtered, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-13" class="section level2">
<h2>Fresh and cultured samples</h2>
<pre><code>Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):
collapsing to unique &#39;x&#39; values

Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):
collapsing to unique &#39;x&#39; values

Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):
collapsing to unique &#39;x&#39; values

Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):
collapsing to unique &#39;x&#39; values

Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):
collapsing to unique &#39;x&#39; values</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/feature%20selection-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-feature-selection-2">
Past versions of “feature selection-2.png”
</button>
</p>
<div id="fig-no-spaces-feature-selection-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/feature selection-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dec.sce.all &lt;- res.all$sco
hvg.all &lt;- res.all$hvg</code></pre>
</div>
</div>
<div id="section-1" class="section level1">
<h1></h1>
<div id="peek-at-highly-variable-genes" class="section level2 tabset">
<h2 class="tabset">Peek at highly variable genes</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-14" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>print(kable_styling(kable(dec.sce.fresh[order(dec.sce.fresh$bio, decreasing=TRUE)[1:10],], digits=2)))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
total
</th>
<th style="text-align:right;">
tech
</th>
<th style="text-align:right;">
bio
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
per.block.Fresh_S1.mean
</th>
<th style="text-align:right;">
per.block.Fresh_S1.total
</th>
<th style="text-align:right;">
per.block.Fresh_S1.tech
</th>
<th style="text-align:right;">
per.block.Fresh_S1.bio
</th>
<th style="text-align:right;">
per.block.Fresh_S1.p.value
</th>
<th style="text-align:right;">
per.block.Fresh_S1.FDR
</th>
<th style="text-align:right;">
per.block.Fresh_S2.mean
</th>
<th style="text-align:right;">
per.block.Fresh_S2.total
</th>
<th style="text-align:right;">
per.block.Fresh_S2.tech
</th>
<th style="text-align:right;">
per.block.Fresh_S2.bio
</th>
<th style="text-align:right;">
per.block.Fresh_S2.p.value
</th>
<th style="text-align:right;">
per.block.Fresh_S2.FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PTGDS
</td>
<td style="text-align:right;">
1.28
</td>
<td style="text-align:right;">
3.37
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
2.93
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.43
</td>
<td style="text-align:right;">
4.41
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
3.97
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.13
</td>
<td style="text-align:right;">
2.34
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
1.89
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
DCN
</td>
<td style="text-align:right;">
1.58
</td>
<td style="text-align:right;">
3.33
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
2.87
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.41
</td>
<td style="text-align:right;">
3.25
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
2.81
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.74
</td>
<td style="text-align:right;">
3.40
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
2.94
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
VIM
</td>
<td style="text-align:right;">
3.48
</td>
<td style="text-align:right;">
2.63
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
2.39
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.33
</td>
<td style="text-align:right;">
4.10
</td>
<td style="text-align:right;">
0.21
</td>
<td style="text-align:right;">
3.89
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.63
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
0.90
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL1A1
</td>
<td style="text-align:right;">
1.46
</td>
<td style="text-align:right;">
2.72
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
2.33
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
0.95
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
0.56
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.18
</td>
<td style="text-align:right;">
4.48
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
4.09
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL1A2
</td>
<td style="text-align:right;">
1.55
</td>
<td style="text-align:right;">
2.73
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
2.31
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
1.53
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
1.09
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.10
</td>
<td style="text-align:right;">
3.93
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
3.52
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL3A1
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
2.30
</td>
<td style="text-align:right;">
0.38
</td>
<td style="text-align:right;">
1.93
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
0.74
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.04
</td>
<td style="text-align:right;">
3.86
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
3.45
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL6A2
</td>
<td style="text-align:right;">
1.56
</td>
<td style="text-align:right;">
2.34
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
1.89
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
2.29
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
1.85
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.85
</td>
<td style="text-align:right;">
2.38
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
1.93
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL6A1
</td>
<td style="text-align:right;">
1.53
</td>
<td style="text-align:right;">
2.27
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
1.83
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
1.89
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
1.45
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.90
</td>
<td style="text-align:right;">
2.64
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
2.20
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
CXCL14
</td>
<td style="text-align:right;">
1.30
</td>
<td style="text-align:right;">
2.27
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
1.81
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.07
</td>
<td style="text-align:right;">
1.48
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
1.04
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.53
</td>
<td style="text-align:right;">
3.05
</td>
<td style="text-align:right;">
0.47
</td>
<td style="text-align:right;">
2.58
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
IGFBP7
</td>
<td style="text-align:right;">
2.03
</td>
<td style="text-align:right;">
2.08
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
1.69
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.93
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:right;">
0.38
</td>
<td style="text-align:right;">
1.85
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.13
</td>
<td style="text-align:right;">
1.93
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
1.54
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh and cultured samples \n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-14" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>print(kable_styling(kable(dec.sce.all[order(dec.sce.all$bio, decreasing=TRUE)[1:10],], digits=2)))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
total
</th>
<th style="text-align:right;">
tech
</th>
<th style="text-align:right;">
bio
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:right;">
per.block.Culture_S1.mean
</th>
<th style="text-align:right;">
per.block.Culture_S1.total
</th>
<th style="text-align:right;">
per.block.Culture_S1.tech
</th>
<th style="text-align:right;">
per.block.Culture_S1.bio
</th>
<th style="text-align:right;">
per.block.Culture_S1.p.value
</th>
<th style="text-align:right;">
per.block.Culture_S1.FDR
</th>
<th style="text-align:right;">
per.block.Culture_S2.mean
</th>
<th style="text-align:right;">
per.block.Culture_S2.total
</th>
<th style="text-align:right;">
per.block.Culture_S2.tech
</th>
<th style="text-align:right;">
per.block.Culture_S2.bio
</th>
<th style="text-align:right;">
per.block.Culture_S2.p.value
</th>
<th style="text-align:right;">
per.block.Culture_S2.FDR
</th>
<th style="text-align:right;">
per.block.Culture_S4.mean
</th>
<th style="text-align:right;">
per.block.Culture_S4.total
</th>
<th style="text-align:right;">
per.block.Culture_S4.tech
</th>
<th style="text-align:right;">
per.block.Culture_S4.bio
</th>
<th style="text-align:right;">
per.block.Culture_S4.p.value
</th>
<th style="text-align:right;">
per.block.Culture_S4.FDR
</th>
<th style="text-align:right;">
per.block.Culture_S5.mean
</th>
<th style="text-align:right;">
per.block.Culture_S5.total
</th>
<th style="text-align:right;">
per.block.Culture_S5.tech
</th>
<th style="text-align:right;">
per.block.Culture_S5.bio
</th>
<th style="text-align:right;">
per.block.Culture_S5.p.value
</th>
<th style="text-align:right;">
per.block.Culture_S5.FDR
</th>
<th style="text-align:right;">
per.block.Fresh_S1.mean
</th>
<th style="text-align:right;">
per.block.Fresh_S1.total
</th>
<th style="text-align:right;">
per.block.Fresh_S1.tech
</th>
<th style="text-align:right;">
per.block.Fresh_S1.bio
</th>
<th style="text-align:right;">
per.block.Fresh_S1.p.value
</th>
<th style="text-align:right;">
per.block.Fresh_S1.FDR
</th>
<th style="text-align:right;">
per.block.Fresh_S2.mean
</th>
<th style="text-align:right;">
per.block.Fresh_S2.total
</th>
<th style="text-align:right;">
per.block.Fresh_S2.tech
</th>
<th style="text-align:right;">
per.block.Fresh_S2.bio
</th>
<th style="text-align:right;">
per.block.Fresh_S2.p.value
</th>
<th style="text-align:right;">
per.block.Fresh_S2.FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CXCL8
</td>
<td style="text-align:right;">
1.57
</td>
<td style="text-align:right;">
3.23
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
2.88
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.08
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
3.93
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
2.46
</td>
<td style="text-align:right;">
4.78
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
4.46
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.85
</td>
<td style="text-align:right;">
3.68
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
3.20
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.17
</td>
<td style="text-align:right;">
4.60
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
4.25
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
1.06
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
0.81
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.47
</td>
<td style="text-align:right;">
0.93
</td>
<td style="text-align:right;">
0.30
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
MT2A
</td>
<td style="text-align:right;">
2.67
</td>
<td style="text-align:right;">
2.41
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
2.09
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.00
</td>
<td style="text-align:right;">
2.51
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
2.10
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
3.22
</td>
<td style="text-align:right;">
2.55
</td>
<td style="text-align:right;">
0.30
</td>
<td style="text-align:right;">
2.24
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.56
</td>
<td style="text-align:right;">
2.34
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
1.99
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.13
</td>
<td style="text-align:right;">
3.27
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
2.99
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.63
</td>
<td style="text-align:right;">
2.22
</td>
<td style="text-align:right;">
0.26
</td>
<td style="text-align:right;">
1.97
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
2.45
</td>
<td style="text-align:right;">
1.58
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
1.26
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
MMP1
</td>
<td style="text-align:right;">
1.21
</td>
<td style="text-align:right;">
2.36
</td>
<td style="text-align:right;">
0.31
</td>
<td style="text-align:right;">
2.04
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.25
</td>
<td style="text-align:right;">
2.68
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
2.27
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
2.03
</td>
<td style="text-align:right;">
4.33
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
3.93
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.73
</td>
<td style="text-align:right;">
3.35
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
2.86
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.00
</td>
<td style="text-align:right;">
3.57
</td>
<td style="text-align:right;">
0.37
</td>
<td style="text-align:right;">
3.20
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.56
</td>
<td style="text-align:right;">
0.18
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
0.14
</td>
<td style="text-align:right;">
-0.02
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.75
</td>
</tr>
<tr>
<td style="text-align:left;">
MMP3
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
2.00
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
1.72
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.64
</td>
<td style="text-align:right;">
1.56
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
1.21
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1.33
</td>
<td style="text-align:right;">
3.78
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
3.37
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.92
</td>
<td style="text-align:right;">
2.49
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
2.04
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.51
</td>
<td style="text-align:right;">
4.10
</td>
<td style="text-align:right;">
0.38
</td>
<td style="text-align:right;">
3.72
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
-0.01
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.79
</td>
</tr>
<tr>
<td style="text-align:left;">
KRT14
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
1.97
</td>
<td style="text-align:right;">
0.28
</td>
<td style="text-align:right;">
1.69
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.72
</td>
<td style="text-align:right;">
5.04
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
4.59
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.33
</td>
<td style="text-align:right;">
0.88
</td>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.52
</td>
<td style="text-align:right;">
1.55
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
1.21
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
1.03
</td>
<td style="text-align:right;">
0.17
</td>
<td style="text-align:right;">
0.86
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.60
</td>
<td style="text-align:right;">
2.39
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
2.05
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.24
</td>
<td style="text-align:right;">
0.94
</td>
<td style="text-align:right;">
0.17
</td>
<td style="text-align:right;">
0.76
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
DCN
</td>
<td style="text-align:right;">
1.12
</td>
<td style="text-align:right;">
2.03
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
1.63
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.53
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
0.23
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.11
</td>
<td style="text-align:right;">
1.17
</td>
<td style="text-align:right;">
1.78
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
1.39
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
1.13
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
0.71
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.22
</td>
<td style="text-align:right;">
1.94
</td>
<td style="text-align:right;">
0.38
</td>
<td style="text-align:right;">
1.56
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.35
</td>
<td style="text-align:right;">
3.26
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
2.82
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1.67
</td>
<td style="text-align:right;">
3.52
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
3.06
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
VIM
</td>
<td style="text-align:right;">
3.25
</td>
<td style="text-align:right;">
1.88
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
1.59
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.51
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
1.94
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
3.75
</td>
<td style="text-align:right;">
1.08
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
0.69
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.03
</td>
<td style="text-align:right;">
1.67
</td>
<td style="text-align:right;">
0.31
</td>
<td style="text-align:right;">
1.35
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.31
</td>
<td style="text-align:right;">
1.15
</td>
<td style="text-align:right;">
0.28
</td>
<td style="text-align:right;">
0.86
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.29
</td>
<td style="text-align:right;">
4.02
</td>
<td style="text-align:right;">
0.21
</td>
<td style="text-align:right;">
3.81
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
3.58
</td>
<td style="text-align:right;">
1.15
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
0.88
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
CXCL3
</td>
<td style="text-align:right;">
1.05
</td>
<td style="text-align:right;">
1.94
</td>
<td style="text-align:right;">
0.37
</td>
<td style="text-align:right;">
1.56
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
1.91
</td>
<td style="text-align:right;">
0.39
</td>
<td style="text-align:right;">
1.52
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1.27
</td>
<td style="text-align:right;">
2.42
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
2.01
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.25
</td>
<td style="text-align:right;">
2.43
</td>
<td style="text-align:right;">
0.48
</td>
<td style="text-align:right;">
1.95
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.82
</td>
<td style="text-align:right;">
3.36
</td>
<td style="text-align:right;">
0.38
</td>
<td style="text-align:right;">
2.98
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.48
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.48
</td>
<td style="text-align:right;">
0.72
</td>
<td style="text-align:right;">
0.31
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
PTGDS
</td>
<td style="text-align:right;">
0.70
</td>
<td style="text-align:right;">
1.68
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
1.35
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
0.38
</td>
<td style="text-align:right;">
0.20
</td>
<td style="text-align:right;">
0.18
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
1.32
</td>
<td style="text-align:right;">
0.35
</td>
<td style="text-align:right;">
0.97
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.36
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
0.26
</td>
<td style="text-align:right;">
0.40
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
0.80
</td>
<td style="text-align:right;">
0.25
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1.37
</td>
<td style="text-align:right;">
4.43
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
3.98
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1.01
</td>
<td style="text-align:right;">
2.46
</td>
<td style="text-align:right;">
0.43
</td>
<td style="text-align:right;">
2.03
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
<tr>
<td style="text-align:left;">
SOD2
</td>
<td style="text-align:right;">
2.47
</td>
<td style="text-align:right;">
1.68
</td>
<td style="text-align:right;">
0.34
</td>
<td style="text-align:right;">
1.35
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.16
</td>
<td style="text-align:right;">
1.69
</td>
<td style="text-align:right;">
0.38
</td>
<td style="text-align:right;">
1.32
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
2.72
</td>
<td style="text-align:right;">
1.66
</td>
<td style="text-align:right;">
0.29
</td>
<td style="text-align:right;">
1.37
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.76
</td>
<td style="text-align:right;">
1.73
</td>
<td style="text-align:right;">
0.32
</td>
<td style="text-align:right;">
1.41
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
3.13
</td>
<td style="text-align:right;">
1.96
</td>
<td style="text-align:right;">
0.27
</td>
<td style="text-align:right;">
1.69
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
2.15
</td>
<td style="text-align:right;">
1.83
</td>
<td style="text-align:right;">
0.33
</td>
<td style="text-align:right;">
1.51
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
1.94
</td>
<td style="text-align:right;">
1.22
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
0.79
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="subset-selection-of-hvgs" class="section level2">
<h2>Subset selection of HVGs</h2>
<pre class="r"><code># fresh
rowData(sce_fresh_filtered)[,&quot;is_hvg&quot;] &lt;- F
rowData(sce_fresh_filtered)[hvg.fresh,&quot;is_hvg&quot;] &lt;- T

# fresh and culture
rowData(sce_all_filtered)[,&quot;is_hvg&quot;] &lt;- F
rowData(sce_all_filtered)[hvg.all,&quot;is_hvg&quot;] &lt;- T


# if we want to delete the other genes
# sce &lt;- sce[hvg,]</code></pre>
</div>
</div>
<div id="batch-correction" class="section level1 tabset">
<h1 class="tabset">Batch correction</h1>
<p>For batch correction we use the Harmony method. We only run batch correction for the single-cell experiments with merged samples.</p>
<pre class="r"><code>run_batch_correction &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n## &quot;, sce_name, &quot; {.tabset}\n\n&quot;))

  # Before batch effects
  # run UMAP
  set.seed(100)
  sce &lt;- runPCA(sce, exprs_values=&quot;logcounts&quot;, subset_row=rowData(sce)$is_hvg)

  set.seed(100)
  sce &lt;- runUMAP(sce, dimred=&quot;PCA&quot;, n_neighbors=50, min_dist = 0.1, subset_row=rowData(sce)$is_hvg)

  cat(&quot;\n\n### UMAP uncorrected \n\n&quot;)
  print(plotReducedDim(sce, &quot;UMAP&quot;, colour_by=&quot;Sample&quot;, point_size=0.5) +
    ggtitle(&quot;UMAP prior to batch correction&quot;) +
    guides(colour=guide_legend(override.aes=list(size=4))) +
    theme_ipsum_rc())

  # correct batch effects with Harmony
  # set.seed(200)

  meta_data &lt;- colData(sce) %&gt;%
    as.data.frame() %&gt;%
    dplyr::select(c(Protocol, Sample))
  
  set.seed(100)
  reducedDim(sce, &quot;harmony&quot;) &lt;- harmony::HarmonyMatrix(data_mat=reducedDim(sce, &quot;PCA&quot;),
                                                meta_data= meta_data,
                                                vars_use=c(&quot;Sample&quot;),
                                                do_pca = FALSE, verbose=FALSE)


  set.seed(100)
  sce &lt;- runUMAP(sce, dimred=&quot;harmony&quot;, n_neighbors=50, min_dist = 0.1) #, subset_row=rowData(sce)$is_hvg

  cat(&quot;\n\n### UMAP corrected \n\n&quot;)
  print(plotReducedDim(sce, &quot;UMAP&quot;, colour_by=&quot;Sample&quot;, point_size=0.5) +
    ggtitle(&quot;UMAP post batch correction - Harmony&quot;) +
    guides(colour=guide_legend(override.aes=list(size=4))) +
    theme_ipsum_rc())

    sce
}

sce_fresh_filtered &lt;- run_batch_correction(sce_fresh_filtered, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-15" class="section level2 tabset">
<h2 class="tabset">Fresh samples only</h2>
<div id="umap-uncorrected" class="section level3">
<h3>UMAP uncorrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-batch-correction-harmony-1">
Past versions of “batch correction harmony-1.png”
</button>
</p>
<div id="fig-no-spaces-batch-correction-harmony-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/batch correction harmony-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="umap-corrected" class="section level3">
<h3>UMAP corrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-batch-correction-harmony-2">
Past versions of “batch correction harmony-2.png”
</button>
</p>
<div id="fig-no-spaces-batch-correction-harmony-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/batch correction harmony-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>sce_all_filtered &lt;- run_batch_correction(sce_all_filtered, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
</div>
<div id="fresh-and-cultured-samples-15" class="section level2 tabset">
<h2 class="tabset">Fresh and cultured samples</h2>
<div id="umap-uncorrected-1" class="section level3">
<h3>UMAP uncorrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-3.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-batch-correction-harmony-3">
Past versions of “batch correction harmony-3.png”
</button>
</p>
<div id="fig-no-spaces-batch-correction-harmony-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/batch correction harmony-3.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="umap-corrected-1" class="section level3">
<h3>UMAP corrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-4.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-batch-correction-harmony-4">
Past versions of “batch correction harmony-4.png”
</button>
</p>
<div id="fig-no-spaces-batch-correction-harmony-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/batch correction harmony-4.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="clustering-with-k50" class="section level1">
<h1>Clustering with k=50</h1>
<pre class="r fold-show"><code>run_clustering &lt;- function(sce){

  set.seed(100)
  g50 &lt;- buildSNNGraph(sce, k=50, use.dimred = &#39;harmony&#39;, subset.row=rowData(sce)$is_hvg) # build graph

  set.seed(100)
  clust50 &lt;- igraph::cluster_walktrap(g50)$membership # find groups

  # Assign labels based on clusters
  colLabels(sce) &lt;- factor(clust50)

  sce
}

sce_fresh_filtered &lt;- run_clustering(sce_fresh_filtered)
sce_all_filtered &lt;- run_clustering(sce_all_filtered)</code></pre>
<div id="visualise-clusters-and-samples" class="section level2 tabset">
<h2 class="tabset">Visualise clusters and samples</h2>
<pre class="r"><code># fresh data
cat(&quot;\n\n### Fresh samples only  {.tabset}\n\n&quot;)</code></pre>
<div id="fresh-samples-only-16" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<pre class="r"><code># by sample
cat(&quot;\n\n#### Sample\n\n&quot;)</code></pre>
<div id="sample" class="section level4">
<h4>Sample</h4>
<pre class="r"><code>plotReducedDim(sce_fresh_filtered, &quot;UMAP&quot;, point_size=1,
               colour_by=&quot;Sample&quot;) +
  labs(&quot;UMAP coloured by sample&quot;) +
  guides(colour = guide_legend(override.aes = list(size=2))) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/_%20UMAP%20visualisation-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-_-UMAP-visualisation-1">
Past versions of “_ UMAP visualisation-1.png”
</button>
</p>
<div id="fig-no-spaces-_-UMAP-visualisation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-1.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># by label
cat(&quot;\n\n#### Label\n\n&quot;)</code></pre>
</div>
<div id="label" class="section level4">
<h4>Label</h4>
<pre class="r"><code>plotReducedDim(sce_fresh_filtered, &quot;UMAP&quot;, point_size=1,
               colour_by=&quot;label&quot;,
               text_by = &quot;label&quot;) +
  labs(&quot;UMAP coloured by cluster label&quot;) +
  guides(colour = guide_legend(override.aes = list(size=2))) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/_%20UMAP%20visualisation-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-_-UMAP-visualisation-2">
Past versions of “_ UMAP visualisation-2.png”
</button>
</p>
<div id="fig-no-spaces-_-UMAP-visualisation-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-2.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Fresh and cultured data
cat(&quot;\n\n### Fresh and cultured samples {.tabset}\n\n&quot;)</code></pre>
</div>
</div>
<div id="fresh-and-cultured-samples-16" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<pre class="r"><code># by sample
cat(&quot;\n\n#### Sample\n\n&quot;)</code></pre>
<div id="sample-1" class="section level4">
<h4>Sample</h4>
<pre class="r"><code>plotReducedDim(sce_all_filtered, &quot;UMAP&quot;, point_size=1,
               colour_by=&quot;Sample&quot;) +
  labs(&quot;UMAP coloured by sample&quot;) +
  guides(colour = guide_legend(override.aes = list(size=2))) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/_%20UMAP%20visualisation-3.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-_-UMAP-visualisation-3">
Past versions of “_ UMAP visualisation-3.png”
</button>
</p>
<div id="fig-no-spaces-_-UMAP-visualisation-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-3.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-3.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># by label
cat(&quot;\n\n#### Label\n\n&quot;)</code></pre>
</div>
<div id="label-1" class="section level4">
<h4>Label</h4>
<pre class="r"><code>plotReducedDim(sce_all_filtered, &quot;UMAP&quot;, point_size=1,
               colour_by=&quot;label&quot;,
               text_by = &quot;label&quot;) +
  labs(&quot;UMAP coloured by cluster label&quot;) +
  guides(colour = guide_legend(override.aes = list(size=2))) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/_%20UMAP%20visualisation-4.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-_-UMAP-visualisation-4">
Past versions of “_ UMAP visualisation-4.png”
</button>
</p>
<div id="fig-no-spaces-_-UMAP-visualisation-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-4.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-4.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># by Protocol
cat(&quot;\n\n#### Protocol\n\n&quot;)</code></pre>
</div>
<div id="protocol" class="section level4">
<h4>Protocol</h4>
<pre class="r"><code>plotReducedDim(sce_all_filtered, &quot;UMAP&quot;, point_size=1,
               colour_by=&quot;Protocol&quot;) +
  labs(&quot;UMAP coloured by protocol&quot;) +
  guides(colour = guide_legend(override.aes = list(size=2))) +
  theme_ipsum_rc()</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/_%20UMAP%20visualisation-5.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-_-UMAP-visualisation-5">
Past versions of “_ UMAP visualisation-5.png”
</button>
</p>
<div id="fig-no-spaces-_-UMAP-visualisation-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-5.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/_ UMAP visualisation-5.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="show-cluster-to-sample-mapping" class="section level2 tabset">
<h2 class="tabset">Show cluster to sample mapping</h2>
<p>The table below shows the mappings from unsupervised clustering to each individual sample. The clusters refer to the clustering performed on the dataset with all samples merged.</p>
<pre class="r"><code>plot_cluster_to_sample_mapping &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; \n\n&quot;))
  
  tab &lt;- table(sce$label, sce$Sample)
  tab_annotation &lt;- tab
  tab_annotation[tab_annotation==0] &lt;- &quot;&quot;

  as.ggplot(pheatmap(tab, color=brewer.pal(n=9, name=&quot;Oranges&quot;), display_numbers=tab_annotation, silent=T)) +
    theme_ipsum_rc() +
    labs(title=&quot;Clusters and samples&quot;) +
    theme(axis.line=element_blank(), 
          axis.text.x=element_blank(),
          axis.text.y=element_blank(), 
          axis.ticks=element_blank(),
          axis.title.x=element_blank(),
          axis.title.y=element_blank(), 
          legend.position=&quot;none&quot;,
          panel.background=element_blank(),
          panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          plot.background=element_blank())
}

plot_cluster_to_sample_mapping(sce_fresh_filtered, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-17" class="section level3">
<h3>Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/cluster%20to%20sample%20assignment-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-cluster-to-sample-assignment-1">
Past versions of “cluster to sample assignment-1.png”
</button>
</p>
<div id="fig-no-spaces-cluster-to-sample-assignment-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/c4b210d95296965afb4378ce3e6c9dbef53901f5/docs/figure/P2_01_data_comparison.Rmd/cluster to sample assignment-1.png" target="_blank">c4b210d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-02-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/cluster to sample assignment-1.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/cluster to sample assignment-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_cluster_to_sample_mapping(sce_all_filtered, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-17" class="section level3">
<h3>Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/cluster%20to%20sample%20assignment-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-cluster-to-sample-assignment-2">
Past versions of “cluster to sample assignment-2.png”
</button>
</p>
<div id="fig-no-spaces-cluster-to-sample-assignment-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/c4b210d95296965afb4378ce3e6c9dbef53901f5/docs/figure/P2_01_data_comparison.Rmd/cluster to sample assignment-2.png" target="_blank">c4b210d</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-02-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/97b097bcd93105d275ebb47c066a8129eb61b5b2/docs/figure/P2_01_data_comparison.Rmd/cluster to sample assignment-2.png" target="_blank">97b097b</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/cluster to sample assignment-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="marker-genes" class="section level1">
<h1>Marker genes</h1>
<p>Marker genes can help us identify clusters as specific cell types and also discover smaller changes between subclusters (e.g., changes in activation or differentiation state). We use the wilcoxon rank sum test.</p>
<pre class="r"><code>wb_fresh &lt;- createWorkbook()
addWorksheet(wb_fresh, &quot;Intro&quot;)
writeData(wb_fresh, &quot;Intro&quot;, data.frame(&quot;Marker genes for fresh samples only&quot;), colNames=FALSE)

wb_all &lt;- createWorkbook()
addWorksheet(wb_all, &quot;Intro&quot;)
writeData(wb_all, &quot;Intro&quot;, data.frame(&quot;Marker genes for fresh and cultured samples&quot;), colNames=FALSE)</code></pre>
<pre class="r"><code>get_marker_genes &lt;- function(sce, sce_name, test_name, type_comparison, excel_workbook=NA, worksheet_prefix=&quot;&quot;){
  
  cat(paste0(&quot;\n\n## &quot;, sce_name, &quot; {.tabset} \n\n&quot;))
  
  prefix &lt;- ifelse(test_name==&quot;t&quot;, &quot;logFC&quot;, &quot;AUC&quot;)
  
  # set limits depending on the marker method used (AUC only goes from 0 to 1)
  heatmap_lower_limit &lt;- ifelse(test_name==&quot;t&quot;, heatmaplimits[0], 0)
  heatmap_upper_limit &lt;- ifelse(test_name==&quot;t&quot;, heatmaplimits[1], 1)

  # we filter for only the highly variable genes. Otherwise we will see many genes which
  # are *significantly* different (high p-values), but not marginally different (low logFCs)
  markers.all.up &lt;- findMarkers(sce[rowData(sce)$is_hvg], 
                                test.type=test_name,
                                pval.type=type_comparison,
                                direction=&quot;up&quot;, 
                                block=sce$Sample)

  for (i in 1:length(unique(names(markers.all.up)))){

    cat(&quot;\n\n### &quot;, &quot;Cluster &quot;, i, &quot; \n\n&quot;)

    interesting.all.up &lt;- markers.all.up[[i]]

    # print(kable_styling(kable(interesting.all.up[1:10,], digits=2)))
    # cat(&quot;\n&quot;)

    # Per default we only look at union of the top 5 markers per pairwise comparison
    if (type_comparison==&quot;any&quot;){
      cluster_markers &lt;- interesting.all.up[interesting.all.up$Top &lt;= 10,]  
    } else {
      cluster_markers &lt;- interesting.all.up[1:50,]
    }
    
    cluster_markers_logFCs &lt;- getMarkerEffects(cluster_markers, prefix = prefix)
    cluster_markers_logFCs[is.na(cluster_markers_logFCs)] &lt;- 0
    
    # Create pheatmap plot per cluster
    heatmap_plt &lt;- as.ggplot(pheatmap(cluster_markers_logFCs, breaks=seq(heatmap_lower_limit, heatmap_upper_limit, length.out=101), silent=T, cluster_cols=FALSE)) +
      labs(title=paste0(&quot;Cluster &quot;, i), subtitle=&quot;Pairwise comparisons&quot;) 
      theme_ipsum_rc() +
      theme(axis.title.x=element_blank(),
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_blank(),
      axis.ticks.y=element_blank(),
      panel.grid.major = element_blank(), 
      panel.grid.minor = element_blank())
    
    print(heatmap_plt)
    
    # write to excel if a workbook was specified. Else skip
    if (is.na(excel_workbook)==FALSE){
        sheet_name &lt;- paste0(worksheet_prefix, &quot;_c&quot;, i)
        addWorksheet(excel_workbook, sheet_name)
        cluster_markers_excel_out &lt;- cbind(&quot;Gene&quot;=rownames(interesting.all.up), interesting.all.up)
        writeData(excel_workbook, sheet_name, cluster_markers_excel_out, rowNames = TRUE)
    }
  }
  
  # Either only return the marker results or also the updated excel
  if (is.na(excel_workbook) == TRUE){
    markers.all.up
  } else {
    return(list(&quot;marker_results&quot;=markers.all.up,
                &quot;excel_workbook&quot;=excel_workbook))
  }

}


res_fresh &lt;- get_marker_genes(sce_fresh_filtered, &quot;Fresh samples only&quot;, test=&quot;wilcox&quot;, type_comparison=&quot;any&quot;, excel_workbook=wb_fresh)</code></pre>
<div id="fresh-samples-only-18" class="section level2 tabset">
<h2 class="tabset">Fresh samples only</h2>
<div id="cluster-1" class="section level3">
<h3>Cluster 1</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-1">
Past versions of “Marker genes with function-1.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-2" class="section level3">
<h3>Cluster 2</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-2">
Past versions of “Marker genes with function-2.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-3" class="section level3">
<h3>Cluster 3</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-3.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-3">
Past versions of “Marker genes with function-3.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-3.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-4" class="section level3">
<h3>Cluster 4</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-4.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-4">
Past versions of “Marker genes with function-4.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-4.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-5" class="section level3">
<h3>Cluster 5</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-5.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-5">
Past versions of “Marker genes with function-5.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-5.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-6" class="section level3">
<h3>Cluster 6</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-6.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-6">
Past versions of “Marker genes with function-6.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-6.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-7" class="section level3">
<h3>Cluster 7</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-7.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-7">
Past versions of “Marker genes with function-7.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-7.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-8" class="section level3">
<h3>Cluster 8</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-8.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-8">
Past versions of “Marker genes with function-8.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-8.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-9" class="section level3">
<h3>Cluster 9</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;

Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-9.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-9">
Past versions of “Marker genes with function-9.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-9.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>markers_fresh &lt;- res_fresh$marker_results
wb_fresh &lt;- res_fresh$excel_workbook

# Fresh and cultured
res_all &lt;- get_marker_genes(sce_all_filtered, &quot;Fresh and cultured samples&quot;, test=&quot;wilcox&quot;, type_comparison=&quot;any&quot;, excel_workbook=wb_all)</code></pre>
</div>
</div>
<div id="fresh-and-cultured-samples-18" class="section level2 tabset">
<h2 class="tabset">Fresh and cultured samples</h2>
<div id="cluster-1-1" class="section level3">
<h3>Cluster 1</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-10.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-10">
Past versions of “Marker genes with function-10.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-10.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-2-1" class="section level3">
<h3>Cluster 2</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-11.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-11">
Past versions of “Marker genes with function-11.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-11.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-3-1" class="section level3">
<h3>Cluster 3</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-12.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-12">
Past versions of “Marker genes with function-12.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-12.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-4-1" class="section level3">
<h3>Cluster 4</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-13.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-13">
Past versions of “Marker genes with function-13.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-13.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-5-1" class="section level3">
<h3>Cluster 5</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-14.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-14">
Past versions of “Marker genes with function-14.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-14.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-6-1" class="section level3">
<h3>Cluster 6</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-15.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-15">
Past versions of “Marker genes with function-15.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-15.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-7-1" class="section level3">
<h3>Cluster 7</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-16.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-16">
Past versions of “Marker genes with function-16.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-16.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-8-1" class="section level3">
<h3>Cluster 8</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-17.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-17">
Past versions of “Marker genes with function-17.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-17.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-9-1" class="section level3">
<h3>Cluster 9</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-18.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-18">
Past versions of “Marker genes with function-18.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-18.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-10" class="section level3">
<h3>Cluster 10</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-19.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-19">
Past versions of “Marker genes with function-19.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-19.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-11" class="section level3">
<h3>Cluster 11</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-20.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-20">
Past versions of “Marker genes with function-20.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-20.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-12" class="section level3">
<h3>Cluster 12</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-21.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-21">
Past versions of “Marker genes with function-21.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-21.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-13" class="section level3">
<h3>Cluster 13</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-22.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-22">
Past versions of “Marker genes with function-22.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-22.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-14" class="section level3">
<h3>Cluster 14</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-23.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-23">
Past versions of “Marker genes with function-23.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-23.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-15" class="section level3">
<h3>Cluster 15</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-24.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-24">
Past versions of “Marker genes with function-24.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-24.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster-16" class="section level3">
<h3>Cluster 16</h3>
<pre><code>Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;

Warning in is.na(excel_workbook): is.na() applied to non-(list or vector) of
type &#39;S4&#39;</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Marker%20genes%20with%20function-25.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Marker-genes-with-function-25">
Past versions of “Marker genes with function-25.png”
</button>
</p>
<div id="fig-no-spaces-Marker-genes-with-function-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/Marker genes with function-25.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>markers_all &lt;- res_all$marker_results
wb_all &lt;- res_all$excel_workbook</code></pre>
</div>
</div>
</div>
<div id="average-gene-expression-values-for-marker-genes-of-all-clusters" class="section level1 tabset">
<h1 class="tabset">Average gene expression values for marker genes of all clusters</h1>
<pre class="r"><code>plot_heatmap_of_top_genes &lt;- function(sce, markers, top_n=5){
  # function assumes the sce columns: label, Sample

  # get top genes
  top_genes &lt;- lapply(markers, function(marker_set){
    marker_set %&gt;%
      as.data.frame() %&gt;%
      filter(Top &lt;= top_n) %&gt;%
      rownames()
  }) %&gt;%
    unlist(use.names=F) %&gt;%
    unique()

  # Matrix to plot the heatmap of expression values.
  # To avoid a too big plot we only keep a fraction of the cells i there are too many
  nth_cell &lt;- max(1, round(dim(sce)[2] / 1000,0))
  sce_plot &lt;- sce[top_genes, seq(1, dim(sce)[2], nth_cell)]
  # We sort by cluster
  sce_plot &lt;- sce_plot[,order(sce_plot$label, sce_plot$Sample)]
  # Anotations
  column_annotations &lt;- data.frame(&quot;Sample&quot;=sce_plot$Sample, &quot;Cluster&quot;=factor(sce_plot$label), row.names=colnames(sce_plot))
  # clip heatmap values to ensure that colour contrast is high
  sce_plot &lt;- assay(sce_plot, &quot;logcounts&quot;)
  sce_plot[sce_plot &gt;=5] = 5

  print(as.ggplot(pheatmap(sce_plot, annotation_col=column_annotations,
           show_rownames=TRUE, show_colnames=FALSE, cluster_cols=FALSE, silent=TRUE,
           color=inferno(50))) +
    labs(title=&quot;Gene signatures per cluster identified&quot;,
         subtitle=paste0(&quot;Top &quot;, top_n, &quot; differentially expressed genes per cluster; values are log-counts clipped at 5&quot;)) +
    theme_ipsum_rc()+
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
            axis.text.y=element_blank(),axis.ticks=element_blank(),
            axis.title.x=element_blank(),
            axis.title.y=element_blank(),legend.position=&quot;none&quot;,
            panel.background=element_blank(),panel.border=element_blank(),panel.grid.major=element_blank(),
            panel.grid.minor=element_blank(),plot.background=element_blank()))
}</code></pre>
<div id="fresh-samples-only-19" class="section level2">
<h2>Fresh samples only</h2>
<pre class="r"><code>plot_heatmap_of_top_genes(sce_fresh_filtered, markers_fresh)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/average%20marker%20gene%20expressions%20fresh%20only-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-average-marker-gene-expressions-fresh-only-1">
Past versions of “average marker gene expressions fresh only-1.png”
</button>
</p>
<div id="fig-no-spaces-average-marker-gene-expressions-fresh-only-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/average marker gene expressions fresh only-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fresh-and-cultured-samples-19" class="section level2">
<h2>Fresh and cultured samples</h2>
<pre class="r"><code>plot_heatmap_of_top_genes(sce_all_filtered, markers_all)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/average%20marker%20gene%20expressions%20fresh%20and%20culture%20-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-average-marker-gene-expressions-fresh-and-culture--1">
Past versions of “average marker gene expressions fresh and culture -1.png”
</button>
</p>
<div id="fig-no-spaces-average-marker-gene-expressions-fresh-and-culture--1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/average marker gene expressions fresh and culture -1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<!-- # Marker genes identified by entropy & f-stat {.tabset} -->
<!-- The following shows the marker genes as identified via the entropy-fstat-method for the dataset combining all samples as well as the dataset for only the fresh samples combined. -->
<!-- ```{r entropy f-stat, results="asis", fig.width=10, fig.height=7, results="asis", warning=FALSE} -->
<!-- for (annotation_name in c("All", "Fresh_all")){ -->
<!--   if (annotation_name %in% master[["annotation_name"]]){ -->
<!--     sample_name <- master[master$annotation_name==annotation_name, "sample_name"] -->
<!--     cat(paste0("\n\n## ", sample_name, " {.tabset} \n\n")) -->
<!--     sce_object <- sces_filtered[[as.integer(master[master$annotation_name==annotation_name, "num_id"])]] -->
<!--     sce_object <- sce_object[rowData(sce_object)$is_hvg,] -->
<!--     # sce_object <- sce_object[rowData(sce_object)$is_hvg==TRUE,] -->
<!--     shannon_entropy <- function(x, e = 1e-10) { -->
<!--       x <- x+e # allow zeros in x -->
<!--       p <- (x)/sum(x) -->
<!--       -sum((p)*log(p)) -->
<!--     } -->
<!--     aggr_syn_sce <- aggregateAcrossCells(sce_object, sce_object$label) -->
<!--     entropy_fstat_sce <- BiocParallel::bplapply( -->
<!--       seq_along(rownames(aggr_syn_sce)), -->
<!--       function(.x) { -->
<!--         data.frame( -->
<!--           gene = rowData(sce_object)$Symbol[.x], -->
<!--           entropy = shannon_entropy(counts(aggr_syn_sce[.x,])), -->
<!--           fstat = summary(aov(logcounts(sce_object)[.x, ] ~ colLabels(sce_object)))[[1]]$`F value`[1] -->
<!--         ) -->
<!--         }, BPPARAM = BiocParallel::MulticoreParam(workers=n_workers, RNGseed = 123)) %>% -->
<!--       purrr::reduce(rbind) -->
<!--     entropy_fstat_sce <- entropy_fstat_sce %>% -->
<!--       dplyr::mutate(entropy_norm = (entropy+1e-3)/(max(entropy)+1e-3), -->
<!--                     fstat_norm = fstat/max(fstat,na.rm = TRUE), -->
<!--                     ent_fstat_ratio = fstat_norm / entropy_norm, na.rm = TRUE) -->
<!--     topmarkers <- order(entropy_fstat_sce$ent_fstat_ratio, decreasing = TRUE)[1:50] -->
<!--     topmarker_knownmarkers <- unique(entropy_fstat_sce[topmarkers,]) -->
<!--     cat("\n\n### Entropy-fstat plot \n\n") -->
<!--     print(ggplot(entropy_fstat_sce) + -->
<!--       geom_point(aes(x=log(fstat), y=entropy), show.legend = F, colour="#02C39A") + -->
<!--       labs(x="log(F-statistic)", y="Entropy", main="Entropy vs. F-statistic of cluster separation") + -->
<!--       ggrepel::geom_text_repel(data=topmarker_knownmarkers, -->
<!--                                aes(x=log(fstat), y=entropy, label=gene, size=4), -->
<!--                                show.legend=F, -->
<!--                                max.overlaps = 30)) -->
<!--     # write to excel -->
<!--     sheet_name <- paste0(annotation_name, "_fstat_top_markers") -->
<!--     addWorksheet(wb, sheet_name) -->
<!--     writeData(wb, sheet_name, topmarker_knownmarkers, rowNames=TRUE) -->
<!--     # Expression plots -->
<!--     cat("\n\n### Expression plots for all samples combined {.tabset} \n\n") -->
<!--     genes_to_plot <-  unique(entropy_fstat_sce$gene[topmarkers]) -->
<!--     rownames(sce_object) <- rowData(sce_object)$Symbol -->
<!--     for(i in seq_along(genes_to_plot)){ -->
<!--       cat("\n\n#### ", genes_to_plot[i], "\n\n") -->
<!--       print(plotExpression(sce_object, -->
<!--                      entropy_fstat_sce$gene[entropy_fstat_sce$gene == genes_to_plot[i]], -->
<!--                      x = "label", colour_by = "Sample", point_size=0.75)) -->
<!--     } -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<pre class="r"><code># Fresh only
filename_marker_results &lt;- &quot;../docs/output/P2_Fresh_only.xlsx&quot;
saveWorkbook(wb_fresh, file = filename_marker_results, overwrite = TRUE)

# Fresh and cultured
filename_marker_results &lt;- &quot;../docs/output/P2_Fresh_v_culture.xlsx&quot;
saveWorkbook(wb_all, file = filename_marker_results, overwrite = TRUE)</code></pre>
</div>
</div>
<div id="foreign-annotations" class="section level1 tabset">
<h1 class="tabset">Foreign Annotations</h1>
<pre class="r"><code>plot_frequencies &lt;- function(sce_object){
  plt &lt;- table(colData(sce_object)[,annotations_name], colData(sce_object)[,&quot;Sample&quot;]) %&gt;%
    as.data.frame() %&gt;%
    dplyr::rename(&quot;Cell.Type&quot;=&quot;Var1&quot;, &quot;Sample&quot;=&quot;Var2&quot;) %&gt;%
    ggplot(aes(x=Sample, y=Freq, fill=Cell.Type)) +
    geom_bar(stat=&quot;identity&quot;, position = &quot;fill&quot;, width = 0.85) +
    theme(panel.grid = element_blank(),
          axis.text = element_text(size = 10),
          axis.text.x = element_text(angle=30, hjust=1),
          axis.text.y = element_text(size = 10),
          plot.subtitle=element_text(size=12)) +
    labs(title=&quot;Relative cell type abundances across samples&quot;,
         subtitle=paste0(&quot;Labels: &quot;, annotations_name))

  return(plt)
}</code></pre>
<div id="blaz" class="section level2 tabset">
<h2 class="tabset">Blaz</h2>
<pre class="r"><code>csv_name &lt;- &quot;../Metadata/cell_annotation_markers/Markers_Blaz.csv&quot;
group_colname &lt;- &quot;Cell.Type&quot;
gene_colname &lt;- &quot;Markers&quot;
annotations_name &lt;- &quot;labels_blaz&quot;

cat(&quot;\n\n### Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-20" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>sce_fresh_filtered &lt;- annotate_with_other_labels(sce_fresh_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c(&quot;Barchart&quot;, &quot;UMAP&quot;))</code></pre>
<pre><code>Genes in the gene sets NOT available in the dataset: 
    Keratinocyte:   1 (5% of 21)
    Fibroblast:     2 (8% of 26)
    Myeloid cell:   2 (17% of 12)
    Lymphocyte:     2 (20% of 10)
    Melanocyte:     1 (20% of 5)</code></pre>
</div>
<div id="umap" class="section level3">
<h3>UMAP</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/labels%20Blaz-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-labels-Blaz-1">
Past versions of “labels Blaz-1.png”
</button>
</p>
<div id="fig-no-spaces-labels-Blaz-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/labels Blaz-1.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat(&quot;\n\n### Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-20" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>sce_all_filtered &lt;- annotate_with_other_labels(sce_all_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c(&quot;Barchart&quot;, &quot;UMAP&quot;), tabset=&quot;###&quot;)</code></pre>
<pre><code>Genes in the gene sets NOT available in the dataset: 
    Keratinocyte:   1 (5% of 21)
    Fibroblast:     2 (8% of 26)
    Myeloid cell:   2 (17% of 12)
    Lymphocyte:     2 (20% of 10)
    Melanocyte:     1 (20% of 5)</code></pre>
</div>
<div id="umap-1" class="section level3">
<h3>UMAP</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/labels%20Blaz-2.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-labels-Blaz-2">
Past versions of “labels Blaz-2.png”
</button>
</p>
<div id="fig-no-spaces-labels-Blaz-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/DominiquePaul/Skin-cell-protocol-scRNA/blob/1923ae36314462857191f55817eed51512a6c060/docs/figure/P2_01_data_comparison.Rmd/labels Blaz-2.png" target="_blank">1923ae3</a>
</td>
<td>
dominique-paul-uzh
</td>
<td>
2022-01-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<!-- ## Raynolds {.tabset} -->
<!-- ```{r labels Raynolds, fig.width=16, fig.height=10, results="asis"} -->
<!-- csv_name <- "../Metadata/cell_annotation_markers/Markers_Raynolds.csv" -->
<!-- group_colname <- "Cell.Type" -->
<!-- gene_colname <- "Gene" -->
<!-- annotations_name <- "labels_raynolds" -->
<!-- cat("\n\n### Fresh samples only") -->
<!-- sce_fresh_filtered <- annotate_with_other_labels(sce_fresh_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP")) -->
<!-- cat("\n\n### Fresh and cultured samples") -->
<!-- sce_all_filtered <- annotate_with_other_labels(sce_all_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP"), tabset="###") -->
<!-- ``` -->
<!-- ## He et al. {.tabset} -->
<!-- ```{r He et al., fig.width=16, fig.height=10, results="asis"} -->
<!-- csv_name <- "../Metadata/cell_annotation_markers/Markers_He.csv" -->
<!-- group_colname <- "cluster_fine" -->
<!-- gene_colname <- "gene" -->
<!-- annotations_name <- "labels_he_et_al_fine" -->
<!-- cat("\n\n### Fresh samples only") -->
<!-- sce_fresh_filtered <- annotate_with_other_labels(sce_fresh_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP")) -->
<!-- cat("\n\n### Fresh and cultured samples") -->
<!-- sce_all_filtered <- annotate_with_other_labels(sce_all_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP"), tabset="###") -->
<!-- ``` -->
<!-- ## Tabib {.tabset} -->
<!-- ```{r Tabib, fig.width=16, fig.height=10, results="asis"} -->
<!-- csv_name <- "../Metadata/cell_annotation_markers/Markers_Tabib.csv" -->
<!-- group_colname <- "Cell.Type" -->
<!-- gene_colname <- "Marker" -->
<!-- annotations_name <- "labels_tabib" -->
<!-- cat("\n\n### Fresh samples only") -->
<!-- sce_fresh_filtered <- annotate_with_other_labels(sce_fresh_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP")) -->
<!-- cat("\n\n### Fresh and cultured samples") -->
<!-- sce_all_filtered <- annotate_with_other_labels(sce_all_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP"), tabset="###") -->
<!-- ``` -->
<!-- ## Xue {.tabset} -->
<!-- ```{r Xue, fig.width=16, fig.height=10, results="asis"} -->
<!-- csv_name <- "../Metadata/cell_annotation_markers/Markers_Xue.csv" -->
<!-- group_colname <- "Cell.Type" -->
<!-- gene_colname <- "Marker" -->
<!-- annotations_name <- "labels_xue" -->
<!-- sce_fresh_filtered <- annotate_with_other_labels(sce_fresh_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP")) -->
<!-- sce_all_filtered <- annotate_with_other_labels(sce_all_filtered, csv_name, group_colname, gene_colname, annotations_name, visualisations=c("Barchart", "UMAP"), tabset="###") -->
<!-- ``` -->
</div>
</div>
</div>
<div id="save-sce-objects-and-upload-them-to-isee" class="section level1">
<h1>Save SCE objects and upload them to iSEE</h1>
<pre class="r fold-show"><code># Save all objects as one rds

# Fresh samples only
file_path_fresh_unfiltered &lt;- &quot;../data/EOS_files/Fresh_EOS1_sce_unfiltered.rds&quot;
saveRDS(sce_fresh, file=file_path_fresh_unfiltered)

file_path_fresh &lt;- &quot;../data/EOS_files/Fresh_EOS1_sce.rds&quot;
saveRDS(sce_fresh_filtered, file=file_path_fresh)
upload_file_to_iSEE(file_path_fresh)</code></pre>
<pre><code>Command executed: 

scp -i ~/.ssh/id_imls_servers /Users/dominiquepaul/xRobinson\ Group/02_Protocol/data/EOS_Files/Fresh_EOS1_sce.rds dominique@imlspenticton.uzh.ch:/home/Shared/retger/synovial/data/protocol_paper_BBDP/sce/Fresh_EOS1_sce___2022-02-01.rds</code></pre>
<pre class="r fold-show"><code># Fresh and cultured samples
file_path_all_unfiltered &lt;- &quot;../data/EOS_files/FvC_EOS1_sce_unfiltered.rds&quot;
saveRDS(sce_all, file=file_path_all_unfiltered)

file_path_all &lt;- &quot;../data/EOS_files/FvC_EOS1_sce.rds&quot;
saveRDS(sce_all_filtered, file=file_path_all)
upload_file_to_iSEE(file_path_all)</code></pre>
<pre><code>Command executed: 

scp -i ~/.ssh/id_imls_servers /Users/dominiquepaul/xRobinson\ Group/02_Protocol/data/EOS_Files/FvC_EOS1_sce.rds dominique@imlspenticton.uzh.ch:/home/Shared/retger/synovial/data/protocol_paper_BBDP/sce/FvC_EOS1_sce___2022-02-01.rds</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] stringr_1.4.0                  SeuratObject_4.0.4            
 [3] Seurat_4.0.6                   BiocParallel_1.28.3           
 [5] dplyr_1.0.7                    plyr_1.8.6                    
 [7] RColorBrewer_1.1-2             batchelor_1.10.0              
 [9] ggplotify_0.1.0                openxlsx_4.2.5                
[11] tidySingleCellExperiment_1.4.0 viridis_0.6.2                 
[13] viridisLite_0.4.0              scDblFinder_1.8.0             
[15] hrbrthemes_0.8.6               AUCell_1.16.0                 
[17] GSEABase_1.56.0                graph_1.72.0                  
[19] annotate_1.72.0                XML_3.99-0.8                  
[21] AnnotationDbi_1.56.2           pheatmap_1.0.12               
[23] kableExtra_1.3.4               scran_1.22.1                  
[25] scater_1.22.0                  scuttle_1.4.0                 
[27] SingleCellExperiment_1.16.0    SummarizedExperiment_1.24.0   
[29] Biobase_2.54.0                 GenomicRanges_1.46.1          
[31] GenomeInfoDb_1.30.0            IRanges_2.28.0                
[33] S4Vectors_0.32.3               BiocGenerics_0.40.0           
[35] MatrixGenerics_1.6.0           matrixStats_0.61.0            
[37] ggplot2_3.3.5                  workflowr_1.7.0               

loaded via a namespace (and not attached):
  [1] scattermore_0.7           R.methodsS3_1.8.1        
  [3] tidyr_1.1.4               bit64_4.0.5              
  [5] knitr_1.37                irlba_2.3.5              
  [7] DelayedArray_0.20.0       R.utils_2.11.0           
  [9] rpart_4.1-15              data.table_1.14.2        
 [11] KEGGREST_1.34.0           RCurl_1.98-1.5           
 [13] generics_0.1.1            ScaledMatrix_1.2.0       
 [15] callr_3.7.0               cowplot_1.1.1            
 [17] RSQLite_2.2.9             RANN_2.6.1               
 [19] future_1.23.0             bit_4.0.4                
 [21] spatstat.data_2.1-2       webshot_0.5.2            
 [23] xml2_1.3.3                httpuv_1.6.4             
 [25] isoband_0.2.5             assertthat_0.2.1         
 [27] xfun_0.29                 jquerylib_0.1.4          
 [29] evaluate_0.14             promises_1.2.0.1         
 [31] fansi_1.0.2               igraph_1.2.10            
 [33] DBI_1.1.2                 htmlwidgets_1.5.4        
 [35] spatstat.geom_2.3-1       purrr_0.3.4              
 [37] ellipsis_0.3.2            RSpectra_0.16-0          
 [39] deldir_1.0-6              sparseMatrixStats_1.6.0  
 [41] vctrs_0.3.8               ROCR_1.0-11              
 [43] abind_1.4-5               cachem_1.0.6             
 [45] withr_2.4.3               sctransform_0.3.2        
 [47] goftest_1.2-3             svglite_2.0.0            
 [49] cluster_2.1.2             lazyeval_0.2.2           
 [51] crayon_1.4.2              labeling_0.4.2           
 [53] edgeR_3.36.0              pkgconfig_2.0.3          
 [55] nlme_3.1-153              vipor_0.4.5              
 [57] rlang_0.4.12              globals_0.14.0           
 [59] lifecycle_1.0.1           miniUI_0.1.1.1           
 [61] extrafontdb_1.0           rsvd_1.0.5               
 [63] polyclip_1.10-0           rprojroot_2.0.2          
 [65] lmtest_0.9-39             Matrix_1.4-0             
 [67] Rhdf5lib_1.16.0           zoo_1.8-9                
 [69] beeswarm_0.4.0            whisker_0.4              
 [71] ggridges_0.5.3            processx_3.5.2           
 [73] png_0.1-7                 bitops_1.0-7             
 [75] getPass_0.2-2             R.oo_1.24.0              
 [77] rhdf5filters_1.6.0        KernSmooth_2.23-20       
 [79] Biostrings_2.62.0         blob_1.2.2               
 [81] DelayedMatrixStats_1.16.0 parallelly_1.30.0        
 [83] gridGraphics_0.5-1        beachmat_2.10.0          
 [85] scales_1.1.1              memoise_2.0.1            
 [87] magrittr_2.0.1            ica_1.0-2                
 [89] zlibbioc_1.40.0           compiler_4.1.0           
 [91] dqrng_0.3.0               fitdistrplus_1.1-6       
 [93] cli_3.1.0                 XVector_0.34.0           
 [95] listenv_0.8.0             patchwork_1.1.1          
 [97] pbapply_1.5-0             ps_1.6.0                 
 [99] mgcv_1.8-38               MASS_7.3-54              
[101] tidyselect_1.1.1          stringi_1.7.6            
[103] glmGamPoi_1.6.0           highr_0.9                
[105] yaml_2.2.1                BiocSingular_1.10.0      
[107] locfit_1.5-9.4            ggrepel_0.9.1            
[109] grid_4.1.0                sass_0.4.0               
[111] tools_4.1.0               future.apply_1.8.1       
[113] rstudioapi_0.13           bluster_1.4.0            
[115] git2r_0.29.0              metapod_1.2.0            
[117] gridExtra_2.3             farver_2.1.0             
[119] Rtsne_0.15                DropletUtils_1.14.1      
[121] digest_0.6.29             shiny_1.7.1              
[123] Rcpp_1.0.8                harmony_0.1.0            
[125] later_1.3.0               RcppAnnoy_0.0.19         
[127] httr_1.4.2                gdtools_0.2.3            
[129] colorspace_2.0-2          rvest_1.0.2              
[131] fs_1.5.2                  tensor_1.5               
[133] reticulate_1.22           splines_4.1.0            
[135] uwot_0.1.11               yulab.utils_0.0.4        
[137] statmod_1.4.36            spatstat.utils_2.3-0     
[139] xgboost_1.5.0.2           plotly_4.10.0            
[141] systemfonts_1.0.3         xtable_1.8-4             
[143] jsonlite_1.7.3            R6_2.5.1                 
[145] pillar_1.6.4              htmltools_0.5.2          
[147] mime_0.12                 glue_1.6.0               
[149] fastmap_1.1.0             BiocNeighbors_1.12.0     
[151] codetools_0.2-18          utf8_1.2.2               
[153] lattice_0.20-45           bslib_0.3.1              
[155] spatstat.sparse_2.1-0     tibble_3.1.6             
[157] ResidualMatrix_1.4.0      ggbeeswarm_0.6.0         
[159] leiden_0.3.9              zip_2.2.0                
[161] Rttf2pt1_1.3.8            survival_3.2-13          
[163] limma_3.50.0              rmarkdown_2.11           
[165] munsell_0.5.0             rhdf5_2.38.0             
[167] GenomeInfoDbData_1.2.7    HDF5Array_1.22.1         
[169] reshape2_1.4.4            gtable_0.3.0             
[171] spatstat.core_2.3-2       extrafont_0.17           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
