<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dominique Paul" />

<meta name="date" content="2021-12-28" />

<title>Comparison of the ‘new’ skin tissue protocol with skin samples 1,2,4 &amp; 5</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Protocol paper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Comparison of the ‘new’ skin tissue protocol with skin samples 1,2,4 &amp; 5</h1>
<h4 class="author">Dominique Paul</h4>
<h4 class="date">2021-12-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>02_Protocol/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211228code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211228)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211228code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211228)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Repository version:</strong> no version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol" class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the results is critical for reproducibility. To start using Git, open the Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full reproducibility benefits of using workflowr, please see <code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<div id="loading-and-preprocess-the-data" class="section level1">
<h1>Loading and preprocess the data</h1>
<div id="identify-cultured-samples" class="section level2">
<h2>Identify cultured samples</h2>
<pre class="r"><code># Skin control data loaded as one group
samples_skin &lt;- list.files(path=&quot;../data/skin_tissue/scRNA_datasets&quot;,
                        pattern=&quot;matrix.mtx&quot;,
                        full.names=TRUE,
                        recursive=TRUE)

samples_skin &lt;- dirname(samples_skin)
samples_skin &lt;- grep(&quot;control&quot;, samples_skin, value=TRUE) # Only use control data
samples_skin &lt;- grep(&quot;filtered&quot;, samples_skin, value=TRUE)

names(samples_skin) &lt;- sapply(strsplit(samples_skin, &quot;/&quot;), .subset, 5) # This number might have to be adapted for other file paths or folder structures

names(samples_skin) &lt;- gsub(&quot;o24300_1_13-control1_BB&quot;, &quot;Culture_S1&quot;, names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o24300_1_09-control2_MFB&quot;, &quot;Culture_S2&quot;, names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o24793_1_09-BB_skin_control&quot;, &quot;Culture_S3&quot;,names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o25154_1_1-SS08_control&quot;, &quot;Culture_S4&quot;, names(samples_skin))
names(samples_skin) &lt;- gsub(&quot;o25154_1_3-SS18_control&quot;, &quot;Culture_S5&quot;, names(samples_skin))

samples_skin &lt;- samples_skin[!names(samples_skin) %in% &quot;Culture_S3&quot;]</code></pre>
</div>
<div id="load-fresh-samples" class="section level2">
<h2>Load fresh samples</h2>
<pre class="r"><code># Fresh data loaded together
samples_fresh &lt;- list.files(path=&quot;../data/fresh_data&quot;,
                      pattern=&quot;matrix.mtx&quot;,
                      full.names=TRUE,
                      recursive=TRUE)

samples_fresh &lt;- dirname(samples_fresh)
samples_fresh &lt;- grep(&quot;filtered&quot;, samples_fresh, value=TRUE)

names(samples_fresh) &lt;- sapply(strsplit(samples_fresh, &quot;/&quot;), .subset, 4) # This number might have to be adapted for other file paths or folder structures

names(samples_fresh) &lt;- gsub(&quot;01_o25154_1_5-HC_fresh&quot;, &quot;Fresh_S1&quot;, names(samples_fresh))
names(samples_fresh) &lt;- gsub(&quot;02_o25459_1_6-HS_Blaz&quot;, &quot;Fresh_S2&quot;, names(samples_fresh))

# Load data
sce_fresh &lt;- DropletUtils::read10xCounts(samples=samples_fresh)

# Factoring
sce_fresh$Sample &lt;- factor(sce_fresh$Sample) # Factor the sample name to save space

# Renaming
rownames(sce_fresh) &lt;- rowData(sce_fresh)$Symbol # Rename the rows
colnames(sce_fresh) &lt;- paste0(sce_fresh$Sample, &quot;.&quot;, sce_fresh$Barcode) # Rename the columns

# add protocol name
sce_fresh$Protocol &lt;- factor(&quot;Fresh&quot;)</code></pre>
</div>
<div id="load-fresh-and-cultured-samples-as-one-dataset" class="section level2">
<h2>Load fresh and cultured samples as one dataset</h2>
<pre class="r"><code>samples_all &lt;- c(samples_skin, samples_fresh)

# Load data
sce_all &lt;- DropletUtils::read10xCounts(samples=samples_all)

# Factoring
sce_all$Sample &lt;- factor(sce_all$Sample) # Factor the sample name to save space

# Renaming
rownames(sce_all) &lt;- rowData(sce_all)$Symbol # Rename the rows
colnames(sce_all) &lt;- paste0(sce_all$Sample, &quot;.&quot;, sce_all$Barcode) # Rename the columns

# Add column for protocol
sce_all$Protocol &lt;- factor(ifelse(grepl(&quot;Fresh&quot;, sce_all$Sample), &quot;Fresh&quot;, &quot;Culture&quot;))</code></pre>
</div>
<div id="basic-filtering" class="section level2 tabset">
<h2 class="tabset">Basic filtering</h2>
<p>We run a crude filtering on the cells to eliminate the majority of bad reads and speed up the runtime of the code below.</p>
<pre class="r"><code>crude_filtering &lt;- function(sce, sce_name){
  
  original_sce_dimensions &lt;- dim(sce)
  
  sce &lt;- sce[rowSums(counts(sce)&gt;0) &gt; 10,  # We only keep genes expressed in more than ten cells
             colSums(counts(sce) &gt;0) &gt; 150] # We only keep cells with more than 150 genes expressed

  new_sce_dimensions &lt;- dim(sce)

  absolute_changes &lt;- original_sce_dimensions - new_sce_dimensions
  relative_changes &lt;- absolute_changes / original_sce_dimensions

  cat(paste0(&quot;\n\n ### &quot;, sce_name))
  cat(paste0(&quot;\n#### Dropped &quot;, absolute_changes[1], &quot; genes out of &quot;, original_sce_dimensions[1], &quot; (~&quot;, round(100*relative_changes[1],2), &quot;%). &quot;, new_sce_dimensions[1], &quot; genes remaining.\n\n&quot;))
  cat(paste0(&quot;\n#### Dropped &quot;, absolute_changes[2], &quot; cells out of &quot;, original_sce_dimensions[2], &quot; (~&quot;, round(100*relative_changes[2],2), &quot;%). &quot;, new_sce_dimensions[2], &quot; cells remaining.\n\n&quot; ))
  
  sce
}

sce_fresh &lt;- crude_filtering(sce_fresh, &quot;Fresh&quot;)</code></pre>
<p>### Fresh #### Dropped 5267 genes out of 21587 (~24.4%). 16320 genes remaining.</p>
<div id="dropped-189-cells-out-of-9941-1.9.-9752-cells-remaining." class="section level4">
<h4>Dropped 189 cells out of 9941 (~1.9%). 9752 cells remaining.</h4>
<pre class="r"><code>sce_all &lt;- crude_filtering(sce_all, &quot;All&quot;)</code></pre>
<p>### All #### Dropped 4090 genes out of 21587 (~18.95%). 17497 genes remaining.</p>
</div>
<div id="dropped-456-cells-out-of-28171-1.62.-27715-cells-remaining." class="section level4">
<h4>Dropped 456 cells out of 28171 (~1.62%). 27715 cells remaining.</h4>
</div>
</div>
</div>
<div id="quality-control" class="section level1">
<h1>Quality Control</h1>
<div id="mitochondria-check" class="section level2 tabset">
<h2 class="tabset">Mitochondria check</h2>
<p>Do we have mitochondrial genes in the data?</p>
<pre class="r"><code>identify_mitochondrial_genes &lt;- function(sce, sce_name){
  rowData(sce)$is_mitochondrial &lt;- grepl(&quot;^(MT-)&quot;, rowData(sce)$Symbol, ignore.case=TRUE)
  
  cat(paste0(&quot;\n\n ### &quot;, sce_name, &quot; \n\n&quot;))
  print(table(rowData(sce)$is_mitochondrial) %&gt;%
          as.data.frame() %&gt;%
          kable() %&gt;%
          kable_styling())
  
  sce
}

sce_fresh &lt;- identify_mitochondrial_genes(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<p>### Fresh samples only</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Var1
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
16285
</td>
</tr>
<tr>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
35
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>sce_all &lt;- identify_mitochondrial_genes(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
<p>### Fresh and cultured samples</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Var1
</th>
<th style="text-align:right;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
FALSE
</td>
<td style="text-align:right;">
17460
</td>
</tr>
<tr>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
37
</td>
</tr>
</tbody>
</table>
</div>
<div id="add-quality-control-metrics" class="section level2">
<h2>Add quality control metrics</h2>
<pre class="r"><code># Fresh only
sce_fresh &lt;- scater::addPerCellQC(sce_fresh, list(Mt = rowData(sce_fresh)$is_mitochondrial,
                                                  genes= !rowData(sce_fresh)$is_mitochondrial),
                                  percent_top=c(50, 100, 200, 500))

sce_fresh$log_sum &lt;- log10(sce_fresh$sum)
sce_fresh$log_total &lt;- log10(sce_fresh$total)
  
# Fresh and culture
sce_all &lt;- scater::addPerCellQC(sce_all, list(Mt = rowData(sce_all)$is_mitochondrial,
                                                  genes= !rowData(sce_all)$is_mitochondrial),
                                  percent_top=c(50, 100, 200, 500))

sce_all$log_sum &lt;- log10(sce_all$sum)
sce_all$log_total &lt;- log10(sce_all$total)</code></pre>
</div>
<div id="first-qc-charts" class="section level2">
<h2>First QC charts</h2>
<pre class="r"><code>plot_qc_charts &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n ### &quot;, sce_name, &quot; \n\n&quot;))

  # cat(&quot;\n### Number of genes \n\n&quot;)
  plot1 &lt;- ggplot(as.data.frame(colData(sce)), aes(x=Sample, y=log10(detected))) +
    geom_violin(scale=&quot;count&quot;) +
    labs(title=&quot;Number of detected genes&quot;,
         subtitle=sce_name,
         x=&quot;&quot;,
         y=&quot;Number of detected genes (log10)&quot;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))

  # cat(&quot;\n### Number of genes \n\n&quot;)
  plot2 &lt;- ggplot(as.data.frame(colData(sce)), aes(x=Sample, y=log_total)) +
    geom_violin(scale=&quot;count&quot;) +
    labs(title=&quot;Total gene expression counts&quot;,
         subtitle=sce_name,
         x=&quot;&quot;,
         y=&quot;Total gene expression counts (log10)&quot;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))

  # plot3 &lt;- ggplot(as.data.frame(qcs), aes(x=Sample, y=subsets_Mt_percent)) +
  plot3 &lt;- ggplot(as.data.frame(colData(sce)), aes(x=Sample, y=subsets_Mt_percent)) +
    geom_violin(scale=&quot;count&quot;) +
    labs(title=&quot;Share of mitochondrial genes&quot;,
         subtitle=sce_name,
         x=&quot;&quot;,
         y=&quot;Mitochodrial share (percent)&quot;) +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5))

  gridExtra::grid.arrange(plot1, plot2, plot3, nrow=1)
}

plot_qc_charts(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<p>### Fresh samples only</p>
<p><img src="figure/P2_01_data_comparison.Rmd/First%20QC%20charts-1.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_qc_charts(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
<p>### Fresh and cultured samples</p>
<p><img src="figure/P2_01_data_comparison.Rmd/First%20QC%20charts-2.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="mitochondria-qc-charts" class="section level2 tabset">
<h2 class="tabset">Mitochondria QC charts</h2>
<pre class="r"><code>plot_mitochondrial_qc_charts &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n ### &quot;, sce_name, &quot; \n\n&quot;))
  
  mito1 &lt;- ggplot(as.data.frame(colData(sce)),
         aes(x=log_total, y=subsets_genes_detected, color=subsets_Mt_percent)) +
    geom_point(size=0.25) +
    labs(title=&quot;Library size vs. gene count&quot;,
         subtitle=&quot;percentual share of mitochondrial genes in blue&quot;,
         x=&quot;Sum of expression values by cell&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    geom_density_2d() +
    scale_y_log10()

  mito2 &lt;- ggplot(as.data.frame(colData(sce)),
         aes(x=subsets_Mt_percent,
             y=subsets_genes_detected,
             color=log_sum)) +
    geom_point(size=0.25) +
    labs(title=&quot;Mito. content vs. genes detected&quot;,
         subtitle=&quot;coloured by log library size&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d()

  gridExtra::grid.arrange(mito1, mito2, nrow=1)
  
}

plot_mitochondrial_qc_charts(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<p>### Fresh samples only</p>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%202-1.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_mitochondrial_qc_charts(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
<p>### Fresh and cultured samples</p>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%202-2.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="library-size-vs.-mito-share-by-sample" class="section level2">
<h2>Library size vs. mito share by sample</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n ### Fresh samples only \n\n&quot;))</code></pre>
<pre><code>

 ### Fresh samples only </code></pre>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
       aes(x=total, y=subsets_genes_detected, color=subsets_Mt_percent)) +
  geom_point(size=0.25) +
  labs(title=&quot;Library size vs. gene count faceted by sample&quot;,
       subtitle=&quot;percentual share of mitochondrial genes in blue&quot;,
       x=&quot;Sum of expression values by cell&quot;,
       y=&quot;# of non-mitochondrial genes detected&quot;) +
  scale_x_log10() +
  geom_density_2d() +
  scale_y_log10() +
  facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/lib%20size%20vs.%20mito-1.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(paste0(&quot;\n\n ### Fresh and cultured samples \n\n&quot;))</code></pre>
<pre><code>

 ### Fresh and cultured samples </code></pre>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
       aes(x=total, y=subsets_genes_detected, color=subsets_Mt_percent)) +
  geom_point(size=0.25) +
  labs(title=&quot;Library size vs. gene count faceted by sample&quot;,
       subtitle=&quot;percentual share of mitochondrial genes in blue&quot;,
       x=&quot;Sum of expression values by cell&quot;,
       y=&quot;# of non-mitochondrial genes detected&quot;) +
  scale_x_log10() +
  geom_density_2d() +
  scale_y_log10() +
  facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/lib%20size%20vs.%20mito-2.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="mito.-percent-vs.-genes-by-sample" class="section level2">
<h2>Mito. percent vs. genes by sample</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n ### Fresh samples only \n\n&quot;))</code></pre>
<pre><code>

 ### Fresh samples only </code></pre>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
         aes(x=subsets_Mt_percent,
             y=subsets_genes_detected,
             color=log_sum)) +
    geom_point(size=0.25) +
    labs(title=&quot;Mito. content vs. genes detected faceted by sample&quot;,
         subtitle=&quot;Coloured by log library size&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d() +
    facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/unnamed-chunk-1-1.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(paste0(&quot;\n\n ### Fresh and cultured samples \n\n&quot;))</code></pre>
<pre><code>

 ### Fresh and cultured samples </code></pre>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
         aes(x=subsets_Mt_percent,
             y=subsets_genes_detected,
             color=log_sum)) +
    geom_point(size=0.25) +
    labs(title=&quot;Mito. content vs. genes detected faceted by sample&quot;,
         subtitle=&quot;Coloured by log library size&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;# of non-mitochondrial genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d() +
    facet_wrap(~Sample)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/unnamed-chunk-1-2.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="filtering" class="section level2">
<h2>Filtering</h2>
<p>Cells are marked to be dropped if:</p>
<ul>
<li>Sum of all expression values in a given cell is lower than 2 median absolute deviations from the sample mean</li>
<li>Number of genes of a given cell is lower than 2 median absolute deviations from the average number of genes in the given sample</li>
<li>The percentage of mitochondrial gene expression values is higher than 2 median absolute deviations from the sample mean</li>
<li>More than 25% percent of a genes expression values are from mitochondrial genes</li>
<li>The sum of all expression values is less than 1500</li>
<li>If they express less than 500 genes</li>
</ul>
<pre class="r"><code>identify_lowq_cells &lt;- function(sce){
  
  # Relative criteria for filtering
  sce$total_expression_counts_drop &lt;- isOutlier(sce$sum, nmads=2, type=&quot;lower&quot;, batch=sce$Sample, log=TRUE)

  sce$genes_detected_drop &lt;- isOutlier(sce$detected, nmads=2, type=&quot;lower&quot;, batch=sce$Sample, log=TRUE)

  sce$mito_drop &lt;- isOutlier(sce$subsets_Mt_percent, nmads=2, type=&quot;higher&quot;, batch=sce$Sample)

  # Absolute thresholds for filtering

  # Drop cells with a sum expression values lower than 1500
  sce$total_expression_counts_drop_manual_thresh &lt;- sce$sum &lt; 1500
  
  # A manual threshold: we require cells to have at least 500 genes expressed
  sce$genes_detected_manual_thresh &lt;- sce$detected &lt; 500

  # Drop cells with a percentage of mitochondrial expression values above a certain value
  sce$mito_drop_manual_thresh_25 &lt;- sce$subsets_Mt_percent &gt; 25
  
  sce$to_be_dropped &lt;- sce$total_expression_counts_drop |
    sce$genes_detected_drop |
    sce$mito_drop |
    sce$total_expression_counts_drop_manual_thresh | 
    sce$genes_detected_manual_thresh |
    sce$mito_drop_manual_thresh_25
  

 sce
}

sce_fresh &lt;- identify_lowq_cells(sce_fresh)
sce_all &lt;- identify_lowq_cells(sce_all)</code></pre>
<div id="how-many-cells-are-captured-by-each-filter" class="section level3 tabset">
<h3 class="tabset">How many cells are captured by each filter?</h3>
<pre class="r"><code>print_filter_overview &lt;- function(sce, sce_name){
  cat(paste0(&quot;\n\n ### &quot;, sce_name, &quot; \n\n&quot;))
  
  print(t(data.frame(Drop_via_lib_size = sum(sce$total_expression_counts_drop),
           Drop_via_lib_size_manual = sum(sce$total_expression_counts_drop_manual_thresh),
           Drop_via_detected_genes = sum(sce$genes_detected_drop),
           Drop_via_manual_genes_thresh = sum(sce$genes_detected_manual_thresh),
           Drop_via_mito_percentage = sum(sce$mito_drop),
           Drop_via_mito_thresh_25 = sum(sce$mito_drop_manual_thresh_25),
           Drop_via_joint_criteria = sum(sce$total_expression_counts_drop |
           sce$genes_detected_drop |
           sce$mito_drop |
           sce$total_expression_counts_drop_manual_thresh |
           sce$mito_drop_manual_thresh_25 | 
           sce$genes_detected_manual_thresh),
           Total_cells = nrow(colData(sce)))) %&gt;%
  kable())
  # %&gt;%
  # kable_styling())
}

print_filter_overview(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<p>### Fresh samples only</p>
<table>
<tbody>
<tr>
<td style="text-align:left;">
Drop_via_lib_size
</td>
<td style="text-align:right;">
758
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_lib_size_manual
</td>
<td style="text-align:right;">
810
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_detected_genes
</td>
<td style="text-align:right;">
956
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_manual_genes_thresh
</td>
<td style="text-align:right;">
534
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_percentage
</td>
<td style="text-align:right;">
882
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_thresh_25
</td>
<td style="text-align:right;">
2384
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_joint_criteria
</td>
<td style="text-align:right;">
3032
</td>
</tr>
<tr>
<td style="text-align:left;">
Total_cells
</td>
<td style="text-align:right;">
9752
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>print_filter_overview(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
<p>### Fresh and cultured samples</p>
<table>
<tbody>
<tr>
<td style="text-align:left;">
Drop_via_lib_size
</td>
<td style="text-align:right;">
2294
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_lib_size_manual
</td>
<td style="text-align:right;">
3840
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_detected_genes
</td>
<td style="text-align:right;">
3292
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_manual_genes_thresh
</td>
<td style="text-align:right;">
2564
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_percentage
</td>
<td style="text-align:right;">
3048
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_mito_thresh_25
</td>
<td style="text-align:right;">
4418
</td>
</tr>
<tr>
<td style="text-align:left;">
Drop_via_joint_criteria
</td>
<td style="text-align:right;">
7827
</td>
</tr>
<tr>
<td style="text-align:left;">
Total_cells
</td>
<td style="text-align:right;">
27715
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="how-many-cells-are-affected-by-each-filter" class="section level2 tabset">
<h2 class="tabset">How many cells are affected by each filter</h2>
<pre class="r"><code>show_panel_overview_filters &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n ### &quot;, sce_name, &quot; {.tabset }\n\n&quot;))
  
  ### Absolute numbers ### 
  
  absolute_count &lt;- rbind(
    table(sce$Sample),
    table(sce$total_expression_counts_drop, sce$Sample)[2,],
    table(sce$total_expression_counts_drop_manual_thresh, sce$Sample)[2,],
    table(sce$genes_detected_drop, sce$Sample)[2,],
    table(sce$genes_detected_drop, sce$Sample)[2,],
    table(sce$mito_drop, sce$Sample)[2,],
    table(sce$mito_drop_manual_thresh_25, sce$Sample)[2,],
    table(sce$to_be_dropped, sce$Sample)[2,],
    table(sce$to_be_dropped, sce$Sample)[1,]) %&gt;%
    as.data.frame()

  rownames(absolute_count) &lt;- c(&quot;Number of cells&quot;,
               &quot;Library size relative filter&quot;,
               &quot;Library size manual filter&quot;,
               &quot;Number of genes relative filter&quot;,
               &quot;Number of genes manual filter&quot;,
               &quot;Mito. percentage relative threshold&quot;,
               &quot;Mito. percentage manual threshold&quot;,
               &quot;Total dropped&quot;,
               &quot;Cells remaining&quot;)
  
  absolute_count[, &quot;Total&quot;] &lt;- rowSums(absolute_count)
  
  cat(&quot;\n\n#### Table (Absolute numbers) \n\n&quot;)
  print(absolute_count %&gt;% 
    kable() %&gt;%
    kable_styling())
  
  cat(&quot;\n\n#### Panel (Absolute numbers) \n\n&quot;)
  print(absolute_count[2:7,-dim(absolute_count)[2]] %&gt;%
    tibble::rownames_to_column() %&gt;%
    tidyr::gather(colname, value, -rowname) %&gt;%
    ggplot(aes(x = colname, y = rowname, fill = value)) +
    geom_tile() +
    geom_text(aes(label=value), color=&quot;white&quot;) +
    scale_x_discrete(position=&quot;top&quot;) +
    labs(title=&quot;Cells filtered out by respective filter&quot;,
         subtitle=&quot;Number of cells&quot;,
         x=&quot;&quot;,
         y=&quot;&quot;) +
    guides(fill=&quot;none&quot;))
  
  
  ### Relative numbers ### 
  
  # divide by number of cells to give a relative count
  relative_count &lt;- mapply(&#39;/&#39;, absolute_count, absolute_count[1,]) %&gt;%
    round(3) %&gt;%
    as.data.frame()
  
  rownames(relative_count) &lt;- c(&quot;Number of cells&quot;,
                                         &quot;Library size relative filter&quot;,
                                         &quot;Library size manual filter&quot;,
                                         &quot;Number of genes relative filter&quot;,
                                         &quot;Number of genes manual filter&quot;,
                                         &quot;Mito. percentage relative threshold&quot;,
                                         &quot;Mito. percentage manual threshold&quot;,
                                         &quot;Total dropped&quot;,
                                         &quot;Cells remaining&quot;)
  
  cat(&quot;\n\n#### Table (Relative numbers by sample) \n\n&quot;)
  print(relative_count %&gt;% 
          mutate_all(list(~paste0(.*100, &quot;%&quot;))) %&gt;%
    kable() %&gt;%
    kable_styling())
  
  
  cat(&quot;\n\n#### Panel (Relative numbers by sample) \n\n&quot;)
  print(relative_count[2:7,-dim(relative_count)[2]] %&gt;%
    tibble::rownames_to_column() %&gt;%
    tidyr::gather(colname, value, -rowname) %&gt;%
    ggplot(aes(x = colname, y = rowname, fill = value)) +
    geom_tile() +
    geom_text(aes(label=scales::percent(value, accuracy = 0.1)), color=&quot;white&quot;) +
    scale_x_discrete(position=&quot;top&quot;) +
    labs(title=&quot;Cells filtered out by respective filter&quot;,
         subtitle=&quot;Percentage of all cells in given sample before filtering&quot;,
         x=&quot;&quot;,
         y=&quot;&quot;) +
    guides(fill=&quot;none&quot;))
}

show_panel_overview_filters(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<p>### Fresh samples only {.tabset }</p>
<div id="table-absolute-numbers" class="section level4">
<h4>Table (Absolute numbers)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Fresh_S1
</th>
<th style="text-align:right;">
Fresh_S2
</th>
<th style="text-align:right;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
4689
</td>
<td style="text-align:right;">
5063
</td>
<td style="text-align:right;">
9752
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:right;">
292
</td>
<td style="text-align:right;">
466
</td>
<td style="text-align:right;">
758
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:right;">
662
</td>
<td style="text-align:right;">
810
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:right;">
422
</td>
<td style="text-align:right;">
534
</td>
<td style="text-align:right;">
956
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:right;">
422
</td>
<td style="text-align:right;">
534
</td>
<td style="text-align:right;">
956
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:right;">
379
</td>
<td style="text-align:right;">
503
</td>
<td style="text-align:right;">
882
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:right;">
855
</td>
<td style="text-align:right;">
1529
</td>
<td style="text-align:right;">
2384
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:right;">
1111
</td>
<td style="text-align:right;">
1921
</td>
<td style="text-align:right;">
3032
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:right;">
3578
</td>
<td style="text-align:right;">
3142
</td>
<td style="text-align:right;">
6720
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-absolute-numbers" class="section level4">
<h4>Panel (Absolute numbers)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="table-relative-numbers-by-sample" class="section level4">
<h4>Table (Relative numbers by sample)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Fresh_S1
</th>
<th style="text-align:left;">
Fresh_S2
</th>
<th style="text-align:left;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:left;">
6.2%
</td>
<td style="text-align:left;">
9.2%
</td>
<td style="text-align:left;">
7.8%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:left;">
3.2%
</td>
<td style="text-align:left;">
13.1%
</td>
<td style="text-align:left;">
8.3%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:left;">
9%
</td>
<td style="text-align:left;">
10.5%
</td>
<td style="text-align:left;">
9.8%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:left;">
9%
</td>
<td style="text-align:left;">
10.5%
</td>
<td style="text-align:left;">
9.8%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:left;">
8.1%
</td>
<td style="text-align:left;">
9.9%
</td>
<td style="text-align:left;">
9%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:left;">
18.2%
</td>
<td style="text-align:left;">
30.2%
</td>
<td style="text-align:left;">
24.4%
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:left;">
23.7%
</td>
<td style="text-align:left;">
37.9%
</td>
<td style="text-align:left;">
31.1%
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:left;">
76.3%
</td>
<td style="text-align:left;">
62.1%
</td>
<td style="text-align:left;">
68.9%
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-relative-numbers-by-sample" class="section level4">
<h4>Panel (Relative numbers by sample)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-2.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>show_panel_overview_filters(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
<p>### Fresh and cultured samples {.tabset }</p>
</div>
<div id="table-absolute-numbers-1" class="section level4">
<h4>Table (Absolute numbers)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Culture_S1
</th>
<th style="text-align:right;">
Culture_S2
</th>
<th style="text-align:right;">
Culture_S4
</th>
<th style="text-align:right;">
Culture_S5
</th>
<th style="text-align:right;">
Fresh_S1
</th>
<th style="text-align:right;">
Fresh_S2
</th>
<th style="text-align:right;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:right;">
989
</td>
<td style="text-align:right;">
4807
</td>
<td style="text-align:right;">
9181
</td>
<td style="text-align:right;">
2986
</td>
<td style="text-align:right;">
4689
</td>
<td style="text-align:right;">
5063
</td>
<td style="text-align:right;">
27715
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
496
</td>
<td style="text-align:right;">
806
</td>
<td style="text-align:right;">
233
</td>
<td style="text-align:right;">
292
</td>
<td style="text-align:right;">
467
</td>
<td style="text-align:right;">
2294
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:right;">
465
</td>
<td style="text-align:right;">
640
</td>
<td style="text-align:right;">
1675
</td>
<td style="text-align:right;">
250
</td>
<td style="text-align:right;">
148
</td>
<td style="text-align:right;">
662
</td>
<td style="text-align:right;">
3840
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
775
</td>
<td style="text-align:right;">
1184
</td>
<td style="text-align:right;">
376
</td>
<td style="text-align:right;">
423
</td>
<td style="text-align:right;">
534
</td>
<td style="text-align:right;">
3292
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
775
</td>
<td style="text-align:right;">
1184
</td>
<td style="text-align:right;">
376
</td>
<td style="text-align:right;">
423
</td>
<td style="text-align:right;">
534
</td>
<td style="text-align:right;">
3292
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
622
</td>
<td style="text-align:right;">
1074
</td>
<td style="text-align:right;">
412
</td>
<td style="text-align:right;">
379
</td>
<td style="text-align:right;">
503
</td>
<td style="text-align:right;">
3048
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:right;">
266
</td>
<td style="text-align:right;">
346
</td>
<td style="text-align:right;">
1071
</td>
<td style="text-align:right;">
352
</td>
<td style="text-align:right;">
854
</td>
<td style="text-align:right;">
1529
</td>
<td style="text-align:right;">
4418
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:right;">
562
</td>
<td style="text-align:right;">
1170
</td>
<td style="text-align:right;">
2411
</td>
<td style="text-align:right;">
653
</td>
<td style="text-align:right;">
1110
</td>
<td style="text-align:right;">
1921
</td>
<td style="text-align:right;">
7827
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:right;">
427
</td>
<td style="text-align:right;">
3637
</td>
<td style="text-align:right;">
6770
</td>
<td style="text-align:right;">
2333
</td>
<td style="text-align:right;">
3579
</td>
<td style="text-align:right;">
3142
</td>
<td style="text-align:right;">
19888
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-absolute-numbers-1" class="section level4">
<h4>Panel (Absolute numbers)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-3.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="table-relative-numbers-by-sample-1" class="section level4">
<h4>Table (Relative numbers by sample)</h4>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Culture_S1
</th>
<th style="text-align:left;">
Culture_S2
</th>
<th style="text-align:left;">
Culture_S4
</th>
<th style="text-align:left;">
Culture_S5
</th>
<th style="text-align:left;">
Fresh_S1
</th>
<th style="text-align:left;">
Fresh_S2
</th>
<th style="text-align:left;">
Total
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Number of cells
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
<td style="text-align:left;">
100%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size relative filter
</td>
<td style="text-align:left;">
0%
</td>
<td style="text-align:left;">
10.3%
</td>
<td style="text-align:left;">
8.8%
</td>
<td style="text-align:left;">
7.8%
</td>
<td style="text-align:left;">
6.2%
</td>
<td style="text-align:left;">
9.2%
</td>
<td style="text-align:left;">
8.3%
</td>
</tr>
<tr>
<td style="text-align:left;">
Library size manual filter
</td>
<td style="text-align:left;">
47%
</td>
<td style="text-align:left;">
13.3%
</td>
<td style="text-align:left;">
18.2%
</td>
<td style="text-align:left;">
8.4%
</td>
<td style="text-align:left;">
3.2%
</td>
<td style="text-align:left;">
13.1%
</td>
<td style="text-align:left;">
13.9%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes relative filter
</td>
<td style="text-align:left;">
0%
</td>
<td style="text-align:left;">
16.1%
</td>
<td style="text-align:left;">
12.9%
</td>
<td style="text-align:left;">
12.6%
</td>
<td style="text-align:left;">
9%
</td>
<td style="text-align:left;">
10.5%
</td>
<td style="text-align:left;">
11.9%
</td>
</tr>
<tr>
<td style="text-align:left;">
Number of genes manual filter
</td>
<td style="text-align:left;">
0%
</td>
<td style="text-align:left;">
16.1%
</td>
<td style="text-align:left;">
12.9%
</td>
<td style="text-align:left;">
12.6%
</td>
<td style="text-align:left;">
9%
</td>
<td style="text-align:left;">
10.5%
</td>
<td style="text-align:left;">
11.9%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage relative threshold
</td>
<td style="text-align:left;">
5.9%
</td>
<td style="text-align:left;">
12.9%
</td>
<td style="text-align:left;">
11.7%
</td>
<td style="text-align:left;">
13.8%
</td>
<td style="text-align:left;">
8.1%
</td>
<td style="text-align:left;">
9.9%
</td>
<td style="text-align:left;">
11%
</td>
</tr>
<tr>
<td style="text-align:left;">
Mito. percentage manual threshold
</td>
<td style="text-align:left;">
26.9%
</td>
<td style="text-align:left;">
7.2%
</td>
<td style="text-align:left;">
11.7%
</td>
<td style="text-align:left;">
11.8%
</td>
<td style="text-align:left;">
18.2%
</td>
<td style="text-align:left;">
30.2%
</td>
<td style="text-align:left;">
15.9%
</td>
</tr>
<tr>
<td style="text-align:left;">
Total dropped
</td>
<td style="text-align:left;">
56.8%
</td>
<td style="text-align:left;">
24.3%
</td>
<td style="text-align:left;">
26.3%
</td>
<td style="text-align:left;">
21.9%
</td>
<td style="text-align:left;">
23.7%
</td>
<td style="text-align:left;">
37.9%
</td>
<td style="text-align:left;">
28.2%
</td>
</tr>
<tr>
<td style="text-align:left;">
Cells remaining
</td>
<td style="text-align:left;">
43.2%
</td>
<td style="text-align:left;">
75.7%
</td>
<td style="text-align:left;">
73.7%
</td>
<td style="text-align:left;">
78.1%
</td>
<td style="text-align:left;">
76.3%
</td>
<td style="text-align:left;">
62.1%
</td>
<td style="text-align:left;">
71.8%
</td>
</tr>
</tbody>
</table>
</div>
<div id="panel-relative-numbers-by-sample-1" class="section level4">
<h4>Panel (Relative numbers by sample)</h4>
<p><img src="figure/P2_01_data_comparison.Rmd/Overview%20of%20cells%20being%20dropped-4.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plots-cells-that-will-be-dropped" class="section level2 tabset">
<h2 class="tabset">Plots cells that will be dropped</h2>
<pre class="r"><code>show_violin_plots_of_dropped &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; {.tabset}\n\n&quot;))
  
  # for the right colouring we order the levels manually
  sce$to_be_dropped &lt;- factor(sce$to_be_dropped, levels=c(FALSE, TRUE))

  plot1 &lt;- plotColData(sce,
                       x=&quot;Sample&quot;,
                       y=&quot;log_total&quot;,
                       colour_by=&quot;to_be_dropped&quot;,
                       point_size=0.5) +
    labs(title=&quot;Library size&quot;,
         subtitle=&quot;Dropped cells in orange&quot;,
         x=&quot;&quot;,
         y=&quot;Library size (log-scale)&quot;) +
    guides(colour=&quot;none&quot;) +
    theme_ipsum_rc()
  
  plot2 &lt;- plotColData(sce,
                       x=&quot;Sample&quot;,
                       y=&quot;detected&quot;,
                       colour_by=&quot;to_be_dropped&quot;,
                       point_size=0.5) +
    labs(title=&quot;Genes detected&quot;,
         subtitle=&quot;Dropped cells in orange&quot;,
         x=&quot;&quot;,
         y=&quot;Genes detected&quot;) +
    guides(colour=&quot;none&quot;) +
    theme_ipsum_rc()
  
  plot3 &lt;- plotColData(sce,
                       x=&quot;Sample&quot;,
                       y=&quot;subsets_Mt_percent&quot;,
                       colour_by=&quot;to_be_dropped&quot;,
                       point_size=0.5) +
    labs(title=&quot;Mitochondrial counts as % library size&quot;,
         x=&quot;&quot;,
         subtitle=&quot;Dropped cells in orange&quot;,
         y=&quot;Mito. gene expression values as % of total&quot;) +
    guides(colour=&quot;none&quot;) +
    theme_ipsum_rc()

  print(gridExtra::grid.arrange(plot1, plot2, plot3, nrow=1))
}

show_violin_plots_of_dropped(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/Violin%20plot%20of%20cells%20to%20be%20dropped-1.png" width="2496" style="display: block; margin: auto;" />TableGrob (1 x 3) “arrange”: 3 grobs z cells name grob 1 1 (1-1,1-1) arrange gtable[layout] 2 2 (1-1,2-2) arrange gtable[layout] 3 3 (1-1,3-3) arrange gtable[layout]</p>
<pre class="r"><code>show_violin_plots_of_dropped(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/Violin%20plot%20of%20cells%20to%20be%20dropped-2.png" width="2496" style="display: block; margin: auto;" />TableGrob (1 x 3) “arrange”: 3 grobs z cells name grob 1 1 (1-1,1-1) arrange gtable[layout] 2 2 (1-1,2-2) arrange gtable[layout] 3 3 (1-1,3-3) arrange gtable[layout]</p>
</div>
</div>
<div id="mitochondria-plots-after-qc-filtering" class="section level2 tabset">
<h2 class="tabset">Mitochondria plots after QC filtering</h2>
<pre class="r"><code>show_mito_plots &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; {.tabset}\n\n&quot;))
  
  # for the right colouring we order the levels manually
  sce$to_be_dropped &lt;- factor(sce$to_be_dropped, levels=c(TRUE, FALSE))

  mito1 &lt;- ggplot(as.data.frame(colData(sce)), 
                  aes(x=log_total,
                      y=subsets_genes_detected,
                      color=to_be_dropped)) +
    geom_point(size=0.5) +
    labs(title=&quot;Library size vs. genes detected&quot;,
         subtitle=&quot;Cells dropped in red&quot;,
         x=&quot;Sum of expression values by cell&quot;,
         y=&quot;# of non-mitochondrial genes&quot;) +
    geom_density_2d(colour=&quot;black&quot;, alpha=0.6) +
    get_boolean_colour_scale() +
    scale_y_log10() +
    guides(colour=&quot;none&quot;)


  mito2 &lt;- ggplot(as.data.frame(colData(sce)),
                  aes(x=subsets_Mt_percent,
                      y=subsets_genes_detected,
                      color=to_be_dropped)) +
    geom_point(size=0.5) +
    labs(title=&quot;Mito. content v. genes detected&quot;,
         subtitle=&quot;Cells dropped in red&quot;,
         x=&quot;Percentage of mito. content&quot;,
         y=&quot;Genes detected&quot;) +
    scale_y_log10(limits=c(100,NA)) +
    geom_density_2d(colour=&quot;black&quot;, alpha=0.6) +
    get_boolean_colour_scale() +
    guides(colour=&quot;none&quot;)

  gridExtra::grid.arrange(mito1, mito2, nrow=1)
}

show_mito_plots(sce_fresh, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-1" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%20after%20QC-1.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code>show_mito_plots(sce_all, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-1" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/plots%20for%20QC:%20Mitochondrial%20content%20after%20QC-2.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="gene-and-expression-counts-stratified-by-sample" class="section level1 tabset">
<h1 class="tabset">Gene and expression counts stratified by sample</h1>
<div id="coloured-by-to_be_dropped" class="section level2">
<h2>Coloured by to_be_dropped</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-2" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=to_be_dropped)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Cells dropped in red&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  get_boolean_colour_scale() +
  guides(colour=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20samples-1.png" width="1728" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh and cultured samples\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-2" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=to_be_dropped)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Cells dropped in red&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  get_boolean_colour_scale() +
  guides(colour=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20samples-2.png" width="1728" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="coloured-by-mitochondrial-content" class="section level2">
<h2>Coloured by mitochondrial content</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-3" class="section level3">
<h3>Fresh samples only</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_fresh)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=subsets_Mt_percent)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Coloured by fraction of mitochondrial counts&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  scale_colour_viridis_c(option = &quot;inferno&quot;, limits=c(0,50), oob=scales::squish)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20coloured%20by%20mt%20percent-1.png" width="1728" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(paste0(&quot;\n\n### Fresh and cultured samples\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-3" class="section level3">
<h3>Fresh and cultured samples</h3>
<pre class="r"><code>ggplot(as.data.frame(colData(sce_all)),
       aes(x=log_total,
           y=subsets_genes_detected,
           color=subsets_Mt_percent)) +
  geom_point(size=0.1) +
  labs(title=&quot;Library size vs. genes detected&quot;,
       subtitle=&quot;Coloured by fraction of mitochondrial counts&quot;,
       x=&quot;Library size (log10)&quot;,
       y=&quot;# of non-mitochondrial genes&quot;) +
  scale_y_log10() +
  facet_wrap(~Sample) +
  scale_colour_viridis_c(option = &quot;inferno&quot;, limits=c(0,50), oob=scales::squish)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/stratified%20coloured%20by%20mt%20percent-2.png" width="1728" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="library-size-distributions-before-and-after-deleting-cells-marked-in-qc" class="section level1 tabset">
<h1 class="tabset">Library size distributions before and after deleting cells marked in QC</h1>
<div id="by-sample" class="section level2">
<h2>By sample</h2>
<pre class="r"><code># Fresh only before QC
cat(paste0(&quot;\n\n### Fresh samples only - before QC \n\n&quot;))</code></pre>
<div id="fresh-samples-only---before-qc" class="section level3">
<h3>Fresh samples only - before QC</h3>
<pre class="r"><code>sce_fresh %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_fresh)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_fresh)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-1.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fresh only after QC
cat(paste0(&quot;\n\n### Fresh samples only - after QC \n\n&quot;))</code></pre>
</div>
<div id="fresh-samples-only---after-qc" class="section level3">
<h3>Fresh samples only - after QC</h3>
<pre class="r"><code>sce_fresh[, !sce_fresh$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_fresh)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_fresh)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-2.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fresh and culture before QC
cat(paste0(&quot;\n\n### Fresh and cultured samples - before QC\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples---before-qc" class="section level3">
<h3>Fresh and cultured samples - before QC</h3>
<pre class="r"><code>sce_all %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_all)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_all)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-3.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fresh and culture after QC
cat(paste0(&quot;\n\n### Fresh and cultured samples - after QC\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples---after-qc" class="section level3">
<h3>Fresh and cultured samples - after QC</h3>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(log10(total), fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_all)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_all)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20sample-4.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="by-protocol-for-fresh-and-cultured-samples" class="section level2">
<h2>By protocol for fresh and cultured samples</h2>
<pre class="r"><code>cat(&quot;\n\n### Before QC\n\n&quot;)</code></pre>
<div id="before-qc" class="section level3">
<h3>Before QC</h3>
<pre class="r"><code>sce_all %&gt;% 
  ggplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  geom_density(aes(log10(total), fill=Protocol, color=Protocol), alpha=0.3) +
  get_protocol_colours(levels(colData(sce_all)$Protocol), type=&quot;fill&quot;) +
  get_protocol_colours(levels(colData(sce_all)$Protocol))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20protocol-1.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cat(&quot;\n\n### After QC\n\n&quot;)</code></pre>
</div>
<div id="after-qc" class="section level3">
<h3>After QC</h3>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of cell library sizes&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  geom_density(aes(log10(total), fill=Protocol, color=Protocol), alpha=0.3) +
  get_protocol_colours(levels(colData(sce_all)$Protocol), type=&quot;fill&quot;) +
  get_protocol_colours(levels(colData(sce_all)$Protocol))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/log10%20counts%20by%20protocol-2.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="gene-count-distributions-after-deleting-cells-marked-in-qc" class="section level1 tabset">
<h1 class="tabset">Gene count distributions after deleting cells marked in QC</h1>
<div id="by-sample-1" class="section level2">
<h2>By sample</h2>
<pre class="r"><code># Fresh only after QC
cat(paste0(&quot;\n\n### Fresh samples only - after QC \n\n&quot;))</code></pre>
<div id="fresh-samples-only---after-qc-1" class="section level3">
<h3>Fresh samples only - after QC</h3>
<pre class="r"><code>sce_fresh[, !sce_fresh$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(detected, fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of gene counts&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_fresh)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_fresh)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/gene%20counts%20by%20sample-1.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fresh and culture after QC
cat(paste0(&quot;\n\n### Fresh and cultured samples - after QC\n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples---after-qc-1" class="section level3">
<h3>Fresh and cultured samples - after QC</h3>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  geom_density(aes(detected, fill=Sample, color=Sample),alpha=0.1) +
  facet_grid(Protocol~.) +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of gene counts&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  theme(panel.spacing = unit(2, &quot;mm&quot;)) +
  get_sample_colours(levels(colData(sce_all)$Sample), type=&quot;fill&quot;) +
  get_sample_colours(levels(colData(sce_all)$Sample))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/gene%20counts%20by%20sample-2.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="by-protocol-for-fresh-and-cultured-samples-1" class="section level2">
<h2>By protocol for fresh and cultured samples</h2>
<pre class="r"><code>cat(&quot;\n\n### After QC\n\n&quot;)</code></pre>
<div id="after-qc-1" class="section level3">
<h3>After QC</h3>
<pre class="r"><code>sce_all[, !sce_all$to_be_dropped] %&gt;% 
  ggplot() +
  theme_ipsum_rc() +
  labs(title=&quot;Distribution of gene counts&quot;,
       x=&quot;Log total expression&quot;, y=&quot;Density&quot;) +
  geom_density(aes(detected, fill=Protocol, color=Protocol), alpha=0.3) +
  get_protocol_colours(levels(colData(sce_all)$Protocol), type=&quot;fill&quot;) +
  get_protocol_colours(levels(colData(sce_all)$Protocol))</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/geme%20counts%20by%20protocol-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="section" class="section level1 unnumbered">
<h1 class="unnumbered"></h1>
<div id="number-of-cells-dropped-and-kept-per-sample" class="section level2">
<h2>Number of cells dropped and kept per sample</h2>
<pre class="r"><code>### Fresh only ###
cat(paste0(&quot;\n\n### Fresh samples only {.tabset}\n\n&quot;))</code></pre>
<div id="fresh-samples-only-4" class="section level3 tabset">
<h3 class="tabset">Fresh samples only</h3>
<pre class="r"><code># Absolute values
cat(paste0(&quot;\n\n#### Absolute \n\n&quot;))</code></pre>
<div id="absolute" class="section level4">
<h4>Absolute</h4>
<pre class="r"><code>table(sce_fresh$to_be_dropped, sce_fresh$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;stack&quot;) +
  labs(title=&quot;Absolute number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Relative values
cat(paste0(&quot;\n\n#### Relative \n\n&quot;))</code></pre>
</div>
<div id="relative" class="section level4">
<h4>Relative</h4>
<pre class="r"><code>table(sce_fresh$to_be_dropped, sce_fresh$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;fill&quot;) +
  labs(title=&quot;Relative number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Fresh v culture ###
cat(paste0(&quot;\n\n### Fresh and cultured samples {.tabset}\n\n&quot;))</code></pre>
</div>
</div>
<div id="fresh-and-cultured-samples-4" class="section level3 tabset">
<h3 class="tabset">Fresh and cultured samples</h3>
<pre class="r"><code># Absolute values
cat(paste0(&quot;\n\n#### Absolute \n\n&quot;))</code></pre>
<div id="absolute-1" class="section level4">
<h4>Absolute</h4>
<pre class="r"><code>table(sce_all$to_be_dropped, sce_all$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;stack&quot;) +
  labs(title=&quot;Absolute number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-3.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Relative values
cat(paste0(&quot;\n\n#### Relative \n\n&quot;))</code></pre>
</div>
<div id="relative-1" class="section level4">
<h4>Relative</h4>
<pre class="r"><code>table(sce_all$to_be_dropped, sce_all$Sample) %&gt;%
  as.data.frame() %&gt;%
  rename(Dropped=Var1) %&gt;%
  mutate(Dropped=factor(Dropped, levels=c(TRUE, FALSE))) %&gt;%
  ggplot(aes(x=Var2, y=Freq, fill=Dropped)) +
  geom_bar(stat=&quot;identity&quot;, position=&quot;fill&quot;) +
  labs(title=&quot;Relative number of cells dropped by sample&quot;,
       subtitle=&quot;Cells dropped coloured in red&quot;,
       x=&quot;&quot;, y=&quot;Number of cells&quot;) +
  get_boolean_colour_scale(type=&quot;fill&quot;) +
  guides(fill=&quot;none&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/Barplot%20of%20cells%20to%20be%20dropped-4.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="finally-drop-cells-from-dataset" class="section level2">
<h2>Finally drop cells from dataset</h2>
<pre class="r fold-show"><code>sce_fresh_filtered &lt;- sce_fresh[, !sce_fresh$to_be_dropped]
sce_all_filtered &lt;- sce_all[, !sce_all$to_be_dropped]</code></pre>
</div>
</div>
<div id="overview-of-doublets-dropped-comparison" class="section level1">
<h1>Overview of doublets dropped comparison</h1>
<pre class="r"><code># identify doublets
# seed_to_use &lt;- readRDS(&quot;../code/seeds/seed8_fresh.rds&quot;)
# set.seed(seed_to_use)
set.seed(100)
sce_fresh_filtered &lt;- scDblFinder(sce_fresh_filtered, samples=&quot;Sample&quot;)

# seed_to_use &lt;- readRDS(&quot;../code/seeds/seed9_all.rds&quot;)
# set.seed(seed_to_use)
set.seed(100)
sce_all_filtered &lt;- scDblFinder(sce_all_filtered, samples=&quot;Sample&quot;)

# plot the number of doublets per sample

# Fresh
cat(paste0(&quot;\n\n## Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-5" class="section level2">
<h2>Fresh samples only</h2>
<pre class="r"><code>table(sce_fresh_filtered$scDblFinder.class, sce_fresh_filtered$Sample) %&gt;%
  as.data.frame() %&gt;%
  dplyr::rename(&quot;Type&quot;=&quot;Var1&quot;, &quot;Sample&quot;=&quot;Var2&quot;) %&gt;%
  ggplot(aes(x=Sample, y=Freq, fill = Type, label=Freq)) +
  geom_bar(stat=&quot;identity&quot;,
           position = position_stack(reverse = TRUE)) +
  labs(title = &quot;Doublets identified&quot;,
       x=&quot;&quot;,
       y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/remove%20doublets-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Fresh and culture
# plot the number of doublets per sample
cat(paste0(&quot;\n\n## Fresh and cultured samples \n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-5" class="section level2">
<h2>Fresh and cultured samples</h2>
<pre class="r"><code>table(sce_all_filtered$scDblFinder.class, sce_all_filtered$Sample) %&gt;%
  as.data.frame() %&gt;%
  dplyr::rename(&quot;Type&quot;=&quot;Var1&quot;, &quot;Sample&quot;=&quot;Var2&quot;) %&gt;%
  ggplot(aes(x=Sample, y=Freq, fill = Type, label=Freq)) +
  geom_bar(stat=&quot;identity&quot;,
           position = position_stack(reverse = TRUE)) +
  labs(title = &quot;Doublets identified&quot;,
       x=&quot;&quot;,
       y = &quot;Number of Cells&quot;)</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/remove%20doublets-2.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code># finally remove droplets
sce_fresh_filtered &lt;- sce_fresh_filtered[,colData(sce_fresh_filtered)$scDblFinder.class==&quot;singlet&quot;]
sce_all_filtered &lt;- sce_all_filtered[,colData(sce_all_filtered)$scDblFinder.class==&quot;singlet&quot;]</code></pre>
</div>
</div>
<div id="normalisation" class="section level1 tabset">
<h1 class="tabset">Normalisation</h1>
<p>We scale the data by a cell-specific size factor.</p>
<p>If we focus more on DE analysis downstream at a later point of the data analysis, then it might make sense to evaluate other normalisation methods.</p>
<pre class="r"><code>cat(paste0(&quot;\n\n## Fresh samples only \n\n&quot;))</code></pre>
<div id="fresh-samples-only-6" class="section level2">
<h2>Fresh samples only</h2>
<pre class="r"><code>lib.sce_fresh &lt;- librarySizeFactors(sce_fresh_filtered)

ggplot(data.frame(&quot;x&quot;=log10(lib.sce_fresh)), aes(x)) +
  geom_histogram(color=&quot;black&quot;, fill=&quot;grey80&quot;) +
  labs(title=&quot;Distribution of normalisation factors&quot;,
       subtitle=&quot;Each normalisation factors are multiplied with the expression values of a given cell&quot;,
       x=&quot;Log10 of normalisation factor&quot;,
       y=&quot;Frequency&quot;)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/normalisation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sce_fresh_filtered &lt;- logNormCounts(sce_fresh_filtered)



cat(paste0(&quot;\n\n## Fresh and cultured samples \n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-6" class="section level2">
<h2>Fresh and cultured samples</h2>
<pre class="r"><code>lib.sce_all &lt;- librarySizeFactors(sce_all_filtered)

ggplot(data.frame(&quot;x&quot;=log10(lib.sce_all)), aes(x)) +
  geom_histogram(color=&quot;black&quot;, fill=&quot;grey80&quot;) +
  labs(title=&quot;Distribution of normalisation factors&quot;,
       subtitle=&quot;Each normalisation factors are multiplied with the expression values of a given cell&quot;,
       x=&quot;Log10 of normalisation factor&quot;,
       y=&quot;Frequency&quot;)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/P2_01_data_comparison.Rmd/normalisation-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sce_all_filtered &lt;- logNormCounts(sce_all_filtered)</code></pre>
<!-- ```{r} -->
<!-- sce <- sce_all[,sce_all$to_be_dropped==FALSE] -->
<!-- sce1 <- sce[,sce$Protocol=="Fresh"] -->
<!-- sce2 <- sce[,sce$Protocol=="Culture"] -->
<!-- mean(sce1$subsets_Mt_percent) -->
<!-- mean(sce2$subsets_Mt_percent) -->
<!-- ``` -->
</div>
</div>
<div id="feature-selection---hvgs" class="section level1 tabset">
<h1 class="tabset">Feature Selection - HVGs</h1>
<pre class="r"><code>num_hvg=2000

normalise_data_by_library_size &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n### &quot;, sce_name, &quot; \n\n&quot;))

  # model the gene variance
  dec.sce &lt;- scran::modelGeneVar(sce)
  fit.vs &lt;- metadata(dec.sce)

  # Mark all HVGs for visualisation
  hvg &lt;- getTopHVGs(dec.sce, n=num_hvg)
  label &lt;- rep(&quot;No HVG&quot;, nrow(sce))
  names(label) &lt;- names(fit.vs$mean)
  label[hvg] &lt;- &quot;Is HVG&quot;

  # construct a df to make plotting with ggplot easier
  print(data.frame(&quot;x&quot;=fit.vs$mean, &quot;y&quot;=fit.vs$var, &quot;z&quot;=fit.vs$trend(fit.vs$mean), &quot;color&quot;=label) %&gt;%
    ggplot(aes(x=x, y=y, color=color)) +
    geom_point() +
    scale_color_manual(values=c(&quot;orange&quot;,&quot;black&quot;)) +
    geom_line(aes(x, z), colour=&quot;dodgerblue&quot;, size=0.75) +
    labs(x=&quot;Mean of log-expression&quot;,
         y=&quot;Variance of log-expression&quot;,
         title=paste0(&quot;Top &quot;, num_hvg,&quot; highly variable genes&quot;)))


  list(&quot;sco&quot;=dec.sce, &quot;hvg&quot;=hvg)
}

res.fresh &lt;- normalise_data_by_library_size(sce_fresh_filtered, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-7" class="section level3">
<h3>Fresh samples only</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/feature%20selection-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dec.sce.fresh &lt;- res.fresh$sco
hvg.fresh &lt;- res.fresh$hvg

res.all &lt;- normalise_data_by_library_size(sce_all_filtered, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
<div id="fresh-and-cultured-samples-7" class="section level3">
<h3>Fresh and cultured samples</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/feature%20selection-2.png" width="1344" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dec.sce.all &lt;- res.all$sco
hvg.all &lt;- res.all$hvg</code></pre>
</div>
</div>
<div id="section-1" class="section level1">
<h1></h1>
<div id="peek-at-highly-variable-genes" class="section level2 tabset">
<h2 class="tabset">Peek at highly variable genes</h2>
<pre class="r"><code>cat(paste0(&quot;\n\n## Fresh samples only \n\n&quot;))</code></pre>
</div>
<div id="fresh-samples-only-8" class="section level2">
<h2>Fresh samples only</h2>
<pre class="r"><code>print(kable_styling(kable(dec.sce.fresh[order(dec.sce.fresh$bio, decreasing=TRUE)[1:10],], digits=2)))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
total
</th>
<th style="text-align:right;">
tech
</th>
<th style="text-align:right;">
bio
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PTGDS
</td>
<td style="text-align:right;">
1.88
</td>
<td style="text-align:right;">
8.37
</td>
<td style="text-align:right;">
1.29
</td>
<td style="text-align:right;">
7.08
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
DCN
</td>
<td style="text-align:right;">
2.30
</td>
<td style="text-align:right;">
8.26
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
6.95
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL1A1
</td>
<td style="text-align:right;">
2.09
</td>
<td style="text-align:right;">
7.56
</td>
<td style="text-align:right;">
1.32
</td>
<td style="text-align:right;">
6.24
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL1A2
</td>
<td style="text-align:right;">
2.22
</td>
<td style="text-align:right;">
7.32
</td>
<td style="text-align:right;">
1.32
</td>
<td style="text-align:right;">
6.00
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
VIM
</td>
<td style="text-align:right;">
5.26
</td>
<td style="text-align:right;">
6.20
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
5.59
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL3A1
</td>
<td style="text-align:right;">
1.87
</td>
<td style="text-align:right;">
6.66
</td>
<td style="text-align:right;">
1.29
</td>
<td style="text-align:right;">
5.37
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL6A2
</td>
<td style="text-align:right;">
2.28
</td>
<td style="text-align:right;">
6.14
</td>
<td style="text-align:right;">
1.32
</td>
<td style="text-align:right;">
4.83
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
COL6A1
</td>
<td style="text-align:right;">
2.23
</td>
<td style="text-align:right;">
6.00
</td>
<td style="text-align:right;">
1.32
</td>
<td style="text-align:right;">
4.68
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
SPARCL1
</td>
<td style="text-align:right;">
2.53
</td>
<td style="text-align:right;">
5.67
</td>
<td style="text-align:right;">
1.29
</td>
<td style="text-align:right;">
4.38
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
IGFBP7
</td>
<td style="text-align:right;">
2.98
</td>
<td style="text-align:right;">
5.55
</td>
<td style="text-align:right;">
1.19
</td>
<td style="text-align:right;">
4.37
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat(paste0(&quot;\n\n## Fresh and cultured samples \n\n&quot;))</code></pre>
</div>
<div id="fresh-and-cultured-samples-8" class="section level2">
<h2>Fresh and cultured samples</h2>
<pre class="r"><code>print(kable_styling(kable(dec.sce.all[order(dec.sce.all$bio, decreasing=TRUE)[1:10],], digits=2)))</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
mean
</th>
<th style="text-align:right;">
total
</th>
<th style="text-align:right;">
tech
</th>
<th style="text-align:right;">
bio
</th>
<th style="text-align:right;">
p.value
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
CXCL8
</td>
<td style="text-align:right;">
2.27
</td>
<td style="text-align:right;">
9.25
</td>
<td style="text-align:right;">
1.44
</td>
<td style="text-align:right;">
7.81
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
MMP1
</td>
<td style="text-align:right;">
1.88
</td>
<td style="text-align:right;">
7.74
</td>
<td style="text-align:right;">
1.35
</td>
<td style="text-align:right;">
6.39
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
MT2A
</td>
<td style="text-align:right;">
4.19
</td>
<td style="text-align:right;">
6.22
</td>
<td style="text-align:right;">
0.87
</td>
<td style="text-align:right;">
5.34
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
MMP3
</td>
<td style="text-align:right;">
1.16
</td>
<td style="text-align:right;">
5.63
</td>
<td style="text-align:right;">
1.08
</td>
<td style="text-align:right;">
4.54
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
DCN
</td>
<td style="text-align:right;">
1.72
</td>
<td style="text-align:right;">
5.61
</td>
<td style="text-align:right;">
1.31
</td>
<td style="text-align:right;">
4.30
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
CXCL3
</td>
<td style="text-align:right;">
1.62
</td>
<td style="text-align:right;">
5.38
</td>
<td style="text-align:right;">
1.28
</td>
<td style="text-align:right;">
4.10
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
IER3
</td>
<td style="text-align:right;">
3.94
</td>
<td style="text-align:right;">
5.00
</td>
<td style="text-align:right;">
0.97
</td>
<td style="text-align:right;">
4.04
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
SOD2
</td>
<td style="text-align:right;">
3.90
</td>
<td style="text-align:right;">
4.94
</td>
<td style="text-align:right;">
0.98
</td>
<td style="text-align:right;">
3.96
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
VIM
</td>
<td style="text-align:right;">
5.13
</td>
<td style="text-align:right;">
4.71
</td>
<td style="text-align:right;">
0.83
</td>
<td style="text-align:right;">
3.88
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
PTGS2
</td>
<td style="text-align:right;">
1.63
</td>
<td style="text-align:right;">
4.93
</td>
<td style="text-align:right;">
1.28
</td>
<td style="text-align:right;">
3.65
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
</tbody>
</table>
</div>
<div id="subset-selection-of-top-2000-hvgs" class="section level2">
<h2>Subset selection of top 2000 HVGs</h2>
<p>We mark the top 2000 HVGs of every dataset which we will use for dimension reduction and other steps</p>
<pre class="r"><code># fresh
rowData(sce_fresh_filtered)[,&quot;is_hvg&quot;] &lt;- F
rowData(sce_fresh_filtered)[hvg.fresh,&quot;is_hvg&quot;] &lt;- T

# fresh and culture
rowData(sce_all_filtered)[,&quot;is_hvg&quot;] &lt;- F
rowData(sce_all_filtered)[hvg.all,&quot;is_hvg&quot;] &lt;- T


# if we want to delete the other genes
# sce &lt;- sce[hvg,]</code></pre>
</div>
</div>
<div id="batch-correction" class="section level1 tabset">
<h1 class="tabset">Batch correction</h1>
<p>For batch correction we use the Harmony method. We only run batch correction for the single-cell experiments with merged samples.</p>
<pre class="r"><code>run_batch_correction &lt;- function(sce, sce_name){
  
  cat(paste0(&quot;\n\n## &quot;, sce_name, &quot; {.tabset}\n\n&quot;))

  # Before batch effects
  # run UMAP
  set.seed(100)
  sce &lt;- runPCA(sce, exprs_values=&quot;logcounts&quot;, subset_row=rowData(sce)$is_hvg)

  set.seed(100)
  sce &lt;- runUMAP(sce, dimred=&quot;PCA&quot;, n_neighbors=50, min_dist = 0.1) #, subset_row=rowData(sce)$is_hvg

  cat(&quot;\n\n### UMAP uncorrected \n\n&quot;)
  print(plotReducedDim(sce, &quot;UMAP&quot;, colour_by=&quot;Sample&quot;, point_size=0.5) +
    ggtitle(&quot;UMAP prior to batch correction&quot;) +
    guides(colour=guide_legend(override.aes=list(size=4))) +
    theme_ipsum_rc())

  # correct batch effects with Harmony
  # set.seed(200)

  meta_data &lt;- colData(sce) %&gt;%
    as.data.frame() %&gt;%
    dplyr::select(c(Protocol, Sample))
  
  set.seed(100)
  reducedDim(sce, &quot;harmony&quot;) &lt;- harmony::HarmonyMatrix(data_mat=reducedDim(sce, &quot;PCA&quot;),
                                                meta_data= meta_data,
                                                vars_use=c(&quot;Sample&quot;),
                                                do_pca = FALSE, verbose=FALSE)


  set.seed(100)
  sce &lt;- runUMAP(sce, dimred=&quot;harmony&quot;, n_neighbors=50, min_dist = 0.1) #, subset_row=rowData(sce)$is_hvg

  cat(&quot;\n\n### UMAP corrected \n\n&quot;)
  print(plotReducedDim(sce, &quot;UMAP&quot;, colour_by=&quot;Sample&quot;, point_size=0.5) +
    ggtitle(&quot;UMAP post batch correction - Harmony&quot;) +
    guides(colour=guide_legend(override.aes=list(size=4))) +
    theme_ipsum_rc())

    sce
}

sce_fresh_filtered &lt;- run_batch_correction(sce_fresh_filtered, &quot;Fresh samples only&quot;)</code></pre>
<div id="fresh-samples-only-9" class="section level2 tabset">
<h2 class="tabset">Fresh samples only</h2>
<div id="umap-uncorrected" class="section level3">
<h3>UMAP uncorrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="umap-corrected" class="section level3">
<h3>UMAP corrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-2.png" width="1536" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sce_all_filtered &lt;- run_batch_correction(sce_all_filtered, &quot;Fresh and cultured samples&quot;)</code></pre>
</div>
</div>
<div id="fresh-and-cultured-samples-9" class="section level2 tabset">
<h2 class="tabset">Fresh and cultured samples</h2>
<div id="umap-uncorrected-1" class="section level3">
<h3>UMAP uncorrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-3.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="umap-corrected-1" class="section level3">
<h3>UMAP corrected</h3>
<p><img src="figure/P2_01_data_comparison.Rmd/batch%20correction%20harmony-4.png" width="1536" style="display: block; margin: auto;" /></p>
<!-- # Clustering with k=50 -->
<!-- ```{r Clustering, results="asis", class.source="fold-show"} -->
<!-- run_clustering <- function(sce, sce_name){ -->
<!--   set.seed(100) -->
<!--   g50 <- buildSNNGraph(sce, k=50, use.dimred = 'harmony')  #, subset.row=rowData(sce)$is_hvg) # build graph -->
<!--   set.seed(100) -->
<!--   clust50 <- igraph::cluster_walktrap(g50)$membership # find groups -->
<!--   # Assign labels based on clusters -->
<!--   colLabels(sce) <- factor(clust50) -->
<!--   sce -->
<!-- } -->
<!-- sce_fresh_filtered <- run_clustering(sce_fresh_filtered, ) -->
<!-- sce_all_filtered <- run_clustering(sce_all_filtered) -->
<!-- ``` -->
<!-- ## Visualise clusters and samples {.tabset} -->
<!-- ```{r _ UMAP visualisation, results="asis", fig.width=16, fig.height=10} -->
<!-- # seed_used <- .Random.seed -->
<!-- # saveRDS(seed_used, "../code/seeds/seed1.rds") -->
<!-- # tes <- readRDS("../code/seeds/seed1.rds") -->
<!-- # fresh data -->
<!-- cat("\n\n### Fresh samples only  {.tabset}\n\n") -->
<!-- # by sample -->
<!-- cat("\n\n#### Sample\n\n") -->
<!-- plotReducedDim(sce_fresh_filtered, "UMAP", point_size=1, -->
<!--                colour_by="Sample") + -->
<!--   labs("UMAP coloured by sample") -->
<!--   guides(colour = guide_legend(override.aes = list(size=2))) + -->
<!--   theme_ipsum_rc() -->
<!-- # by label -->
<!-- cat("\n\n#### Label\n\n") -->
<!-- plotReducedDim(sce_fresh_filtered, "UMAP", point_size=1, -->
<!--                colour_by="label", -->
<!--                text_by = "label") + -->
<!--   labs("UMAP coloured by sample") -->
<!--   guides(colour = guide_legend(override.aes = list(size=2))) + -->
<!--   theme_ipsum_rc() -->
<!-- # Fresh and cultured data -->
<!-- cat("\n\n### Fresh and cultured samples {.tabset}\n\n") -->
<!-- # by sample -->
<!-- cat("\n\n#### Sample\n\n") -->
<!-- plotReducedDim(sce_all_filtered, "UMAP", point_size=1, -->
<!--                colour_by="Sample") + -->
<!--   labs("UMAP coloured by sample") -->
<!--   guides(colour = guide_legend(override.aes = list(size=2))) + -->
<!--   theme_ipsum_rc() -->
<!-- # by label -->
<!-- cat("\n\n#### Label\n\n") -->
<!-- plotReducedDim(sce_all_filtered, "UMAP", point_size=1, -->
<!--                colour_by="label", -->
<!--                text_by = "label") + -->
<!--   labs("UMAP coloured by sample") -->
<!--   guides(colour = guide_legend(override.aes = list(size=2))) + -->
<!--   theme_ipsum_rc() -->
<!-- ``` -->
<!-- ## Combined data coloured by protocol {.tabset} -->
<!-- ```{r visualise by protocol, fig.width=14, fig.height=10, results="asis"} -->
<!-- if ("All" %in% master[["annotation_name"]]){ -->
<!--   sce <- sces_filtered[[as.integer(master[master$annotation_name=="All", "num_id"])]] -->
<!--   for (dimred in c("UMAP")){  -->
<!--     cat("\n\n### ", dimred, "\n\n") -->
<!--     print(plotReducedDim(sce, dimred, point_size=1, colour_by="Protocol") + -->
<!--       labs(title=paste0(master[idx, "sample_name"], " displayed via ", dimred, " and coloured by protocol")) + -->
<!--       guides(colour = guide_legend(override.aes = list(size=2))) + -->
<!--       theme_ipsum_rc()) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ## Show cluster to sample mapping -->
<!-- The table below shows the mappings from unsupervised clustering to each individual sample. The clusters refer to the clustering performed on the dataset with all samples merged. -->
<!-- ```{r cluster to sample assignment, fig.height=6, fig.width=10} -->
<!-- if ("All" %in% master[["annotation_name"]]){ -->
<!--   sce <- sces_filtered[[as.integer(master[master$annotation_name=="All", "num_id"])]] -->
<!--   tab <- table(sce$label, sce$Sample) -->
<!--   tab_annotation <- tab -->
<!--   tab_annotation[tab_annotation==0] <- "" -->
<!--   as.ggplot(pheatmap(tab, color=brewer.pal(n=9, name="Oranges"), display_numbers=tab_annotation, silent=T)) + -->
<!--     theme_ipsum_rc() + -->
<!--     labs(title="Clusters and samples") + -->
<!--     theme(axis.line=element_blank(),axis.text.x=element_blank(), -->
<!--             axis.text.y=element_blank(),axis.ticks=element_blank(), -->
<!--             axis.title.x=element_blank(), -->
<!--             axis.title.y=element_blank(),legend.position="none", -->
<!--             panel.background=element_blank(),panel.border=element_blank(),panel.grid.major=element_blank(), -->
<!--             panel.grid.minor=element_blank(),plot.background=element_blank()) -->
<!-- } -->
<!-- ``` -->
<!-- # Marker genes {.tabset} -->
<!-- Marker genes can help us identify clusters as specific cell types and also discover smaller changes between subclusters (e.g., changes in activation or differentiation state) -->
<!-- ```{r prepare sheet to save data to} -->
<!-- wb <- createWorkbook() -->
<!-- addWorksheet(wb, "Intro") -->
<!-- writeData(wb, "Intro", data.frame("Marker genes for the protocol comparison"), colNames=FALSE) -->
<!-- ``` -->
<!-- ```{r function for top_genes} -->
<!-- plot_heatmap_of_top_genes <- function(sce, markers, top_n=5){ -->
<!--   # function assumes the sce columns: label, Sample -->
<!--   # get top genes -->
<!--   top_genes <- lapply(markers, function(marker_set){ -->
<!--     marker_set %>% -->
<!--       as.data.frame() %>% -->
<!--       filter(Top <= top_n) %>% -->
<!--       rownames() -->
<!--   }) %>% -->
<!--     unlist(use.names=F) %>% -->
<!--     unique() -->
<!--   # Matrix to plot the heatmap of expression values. -->
<!--   # To avoid a too big plot we only keep a fraction of the cells i there are too many -->
<!--   nth_cell <- max(1, round(dim(sce)[2] / 1000,0)) -->
<!--   sce_plot <- sce[top_genes, seq(1, dim(sce)[2], nth_cell)] -->
<!--   # We sort by cluster -->
<!--   sce_plot <- sce_plot[,order(sce_plot$label, sce_plot$Sample)] -->
<!--   # Anotations -->
<!--   column_annotations <- data.frame("Sample"=sce_plot$Sample, "Cluster"=factor(sce_plot$label), row.names=colnames(sce_plot)) -->
<!--   # clip heatmap values to ensure that colour contrast is high -->
<!--   sce_plot <- assay(sce_plot, "logcounts") -->
<!--   sce_plot[sce_plot >=5] = 5 -->
<!--   print(as.ggplot(pheatmap(sce_plot, annotation_col=column_annotations, -->
<!--            show_rownames=TRUE, show_colnames=FALSE, cluster_cols=FALSE, silent=TRUE, -->
<!--            color=inferno(50))) + -->
<!--     labs(title="Gene signatures per cluster identified", -->
<!--          subtitle=paste0("Top ", top_n, " differentially expressed genes per cluster; values are log-counts clipped at 5")) + -->
<!--     theme_ipsum_rc()+ -->
<!--     theme(axis.line=element_blank(),axis.text.x=element_blank(), -->
<!--             axis.text.y=element_blank(),axis.ticks=element_blank(), -->
<!--             axis.title.x=element_blank(), -->
<!--             axis.title.y=element_blank(),legend.position="none", -->
<!--             panel.background=element_blank(),panel.border=element_blank(),panel.grid.major=element_blank(), -->
<!--             panel.grid.minor=element_blank(),plot.background=element_blank())) -->
<!-- } -->
<!-- ``` -->
<!-- ```{r marker genes, fig.height=12, fig.width=6, results="asis"} -->
<!-- top_markers <- lapply(1:nrow(master), function(idx){ -->
<!--   cat(paste0("\n\n## ", master[idx, "sample_name"], " {.tabset} \n\n")) -->
<!--   sce_filtered <- sces_filtered[[idx]] -->
<!--   # we filter for only the highly variable genes. Otherwise we will see many genes which -->
<!--   # are *significantly* different (high p-values), but not marginally different (low logFCs) -->
<!--   markers.all.up <- findMarkers(sce_filtered[rowData(sce_filtered)$is_hvg,], -->
<!--                                 direction="up", -->
<!--                                 lfc=1, -->
<!--                                 block=sce_filtered$Sample) -->
<!--   for (i in 1:length(unique(names(markers.all.up)))){ -->
<!--     cat("\n\n### ", "Cluster ", i, " \n\n") -->
<!--     chosen <- i -->
<!--     interesting.all.up <- markers.all.up[[chosen]] -->
<!--     print(kable_styling(kable(interesting.all.up[1:10,], digits=2))) -->
<!--     cat("\n") -->
<!--     best.set <- interesting.all.up[1:50,] -->
<!--     logFCs <- getMarkerEffects(best.set) -->
<!--     # write to excel -->
<!--     sheet_name <- paste0(master[idx, "annotation_name"], "_ttest_pairs_cl_", i) -->
<!--     addWorksheet(wb, sheet_name) -->
<!--     writeData(wb, sheet_name, logFCs, rowNames=TRUE) -->
<!--     # set all NAs to 0 -->
<!--     cat(paste0(sum(is.na(logFCs)), " NA values in cluster ", chosen, "\n")) -->
<!--     logFCs[is.na(logFCs)] <- 0 -->
<!--     pheatmap(logFCs, breaks=seq(-5, 5, length.out=101), main=paste0("Cluster ",i, ": Marker genes, corrected for batch effects \n (upregulated only, pval.type='all')")) -->
<!--   } -->
<!--   markers.all.up -->
<!-- }) -->
<!-- ``` -->
<!-- # {-} -->
<!-- ## Averaged expression values for top markers {.tabset} -->
<!-- ```{r averaged expression values for markers, fig.width=14, fig.height=35, results="asis"} -->
<!-- for (idx in 1:nrow(master)){ -->
<!--   cat(paste0("\n\n### ", master[idx, "sample_name"], " \n\n")) -->
<!--   plot_heatmap_of_top_genes(sces_filtered[[idx]], top_markers[[idx]], top_n=5) -->
<!-- } -->
<!-- ``` -->
<!-- # Marker genes identified by entropy & f-stat {.tabset} -->
<!-- The following shows the marker genes as identified via the entropy-fstat-method for the dataset combining all samples as well as the dataset for only the fresh samples combined. -->
<!-- ```{r entropy f-stat, results="asis", fig.width=10, fig.height=7, results="asis", warning=FALSE} -->
<!-- for (annotation_name in c("All", "Fresh_all")){ -->
<!--   if (annotation_name %in% master[["annotation_name"]]){ -->
<!--     sample_name <- master[master$annotation_name==annotation_name, "sample_name"] -->
<!--     cat(paste0("\n\n## ", sample_name, " {.tabset} \n\n")) -->
<!--     sce_object <- sces_filtered[[as.integer(master[master$annotation_name==annotation_name, "num_id"])]] -->
<!--     sce_object <- sce_object[rowData(sce_object)$is_hvg,] -->
<!--     # sce_object <- sce_object[rowData(sce_object)$is_hvg==TRUE,] -->
<!--     shannon_entropy <- function(x, e = 1e-10) { -->
<!--       x <- x+e # allow zeros in x -->
<!--       p <- (x)/sum(x) -->
<!--       -sum((p)*log(p)) -->
<!--     } -->
<!--     aggr_syn_sce <- aggregateAcrossCells(sce_object, sce_object$label) -->
<!--     entropy_fstat_sce <- BiocParallel::bplapply( -->
<!--       seq_along(rownames(aggr_syn_sce)), -->
<!--       function(.x) { -->
<!--         data.frame( -->
<!--           gene = rowData(sce_object)$Symbol[.x], -->
<!--           entropy = shannon_entropy(counts(aggr_syn_sce[.x,])), -->
<!--           fstat = summary(aov(logcounts(sce_object)[.x, ] ~ colLabels(sce_object)))[[1]]$`F value`[1] -->
<!--         ) -->
<!--         }, BPPARAM = BiocParallel::MulticoreParam(workers=n_workers, RNGseed = 123)) %>% -->
<!--       purrr::reduce(rbind) -->
<!--     entropy_fstat_sce <- entropy_fstat_sce %>% -->
<!--       dplyr::mutate(entropy_norm = (entropy+1e-3)/(max(entropy)+1e-3), -->
<!--                     fstat_norm = fstat/max(fstat,na.rm = TRUE), -->
<!--                     ent_fstat_ratio = fstat_norm / entropy_norm, na.rm = TRUE) -->
<!--     topmarkers <- order(entropy_fstat_sce$ent_fstat_ratio, decreasing = TRUE)[1:50] -->
<!--     topmarker_knownmarkers <- unique(entropy_fstat_sce[topmarkers,]) -->
<!--     cat("\n\n### Entropy-fstat plot \n\n") -->
<!--     print(ggplot(entropy_fstat_sce) + -->
<!--       geom_point(aes(x=log(fstat), y=entropy), show.legend = F, colour="#02C39A") + -->
<!--       labs(x="log(F-statistic)", y="Entropy", main="Entropy vs. F-statistic of cluster separation") + -->
<!--       ggrepel::geom_text_repel(data=topmarker_knownmarkers, -->
<!--                                aes(x=log(fstat), y=entropy, label=gene, size=4), -->
<!--                                show.legend=F, -->
<!--                                max.overlaps = 30)) -->
<!--     # write to excel -->
<!--     sheet_name <- paste0(annotation_name, "_fstat_top_markers") -->
<!--     addWorksheet(wb, sheet_name) -->
<!--     writeData(wb, sheet_name, topmarker_knownmarkers, rowNames=TRUE) -->
<!--     # Expression plots -->
<!--     cat("\n\n### Expression plots for all samples combined {.tabset} \n\n") -->
<!--     genes_to_plot <-  unique(entropy_fstat_sce$gene[topmarkers]) -->
<!--     rownames(sce_object) <- rowData(sce_object)$Symbol -->
<!--     for(i in seq_along(genes_to_plot)){ -->
<!--       cat("\n\n#### ", genes_to_plot[i], "\n\n") -->
<!--       print(plotExpression(sce_object,   -->
<!--                      entropy_fstat_sce$gene[entropy_fstat_sce$gene == genes_to_plot[i]], -->
<!--                      x = "label", colour_by = "Sample", point_size=0.75)) -->
<!--     } -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ```{r save marker genes to excel} -->
<!-- # save results -->
<!-- filename_marker_results <- "../docs/output/Protocol_analysis_markers_results_hvg_genes.xlsx" -->
<!-- saveWorkbook(wb, file = filename_marker_results, overwrite = TRUE) -->
<!-- ``` -->
<!-- # Foreign Annotations {.tabset} -->
<!-- ```{r plot relative frequencies} -->
<!-- plot_frequencies <- function(sce_object){ -->
<!--   plt <- table(colData(sce_object)[,annotations_name], colData(sce_object)[,"Sample"]) %>% -->
<!--     as.data.frame() %>% -->
<!--     dplyr::rename("Cell.Type"="Var1", "Sample"="Var2") %>% -->
<!--     ggplot(aes(x=Sample, y=Freq, fill=Cell.Type)) + -->
<!--     geom_bar(stat="identity", position = "fill", width = 0.85) + -->
<!--     theme(panel.grid = element_blank(), -->
<!--           axis.text = element_text(size = 10), -->
<!--           axis.text.x = element_text(angle=30, hjust=1), -->
<!--           axis.text.y = element_text(size = 10), -->
<!--           plot.subtitle=element_text(size=12)) + -->
<!--     labs(title="Relative cell type abundances across samples", -->
<!--          subtitle=paste0("Labels: ", annotations_name)) -->
<!--   return(plt) -->
<!-- } -->
<!-- ``` -->
<!-- ```{r function for foreign annotations, echo=FALSE} -->
<!-- annotate_with_other_labels <- function(sce_object, csv_input, group_colname, gene_colname, annotations_name){ -->
<!--   # if a csv file is supplied read it and annotate. The alternative is just to pass the annotations_name which is then used to extract the annotations from the col data -->
<!--   if (grepl(".csv", csv_input)){ -->
<!--     reference <- read.csv(csv_input, sep=";") -->
<!--     unique_cell_types <- as.vector(unique(reference[group_colname])) -->
<!--     rownames(unique_cell_types) <- NULL -->
<!--     cell_type_genes <- vector("list", length(unique_cell_types[,group_colname])) -->
<!--     names(cell_type_genes) <- unique_cell_types[,group_colname] -->
<!--     for (i in 1:length(unique_cell_types[,group_colname])) { -->
<!--       cell_type_genes[[i]] <- unique(reference[reference[,group_colname] == unique_cell_types[i, group_colname], gene_colname]) -->
<!--     } -->
<!--     # GSEABase -->
<!--     kable(cell_type_genes) -->
<!--     all_sets <- lapply(names(cell_type_genes), function(x) { -->
<!--       GeneSet(cell_type_genes[[x]], setName=x)         -->
<!--     }) -->
<!--     all_sets <- GeneSetCollection(all_sets) -->
<!--     # AUCell -->
<!--     rankings <- AUCell_buildRankings(counts(sce_object), plotStats=FALSE, verbose=TRUE) -->
<!--     cell_aucs <- AUCell_calcAUC(all_sets, rankings, aucMaxRank=500) -->
<!--     results_auccell <- t(assay(cell_aucs)) -->
<!--     labels <- colnames(results_auccell)[max.col(results_auccell)] -->
<!--     # add labels to sce object -->
<!--     colData(sce_object)[annotations_name] <- labels -->
<!--     # sce_filtered$labels_blaz <- labels_blaz -->
<!--   } -->
<!--   labels <- colData(sce_object)[[annotations_name]] -->
<!--   tab <- table(Assigned=labels, Cluster=colLabels(sce_object)) -->
<!--   ###################### -->
<!--   # plot counts -->
<!--   sumdata <- data.frame(value=apply(tab, 1, sum)) -->
<!--   sumdata$key=rownames(sumdata) -->
<!--   print(ggplot(data=sumdata, aes(x=key, y=value, fill=key)) + -->
<!--     ggtitle(paste0("Annotated cell type counts for ", annotations_name, " labels")) + -->
<!--     geom_bar(colour="black", stat="identity") + -->
<!--     theme(axis.text.x=element_text(angle=50, hjust=1),)) -->
<!--   print(plot_frequencies(sce_object)) -->
<!--   #print(plotReducedDim(sce_object, dimred="UMAP", colour_by=annotations_name, text_by=annotations_name)) -->
<!--   # plot relative frequencies -->
<!--   return(sce_object) -->
<!-- } -->
<!-- ``` -->
<!-- ## Blaz {.tabset} -->
<!-- ```{r labels Blaz, fig.width=14, fig.height=8, results="asis"} -->
<!-- csv_name <- "../data/skin_tissue/marker_gene_datasets/Cell markers skin tissue sum sorted.csv" -->
<!-- group_colname <- "Cell.Type" -->
<!-- gene_colname <- "Markers" -->
<!-- annotations_name <- "labels_blaz" -->
<!-- sces_filtered <- mapply(function(idx){ -->
<!--   cat(paste0("\n\n### ", master[idx, "sample_name"], " \n\n")) -->
<!--   sce_filtered <- sces_filtered[[idx]] -->
<!--   annotate_with_other_labels(sce_filtered, csv_name, group_colname, gene_colname, annotations_name) -->
<!-- }, 1:nrow(master)) -->
<!-- ``` -->
<!-- ## Raynolds {.tabset} -->
<!-- ```{r labels Raynolds, fig.width=14, fig.height=8, results="asis"} -->
<!-- csv_name <- "../data/skin_tissue/marker_gene_datasets/Raynolds et al skin tissue n=19/cell markers reorganized BB sorted.csv" -->
<!-- group_colname <- "Cell.Type" -->
<!-- gene_colname <- "Gene" -->
<!-- annotations_name <- "labels_raynolds" -->
<!-- sces_filtered <- mapply(function(idx){ -->
<!--   cat(paste0("\n\n### ", master[idx, "sample_name"], " \n\n")) -->
<!--   sce_filtered <- sces_filtered[[idx]] -->
<!--   annotate_with_other_labels(sce_filtered, csv_name, group_colname, gene_colname, annotations_name) -->
<!-- }, 1:nrow(master)) -->
<!-- ``` -->
<!-- ## He et al. {.tabset} -->
<!-- ```{r He et al., fig.width=14, fig.height=8, results="asis"} -->
<!-- csv_name <- "../data/skin_tissue/marker_gene_datasets/He et al atopic dermatitis/He et al cell markers organized BB.csv" -->
<!-- group_colname <- "cluster_fine" -->
<!-- gene_colname <- "gene" -->
<!-- annotations_name <- "labels_he_et_al_fine" -->
<!-- sces_filtered <- mapply(function(idx){ -->
<!--   cat(paste0("\n\n### ", master[idx, "sample_name"], " \n\n")) -->
<!--   sce_filtered <- sces_filtered[[idx]] -->
<!--   annotate_with_other_labels(sce_filtered, csv_name, group_colname, gene_colname, annotations_name) -->
<!-- }, 1:nrow(master))        -->
<!-- ``` -->
<!-- ## Tabib {.tabset} -->
<!-- ```{r Tabib, fig.width=14, fig.height=8, results="asis"} -->
<!-- csv_name <- "../data/skin_tissue/marker_gene_datasets/Tabib et al healty skin n=6/Tabib 2018 cell markers sorted.csv" -->
<!-- group_colname <- "Cell.Type" -->
<!-- gene_colname <- "Marker" -->
<!-- annotations_name <- "labels_tabib" -->
<!-- sces_filtered <- mapply(function(idx){ -->
<!--   cat(paste0("\n\n### ", master[idx, "sample_name"], " \n\n")) -->
<!--   sce_filtered <- sces_filtered[[idx]] -->
<!--   annotate_with_other_labels(sce_filtered, csv_name, group_colname, gene_colname, annotations_name) -->
<!-- }, 1:nrow(master)) -->
<!-- ``` -->
<!-- ## Xue {.tabset} -->
<!-- ```{r Xue, fig.width=14, fig.height=8, results="asis"} -->
<!-- csv_name <- "../data/skin_tissue/marker_gene_datasets/Xue et al healthy skin n=10/Xue cell markers organized BB sorted.csv" -->
<!-- group_colname <- "Cell.Type" -->
<!-- gene_colname <- "Marker" -->
<!-- annotations_name <- "labels_xue" -->
<!-- sces_filtered <- mapply(function(idx){ -->
<!--   cat(paste0("\n\n### ", master[idx, "sample_name"], " \n\n")) -->
<!--   sce_filtered <- sces_filtered[[idx]] -->
<!--   annotate_with_other_labels(sce_filtered, csv_name, group_colname, gene_colname, annotations_name) -->
<!-- }, 1:nrow(master)) -->
<!-- ``` -->
<!-- # Save SCE object -->
<!-- ```{r Save RDS, class.source="fold-show"} -->
<!-- # Save all objects as one rds -->
<!-- saveRDS(sces_filtered, file="../data/fresh_data/Protocol2_all_sces_EOS1.rds") -->
<!-- # Save each SCE object individually so we can upload them to iSee -->
<!-- for (idx in 1:nrow(master)){ -->
<!--   annotation_name <- master[idx, "annotation_name"] -->
<!--   file_path <- paste0("../data/fresh_data/Protocol2_", annotation_name,"_sce_EOS1.rds") -->
<!--   saveRDS(sces_filtered[[idx]], file=file_path) -->
<!-- } -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] BiocParallel_1.28.3            dplyr_1.0.7                   
 [3] plyr_1.8.6                     RColorBrewer_1.1-2            
 [5] batchelor_1.10.0               ggplotify_0.1.0               
 [7] openxlsx_4.2.5                 tidySingleCellExperiment_1.4.0
 [9] viridis_0.6.2                  viridisLite_0.4.0             
[11] scDblFinder_1.8.0              hrbrthemes_0.8.0              
[13] AUCell_1.16.0                  GSEABase_1.56.0               
[15] graph_1.72.0                   annotate_1.72.0               
[17] XML_3.99-0.8                   AnnotationDbi_1.56.2          
[19] pheatmap_1.0.12                kableExtra_1.3.4              
[21] scran_1.22.1                   scater_1.22.0                 
[23] scuttle_1.4.0                  SingleCellExperiment_1.16.0   
[25] SummarizedExperiment_1.24.0    Biobase_2.54.0                
[27] GenomicRanges_1.46.1           GenomeInfoDb_1.30.0           
[29] IRanges_2.28.0                 S4Vectors_0.32.3              
[31] BiocGenerics_0.40.0            MatrixGenerics_1.6.0          
[33] matrixStats_0.61.0             ggplot2_3.3.5                 
[35] workflowr_1.6.2               

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                R.utils_2.11.0           
  [3] tidyselect_1.1.1          RSQLite_2.2.9            
  [5] htmlwidgets_1.5.4         grid_4.1.0               
  [7] DropletUtils_1.14.1       munsell_0.5.0            
  [9] ScaledMatrix_1.2.0        codetools_0.2-18         
 [11] statmod_1.4.36            xgboost_1.5.0.2          
 [13] withr_2.4.3               colorspace_2.0-2         
 [15] highr_0.9                 knitr_1.37               
 [17] rstudioapi_0.13           Rttf2pt1_1.3.9           
 [19] labeling_0.4.2            git2r_0.29.0             
 [21] harmony_0.1.0             GenomeInfoDbData_1.2.7   
 [23] bit64_4.0.5               farver_2.1.0             
 [25] rhdf5_2.38.0              rprojroot_2.0.2          
 [27] vctrs_0.3.8               generics_0.1.1           
 [29] xfun_0.29                 R6_2.5.1                 
 [31] ggbeeswarm_0.6.0          rsvd_1.0.5               
 [33] locfit_1.5-9.4            isoband_0.2.5            
 [35] bitops_1.0-7              rhdf5filters_1.6.0       
 [37] cachem_1.0.6              gridGraphics_0.5-1       
 [39] DelayedArray_0.20.0       assertthat_0.2.1         
 [41] promises_1.2.0.1          scales_1.1.1             
 [43] beeswarm_0.4.0            gtable_0.3.0             
 [45] beachmat_2.10.0           rlang_0.4.12             
 [47] systemfonts_1.0.3         extrafontdb_1.0          
 [49] lazyeval_0.2.2            yaml_2.2.1               
 [51] httpuv_1.6.4              extrafont_0.17           
 [53] tools_4.1.0               ellipsis_0.3.2           
 [55] jquerylib_0.1.4           Rcpp_1.0.7               
 [57] sparseMatrixStats_1.6.0   zlibbioc_1.40.0          
 [59] purrr_0.3.4               RCurl_1.98-1.5           
 [61] cowplot_1.1.1             ggrepel_0.9.1            
 [63] cluster_2.1.2             fs_1.5.2                 
 [65] magrittr_2.0.1            RSpectra_0.16-0          
 [67] data.table_1.14.2         ResidualMatrix_1.4.0     
 [69] mime_0.12                 evaluate_0.14            
 [71] xtable_1.8-4              gridExtra_2.3            
 [73] compiler_4.1.0            tibble_3.1.6             
 [75] crayon_1.4.2              R.oo_1.24.0              
 [77] htmltools_0.5.2           later_1.3.0              
 [79] tidyr_1.1.4               DBI_1.1.2                
 [81] MASS_7.3-54               Matrix_1.4-0             
 [83] cli_3.1.0                 R.methodsS3_1.8.1        
 [85] metapod_1.2.0             igraph_1.2.10            
 [87] pkgconfig_2.0.3           plotly_4.10.0            
 [89] xml2_1.3.3                svglite_2.0.0            
 [91] vipor_0.4.5               bslib_0.3.1              
 [93] dqrng_0.3.0               webshot_0.5.2            
 [95] XVector_0.34.0            rvest_1.0.2              
 [97] yulab.utils_0.0.4         stringr_1.4.0            
 [99] digest_0.6.29             RcppAnnoy_0.0.19         
[101] Biostrings_2.62.0         rmarkdown_2.11           
[103] uwot_0.1.11               edgeR_3.36.0             
[105] DelayedMatrixStats_1.16.0 gdtools_0.2.3            
[107] shiny_1.7.1               lifecycle_1.0.1          
[109] jsonlite_1.7.2            Rhdf5lib_1.16.0          
[111] BiocNeighbors_1.12.0      limma_3.50.0             
[113] fansi_0.5.0               pillar_1.6.4             
[115] lattice_0.20-45           KEGGREST_1.34.0          
[117] fastmap_1.1.0             httr_1.4.2               
[119] glue_1.6.0                zip_2.2.0                
[121] png_0.1-7                 bluster_1.4.0            
[123] bit_4.0.4                 stringi_1.7.6            
[125] sass_0.4.0                HDF5Array_1.22.1         
[127] blob_1.2.2                BiocSingular_1.10.0      
[129] memoise_2.0.1             irlba_2.3.5              </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
